diff --git a/VTKExtensions/Core/vtkCLIOptions.cxx b/VTKExtensions/Core/vtkCLIOptions.cxx
index f7ef870415..5aefd1c95a 100644
--- a/VTKExtensions/Core/vtkCLIOptions.cxx
+++ b/VTKExtensions/Core/vtkCLIOptions.cxx
@@ -52,10 +52,17 @@ class PVFormatter : public CLI::Formatter

 public:
   // overridden to add a new line before each group.
+#if (CLI11_VERSION_MAJOR > 2 || (CLI11_VERSION_MAJOR == 2 && CLI11_VERSION_MINOR >= 5))
+  std::string make_expanded(const CLI::App* sub, CLI::AppFormatMode mode) const override
+  {
+    return "\n" + Superclass::make_expanded(sub, mode);
+  }
+#else
   std::string make_expanded(const CLI::App* sub) const override
   {
     return "\n" + Superclass::make_expanded(sub);
   }
+#endif

   // overridden to ensure good word wrapping for description text.
   std::string make_option_desc(const CLI::Option* option) const override
