--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,7 +57,7 @@ FetchContent_Declare(juce
     SUBBUILD_DIR   "${FETCHCONTENT_BASE_DIR}/JUCE-Subbuild"
     BINARY_DIR     "${FETCHCONTENT_BASE_DIR}/JUCE-Build")
 
-FetchContent_Declare(juce-clap-extensions
+FetchContent_Declare(juce_clap
     GIT_REPOSITORY https://github.com/free-audio/clap-juce-extensions.git
     GIT_TAG        ${LIB_CLAP_TAG}
     GIT_CONFIG     advice.detachedHead=false)
@@ -67,7 +67,7 @@ FetchContent_Declare(chowdsp_utils
     GIT_TAG        ${LIB_CHOWDSP_UTILS_TAG}
     GIT_CONFIG     advice.detachedHead=false)
 
-FetchContent_MakeAvailable(juce juce-clap-extensions chowdsp_utils)
+FetchContent_MakeAvailable(juce juce_clap chowdsp_utils)
 
 # gui resources BinaryData via CMakeLists.txt
 add_subdirectory(src/gui/${GUI})
