diff --git a/src/Views/FirmwareReleaseView.vala b/src/Views/FirmwareReleaseView.vala
index 6efbbec4..6a121a79 100644
--- a/src/Views/FirmwareReleaseView.vala
+++ b/src/Views/FirmwareReleaseView.vala
@@ -194,7 +194,7 @@ public class About.FirmwareReleaseView : Gtk.Box {
         stack.visible_child = scrolled_window;
 
         var release_version = release.get_version ();
-        if (release.get_flags () == Fwupd.RELEASE_FLAG_IS_UPGRADE && release_version != device.get_version ()) {
+        if (release.get_flags () == Fwupd.ReleaseFlags.IS_UPGRADE && release_version != device.get_version ()) {
             update_button.label = _("Update");
             update_button.sensitive = true;
         } else {
diff --git a/src/Views/FirmwareView.vala b/src/Views/FirmwareView.vala
index 7f71fb53..9227f1a3 100644
--- a/src/Views/FirmwareView.vala
+++ b/src/Views/FirmwareView.vala
@@ -124,7 +124,7 @@ public class About.FirmwareView : Switchboard.SettingsPage {
     }
 
     private void add_device (Fwupd.Device device) {
-        if (device.has_flag (Fwupd.DEVICE_FLAG_UPDATABLE)) {
+        if (device.has_flag (Fwupd.DeviceFlags.UPDATABLE)) {
             FirmwareClient.get_upgrades.begin (fwupd_client, device.get_id (), (obj, res) => {
                 Fwupd.Release? release = null;
 
@@ -253,18 +253,16 @@ public class About.FirmwareView : Switchboard.SettingsPage {
     }
 
     private async void continue_update (Fwupd.Device device, Fwupd.Release release) {
-        var path = yield download_file (device, release.get_uri ());
+        var locations = release.get_locations ();
+        var path = yield download_file (device, locations[0]);
 
         try {
             var install_flags = Fwupd.InstallFlags.NONE;
-            if (device.has_flag (Fwupd.DEVICE_FLAG_ONLY_OFFLINE)) {
-                install_flags = Fwupd.InstallFlags.OFFLINE;
-            }
 
             if (yield FirmwareClient.install (fwupd_client, device.get_id (), path, install_flags)) {
-                if (device.has_flag (Fwupd.DEVICE_FLAG_NEEDS_REBOOT)) {
+                if (device.has_flag (Fwupd.DeviceFlags.NEEDS_REBOOT)) {
                     show_reboot_dialog ();
-                } else if (device.has_flag (Fwupd.DEVICE_FLAG_NEEDS_SHUTDOWN)) {
+                } else if (device.has_flag (Fwupd.DeviceFlags.NEEDS_SHUTDOWN)) {
                     show_shutdown_dialog ();
                 }
             }
