diff --git a/dtkwidget.cmake b/dtkwidget.cmake
index 8944646d..72cad0d0 100644
--- a/dtkwidget.cmake
+++ b/dtkwidget.cmake
@@ -50,6 +50,10 @@ find_package(PkgConfig REQUIRED)
 if("${QT_VERSION_MAJOR}" STREQUAL "5")
     find_package(Qt5 REQUIRED COMPONENTS X11Extras)
 endif()
+if(Qt${QT_VERSION_MAJOR}_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt${QT_VERSION_MAJOR} COMPONENTS GuiPrivate WidgetsPrivate PrintSupportPrivate REQUIRED)
+endif()
+
 
 file(GLOB D_HEADERS "${PROJECT_SOURCE_DIR}/include/DWidget/*")
 if("${DTK_VERSION_MAJOR}" STREQUAL "6")
