diff -ru plugins-1.0.2.orig/hdf5/HDF5IO.c plugins-1.0.2/hdf5/HDF5IO.c
--- plugins-1.0.2.orig/hdf5/HDF5IO.c	2025-12-12 22:59:38.602151298 +0100
+++ plugins-1.0.2/hdf5/HDF5IO.c	2025-12-12 22:59:58.114130363 +0100
@@ -229,7 +229,7 @@
 {
     hid_t dataSetID = H5Dopen2(self->fileHandle, datasetName, H5P_DEFAULT);
     H5O_info_t oinfo;
-#if HDF5_VERSION_MAJOR == 1 && HDF5_VERSION_MINOR >= 12
+#if (HDF5_VERSION_MAJOR == 1 && HDF5_VERSION_MINOR >= 12) || HDF5_VERSION_MAJOR > 1
     HDF5ERROR(H5Oget_info1(dataSetID, &oinfo));
 #else
     HDF5ERROR(H5Oget_info(dataSetID, &oinfo));
