diff --git a/src/application/CMakeLists.txt b/src/application/CMakeLists.txt
index 2379c57..49b3344 100755
--- a/src/application/CMakeLists.txt
+++ b/src/application/CMakeLists.txt
@@ -42,7 +42,7 @@ find_package(DevIL COMPONENTS IL ILU) # yields IL_FOUND, IL_LIBRARIES, IL_INCLUD
 if(PopSift_BOOST_USE_STATIC_LIBS)
   set(Boost_USE_STATIC_LIBS ON)
 endif()
-find_package(Boost 1.71.0 REQUIRED COMPONENTS filesystem program_options system)
+find_package(Boost 1.71.0 REQUIRED COMPONENTS filesystem program_options)
 if(WIN32)
   add_definitions("-DBOOST_ALL_NO_LIB")
 endif(WIN32)
