From 5df2899a72b09c4ae341befbe5b7857022db64fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sosth=C3=A8ne=20Gu=C3=A9don?= <sosthene@nitrokey.com>
Date: Mon, 19 Feb 2024 09:35:32 +0100
Subject: [PATCH] Add `#[ignore]` for tests requiring network access

---
 pkcs11/src/backend/session.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pkcs11/src/backend/session.rs b/pkcs11/src/backend/session.rs
index c03cc0f..fcf8d7d 100644
--- a/pkcs11/src/backend/session.rs
+++ b/pkcs11/src/backend/session.rs
@@ -701,6 +701,7 @@ mod test {
     use super::*;
 
     #[test]
+    #[ignore]
     fn parrallel_fetch_all_keys() {
         init_for_tests();
         let slot = get_slot(0).unwrap();
@@ -741,6 +742,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
     fn parrallel_fetch_all_keys_fail() {
         THREADS_ALLOWED.store(false, Ordering::Relaxed);
         init_for_tests();
