--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,11 +62,7 @@ set(LOADTRK_SOURCES
     src/sdl_sound.cpp
     src/sdl_gui.cpp
     src/sdl_io.cpp
-    src/io_helpers.cpp
-    src/rtmidi/rtmidi_c.h
-    src/rtmidi/rtmidi_c.cpp
-    src/rtmidi/RtMidi.h
-    src/rtmidi/RtMidi.cpp)
+    src/io_helpers.cpp)
 
 if(MINGW)
     list(APPEND LOADTRK_SOURCES
@@ -197,6 +193,11 @@ endif()
 
 find_package(PkgConfig REQUIRED)
 
+if (use_midi)
+    pkg_check_modules(rtmidi REQUIRED IMPORTED_TARGET rtmidi)
+    target_link_libraries(loadtrk PkgConfig::rtmidi)
+endif()
+
 pkg_check_modules(EXSID IMPORTED_TARGET libexsid>=2.0)
 if (EXSID_FOUND)
     target_compile_definitions(loadtrk PUBLIC USE_EXSID)
