diff --git a/acquisition/acquisition.go b/acquisition/acquisition.go
index 446b5a6..8a8f041 100644
--- a/acquisition/acquisition.go
+++ b/acquisition/acquisition.go
@@ -17,7 +17,6 @@ import (
 	"time"
 
 	"github.com/botherder/go-savetime/hashes"
-	rt "github.com/botherder/go-savetime/runtime"
 	"github.com/google/uuid"
 	"github.com/mvt-project/androidqf/adb"
 	"github.com/mvt-project/androidqf/log"
@@ -51,7 +50,7 @@ func New(path string) (*Acquisition, error) {
 	}
 
 	if path == "" {
-		acq.StoragePath = filepath.Join(rt.GetExecutableDirectory(), acq.UUID)
+		acq.StoragePath = acq.UUID
 	} else {
 		acq.StoragePath = path
 	}
