diff --git a/CMakeLists.txt b/CMakeLists.txt
index 918e2f1..3c24d73 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,6 +60,10 @@ set(QML_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/qt${QT_VERSION_MAJOR}/qml" CACHE ST
 set(USE_QQuickStylePluginPrivate OFF)
 find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core)
 find_package(Qt${QT_VERSION_MAJOR}QuickControls2)
+if (Qt${QT_VERSION_MAJOR}_VERSION VERSION_GREATER_EQUAL 6.10)
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS CorePrivate QuickPrivate QuickControls2Private REQUIRED)
+endif()
+
 if(EnableQt5)
 if(TARGET Qt::QuickControls2 AND TARGET Qt::QuickControls2Private)
     set(USE_QQuickStylePluginPrivate ON)
