diff --git a/src/lcalc/Lcommandline.cc b/src/lcalc/Lcommandline.cc
index 0abd433..0c9617f 100644
--- a/src/lcalc/Lcommandline.cc
+++ b/src/lcalc/Lcommandline.cc
@@ -479,7 +479,7 @@ int main (int argc, char *argv[])
 
             t2=.5; //t2=.5 because of the GAMMA(s+1/2)
 
-            pari_init_opts(400000000,2,INIT_DFTm); // the last option is to prevent
+            pari_init_opts(400000000,1UL << 20,INIT_DFTm); // the last option is to prevent
             //pari from giving its interrupt signal when its elliptic curve a_p
             //algorithm is called and interrupted with ctrl-c.
 
diff --git a/src/libLfunction/Lelliptic.cc b/src/libLfunction/Lelliptic.cc
index e593934..214b5c8 100644
--- a/src/libLfunction/Lelliptic.cc
+++ b/src/libLfunction/Lelliptic.cc
@@ -60,17 +60,12 @@ void data_E(char *a1, char *a2, char *a3, char *a4, char *a6, int N_terms,Double
 
 
     F = cgetg(6, t_VEC);
-    F[1] = (long) cgeti(BIGDEFAULTPREC);
-    F[2] = (long) cgeti(BIGDEFAULTPREC);
-    F[3] = (long) cgeti(BIGDEFAULTPREC);
-    F[4] = (long) cgeti(BIGDEFAULTPREC);
-    F[5] = (long) cgeti(BIGDEFAULTPREC);
-
-    gaffect(gp_read_str(a1), (GEN) F[1]);
-    gaffect(gp_read_str(a2), (GEN) F[2]);
-    gaffect(gp_read_str(a3), (GEN) F[3]);
-    gaffect(gp_read_str(a4), (GEN) F[4]);
-    gaffect(gp_read_str(a6), (GEN) F[5]);
+
+    gel(F, 1) = gp_read_str(a1);
+    gel(F, 2) = gp_read_str(a2);
+    gel(F, 3) = gp_read_str(a3);
+    gel(F, 4) = gp_read_str(a4);
+    gel(F, 5) = gp_read_str(a6);
 
     E = ellinit(F, NULL, BIGDEFAULTPREC);
 
@@ -89,7 +84,7 @@ void data_E(char *a1, char *a2, char *a3, char *a4, char *a6, int N_terms,Double
     }
 
 
-    gaffsg(1, (GEN) y);
+    affsi(1, y);
     sign = ellrootno(E,y); //sign of the functional equation
 
 
@@ -100,7 +95,7 @@ void data_E(char *a1, char *a2, char *a3, char *a4, char *a6, int N_terms,Double
         if(isprime(n)){
 
             p=n;
-            gaffsg(p,y);
+            affsi(p,y);
             coeff[p] = Double(1.*llrint(gtodouble(ellap(E,y))))/sqrt(Double(1.*p));
 
             if(gtolong(gmod((GEN) E[12],(GEN) y))==0) // if p|discriminant, i.e. bad reduction
