diff --git a/third_party/libwebrtc/third_party/crc32c/src/.gitmodules b/third_party/libwebrtc/third_party/crc32c/src/.gitmodules
new file mode 100644
index 0000000..888541f
--- /dev/null
+++ b/third_party/libwebrtc/third_party/crc32c/src/.gitmodules
@@ -0,0 +1,9 @@
+[submodule "third_party/googletest"]
+	path = third_party/googletest
+	url = https://github.com/google/googletest.git
+[submodule "third_party/glog"]
+	path = third_party/glog
+	url = https://github.com/google/glog.git
+[submodule "third_party/benchmark"]
+	path = third_party/benchmark
+	url = https://github.com/google/benchmark.git
diff --git a/third_party/rust/cubeb-sys/libcubeb/.gitmodules b/third_party/rust/cubeb-sys/libcubeb/.gitmodules
new file mode 100644
index 0000000..41442f2
--- /dev/null
+++ b/third_party/rust/cubeb-sys/libcubeb/.gitmodules
@@ -0,0 +1,14 @@
+[submodule "googletest"]
+	path = googletest
+	url = https://github.com/google/googletest
+[submodule "cmake/sanitizers-cmake"]
+	path = cmake/sanitizers-cmake
+	url = https://github.com/arsenm/sanitizers-cmake
+[submodule "src/cubeb-coreaudio-rs"]
+	path = src/cubeb-coreaudio-rs
+	url = https://github.com/mozilla/cubeb-coreaudio-rs
+	branch = trailblazer
+[submodule "src/cubeb-pulse-rs"]
+	path = src/cubeb-pulse-rs
+	url = https://github.com/mozilla/cubeb-pulse-rs
+	branch = dev
diff --git a/third_party/rust/glslopt/.gitmodules b/third_party/rust/glslopt/.gitmodules
new file mode 100644
index 0000000..415eebb
--- /dev/null
+++ b/third_party/rust/glslopt/.gitmodules
@@ -0,0 +1,3 @@
+[submodule "glsl-optimizer"]
+	path = glsl-optimizer
+	url = https://github.com/jamienicol/glsl-optimizer.git
diff --git a/third_party/rust/minimal-lexical/.gitmodules b/third_party/rust/minimal-lexical/.gitmodules
new file mode 100644
index 0000000..f06dee0
--- /dev/null
+++ b/third_party/rust/minimal-lexical/.gitmodules
@@ -0,0 +1,4 @@
+[submodule "data/test-parse-golang/parse-number-fxx-test-data"]
+	path = data/test-parse-golang/parse-number-fxx-test-data
+	url = https://github.com/nigeltao/parse-number-fxx-test-data
+	shallow = true
diff --git a/third_party/rust/sfv/.gitmodules b/third_party/rust/sfv/.gitmodules
new file mode 100644
index 0000000..cedc6d1
--- /dev/null
+++ b/third_party/rust/sfv/.gitmodules
@@ -0,0 +1,3 @@
+[submodule "tests/spec_tests"]
+	path = tests/spec_tests
+	url = https://github.com/httpwg/structured-header-tests.git
diff --git a/third_party/rust/wasi/.gitmodules b/third_party/rust/wasi/.gitmodules
new file mode 100644
index 0000000..e69de29
diff --git a/third_party/rust/yaml-rust2/.gitmodules b/third_party/rust/yaml-rust2/.gitmodules
new file mode 100644
index 0000000..cbc1e88
--- /dev/null
+++ b/third_party/rust/yaml-rust2/.gitmodules
@@ -0,0 +1,3 @@
+[submodule "tests/yaml-test-suite"]
+	path = tests/yaml-test-suite
+	url = https://github.com/yaml/yaml-test-suite/
