diff --git i/src/lib.rs w/src/lib.rs
index d369a70..24cffb3 100644
--- i/src/lib.rs
+++ w/src/lib.rs
@@ -1021,13 +1021,13 @@ unsafe fn symbolsrccfg_to_rust(cfg: *const sym_src_cfg, cfg_len: u32) -> Option<
         match (*c).src_type {
             blazesym_src_type::SRC_T_ELF => {
                 cfg_rs.push(SymbolSrcCfg::Elf {
-                    file_name: from_cstr((*c).params.elf.file_name),
-                    base_address: (*c).params.elf.base_address,
+                    file_name: from_cstr((&(*c).params.elf).file_name),
+                    base_address: (&(*c).params.elf).base_address,
                 });
             }
             blazesym_src_type::SRC_T_KERNEL => {
-                let kallsyms = (*c).params.kernel.kallsyms;
-                let kernel_image = (*c).params.kernel.kernel_image;
+                let kallsyms = (&(*c).params.kernel).kallsyms;
+                let kernel_image = (&(*c).params.kernel).kernel_image;
                 cfg_rs.push(SymbolSrcCfg::Kernel {
                     kallsyms: if !kallsyms.is_null() {
                         Some(from_cstr(kallsyms))
@@ -1043,8 +1043,8 @@ unsafe fn symbolsrccfg_to_rust(cfg: *const sym_src_cfg, cfg_len: u32) -> Option<
             }
             blazesym_src_type::SRC_T_PROCESS => {
                 cfg_rs.push(SymbolSrcCfg::Process {
-                    pid: if (*c).params.process.pid > 0 {
-                        Some((*c).params.process.pid)
+                    pid: if (&(*c).params.process).pid > 0 {
+                        Some((&(*c).params.process).pid)
                     } else {
                         None
                     },
