From 82e75787f7f622caac43b8cfcb30f50f97684d18 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 15 Oct 2024 13:21:01 +0000
Subject: [PATCH] Translate application name

---
 src/main.rs | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/main.rs b/src/main.rs
index 3b5e32e..d1efca0 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,4 +1,4 @@
-use gettextrs::{bindtextdomain, setlocale, textdomain, LocaleCategory};
+use gettextrs::{bindtextdomain, gettext, setlocale, textdomain, LocaleCategory};
 use gtk::{gio, glib};
 mod application;
 mod config;
@@ -10,7 +10,6 @@ use config::{GETTEXT_PACKAGE, LOCALEDIR, RESOURCES_FILE};
 fn main() -> glib::ExitCode {
     pretty_env_logger::init();
 
-    glib::set_application_name("Obfuscate");
     gtk::init().expect("Unable to start GTK");
     // Prepare i18n
     setlocale(LocaleCategory::LcAll, "");
@@ -20,5 +19,7 @@ fn main() -> glib::ExitCode {
     let res = gio::Resource::load(RESOURCES_FILE).expect("Could not load resources");
     gio::resources_register(&res);
 
+    glib::set_application_name(&gettext("Obfuscate"));
+
     Application::run()
 }
-- 
GitLab

