--- a/src/libcmd/markdown.cc
+++ b/src/libcmd/markdown.cc
@@ -39,7 +39,7 @@ static std::string doRenderMarkdownToTerminal(std::string_view markdown)
         .feat = LOWDOWN_COMMONMARK | LOWDOWN_FENCED | LOWDOWN_DEFLIST | LOWDOWN_TABLES,
         .oflags =
 #  if HAVE_LOWDOWN_1_4
-            LOWDOWN_TERM_NORELLINK // To render full links while skipping relative ones
+            LOWDOWN_NORELLINK // To render full links while skipping relative ones
 #  else
             LOWDOWN_TERM_NOLINK
 #  endif
