From d411cd85d0ea87cd3dc0bcd57c3179919e47bf19 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 17 Oct 2024 02:23:01 +0200
Subject: [PATCH] window: Set icon name

---
 src/window.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/window.py b/src/window.py
index ee20773..4ac2e31 100644
--- a/src/window.py
+++ b/src/window.py
@@ -100,5 +100,6 @@ def open_uri(link: str):
     convertion.connect("converted", converted)
     load_convertion_page(from_currency_value)
     window.set_application(application)
+    window.set_icon_name(application.get_application_id())
     window.load_convertion_page = load_convertion_page
     return window
