From f448cee4c6a1fecf779aeb24c519948ad71e64cb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 17 Oct 2024 08:51:16 +0200
Subject: [PATCH] about: Set license

---
 src/about.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/about.py b/src/about.py
index 0c242d0..085bbc5 100644
--- a/src/about.py
+++ b/src/about.py
@@ -55,5 +55,6 @@ def get_flatpak_info():
         designers=['Brage Fuglseth https://bragefuglseth.dev'],
         issue_url='https://github.com/ideveCore/valuta/issues',
         debug_info=debug_info,
+        license_type=Gtk.License.GPL_3_0,
         copyright='© 2023 Ideve Core'
     )
