From a20209d990a3c92a28e993a2e25a1c62e03dfdc8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 17 Oct 2024 02:13:57 +0200
Subject: [PATCH] requests: Fix regular expressions

---
 src/requests.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/requests.py b/src/requests.py
index 3d661a0..72db677 100644
--- a/src/requests.py
+++ b/src/requests.py
@@ -83,10 +83,10 @@ def mount_url(self):
     def serializer(self, data: bytes):
         data = data.decode('utf-8', errors="replace")
         try:
-            results = re.findall(f'[\d*\,]*\.\d* {CODES[self.to_currency]["name"]}', data)
+            results = re.findall(rf'[\d*\,]*\.\d* {CODES[self.to_currency]["name"]}', data)
             if results.__len__() > 0:
                 converted_amount_str = results[0]
-                converted_currency = re.findall('[\d*\,]*\.\d*', converted_amount_str)[0]
+                converted_currency = re.findall(r'[\d*\,]*\.\d*', converted_amount_str)[0]
                 return self.default_response({
                     "amount": converted_currency,
                     "converted": True,
