--- a/installers/debian/etc/openbve/filesystem.cfg
+++ b/installers/debian/etc/openbve/filesystem.cfg
@@ -4,5 +4,5 @@ ManagedContent = $[Personal]/.local/share/OpenBve/ManagedContent
 Settings = $[ApplicationData]/OpenBve/Settings
 InitialRoute = $[CommonApplicationData]/games/bve/Railway/Route/Birmingham_Cross-City_South
 InitialTrain = $[CommonApplicationData]/games/bve/Train/
-RestartProcess = /usr/games/openbve
+RestartProcess = /usr/bin/openbve
 #RestartArguments = /filesystem=/etc/openbve/filesystem.cfg
