diff --git a/Cargo.toml b/Cargo.toml
index 9c24ece..497385a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,6 +54,9 @@ zbus = { version = "4.2.1", default-features = false, features = ["tokio"] }
 unicode-truncate = "1.0.0"
 unicode-width = "0.1.11"

+[profile.release]
+lto = "thin"
+
 # [patch."https://github.com/pop-os/launcher/"]
 # pop-launcher = { git = "https://github.com/wiiznokes/launcher/", rev = "86a54d54a68b832d404ef83a03af84cf9ef3e694" }
 # pop-launcher-service = { git = "https://github.com/wiiznokes/launcher/", rev = "86a54d54a68b832d404ef83a03af84cf9ef3e694" }
