diff --git a/libpolys/coeffs/coeffs.h b/libpolys/coeffs/coeffs.h
index d4de65f45..184ad9189 100644
--- a/libpolys/coeffs/coeffs.h
+++ b/libpolys/coeffs/coeffs.h
@@ -822,7 +822,7 @@ static FORCE_INLINE BOOLEAN nCoeff_is_Z(const coeffs r)
 }
 
 static FORCE_INLINE BOOLEAN nCoeff_is_Zn(const coeffs r)
-{ assume(r != NULL); return (getCoeffType(r)==n_Zn)&& (r->is_field==0); }
+{ assume(r != NULL); return getCoeffType(r)==n_Zn; }
 
 static FORCE_INLINE BOOLEAN nCoeff_is_Q_or_BI(const coeffs r)
 { assume(r != NULL); return getCoeffType(r)==n_Q; }
