diff -wbBur notion/libtu/objp.h notion.q/libtu/objp.h
--- notion/libtu/objp.h	2025-06-11 21:55:15.632474228 +0300
+++ notion.q/libtu/objp.h	2025-06-11 21:54:59.312474402 +0300
@@ -26,7 +26,7 @@
     ClassDescr *ancestor;
     int funtab_n;
     DynFunTab *funtab;
-    void (*destroy_fn)();
+    void (*destroy_fn)(Obj *obj);
 };
 
 #define OBJ_TYPESTR(OBJ) ((OBJ) ? ((Obj*)OBJ)->obj_type->name : NULL)
