diff -ru nageru-2.3.2.orig/nageru/audio_encoder.cpp nageru-2.3.2/nageru/audio_encoder.cpp
--- nageru-2.3.2.orig/nageru/audio_encoder.cpp	2026-08-05 20:13:21.241836702 +0200
+++ nageru-2.3.2/nageru/audio_encoder.cpp	2026-08-05 20:18:38.294511613 +0200
@@ -41,7 +41,9 @@
 	ctx = avcodec_alloc_context3(codec);
 	ctx->bit_rate = bit_rate;
 	ctx->sample_rate = OUTPUT_FREQUENCY;
-	ctx->sample_fmt = codec->sample_fmts[0];
+	const enum AVSampleFormat *codec_sample_fmts = nullptr;
+	avcodec_get_supported_config(nullptr, codec, AV_CODEC_CONFIG_SAMPLE_FORMAT, 0, (const void**)&codec_sample_fmts, nullptr);
+	ctx->sample_fmt = codec_sample_fmts[0];
 	ctx->ch_layout.order = AV_CHANNEL_ORDER_NATIVE;
 	ctx->ch_layout.nb_channels = 2;
 	ctx->ch_layout.u.mask = AV_CH_LAYOUT_STEREO;
