From 3b2e9d947ac8937392efc5263056d55c4c7dc7d2 Mon Sep 17 00:00:00 2001
From: "Steinar H. Gunderson" <sgunderson@bigfoot.com>
Date: Fri, 7 Mar 2025 21:29:10 +0100
Subject: [PATCH] Fix compilation with SVT-AV1 3.0.0.

---
 nageru/av1_encoder.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/nageru/av1_encoder.cpp b/nageru/av1_encoder.cpp
index bb6c17b..f437cda 100644
--- a/nageru/av1_encoder.cpp
+++ b/nageru/av1_encoder.cpp
@@ -140,7 +140,11 @@ void AV1Encoder::init_av1()
 	// so we do that.
 	memset(&config, 0, sizeof(config));
 
+#if SVT_AV1_CHECK_VERSION(3, 0, 0)
+	EbErrorType ret = svt_av1_enc_init_handle(&encoder, &config);
+#else
 	EbErrorType ret = svt_av1_enc_init_handle(&encoder, nullptr, &config);
+#endif
 	if (ret != EB_ErrorNone) {
 		fprintf(stderr, "Error initializing SVT-AV1 handle (error %08x)\n", ret);
 		exit(EXIT_FAILURE);
@@ -297,12 +301,14 @@ void AV1Encoder::encode_frame(AV1Encoder::QueuedFrame qf)
 		pic.y_stride = global_flags.width;  // In pixels, so no bytes_per_pixel.
 		pic.cb_stride = global_flags.width / 2;  // Likewise.
 		pic.cr_stride = global_flags.width / 2;  // Likewise.
+#if !SVT_AV1_CHECK_VERSION(3, 0, 0)
 		pic.width = global_flags.width;
 		pic.height = global_flags.height;
 		pic.org_x = 0;
 		pic.org_y = 0;
 		pic.color_fmt = EB_YUV420;
 		pic.bit_depth = global_flags.bit_depth > 8 ? EB_TEN_BIT : EB_EIGHT_BIT;
+#endif
 
 		EbBufferHeaderType hdr;
 		hdr.p_buffer      = reinterpret_cast<uint8_t *>(&pic);
-- 
2.39.5

