diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 71086b0..ab99cf0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,6 +3,9 @@ if("${QT_VERSION_MAJOR}" STREQUAL "6")
 else()
     find_package(Qt5 REQUIRED COMPONENTS WaylandClient XkbCommonSupport)
 endif()
+if(Qt${QT_VERSION_MAJOR}_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt${QT_VERSION_MAJOR} COMPONENTS CorePrivate GuiPrivate WaylandClientPrivate REQUIRED)
+endif()
 
 add_library(${LIB_NAME} SHARED)
 
diff --git a/src/kernel/dguiapplicationhelper.cpp b/src/kernel/dguiapplicationhelper.cpp
index abf55ce..44cc334 100644
--- a/src/kernel/dguiapplicationhelper.cpp
+++ b/src/kernel/dguiapplicationhelper.cpp
@@ -392,7 +392,11 @@ void DGuiApplicationHelperPrivate::notifyAppThemeChanged()
 
 void DGuiApplicationHelperPrivate::notifyAppThemeChangedByEvent()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 10, 0)
+    QWindowSystemInterfacePrivate::ThemeChangeEvent event;
+#else
     QWindowSystemInterfacePrivate::ThemeChangeEvent event(nullptr);
+#endif
     // 此事件会促使QGuiApplication重新从QPlatformTheme中获取系统级别的QPalette.
     // 而在deepin平台下, 系统级别的QPalette来源自 \a applicationPalette()
     QGuiApplicationPrivate::processThemeChanged(&event);
diff --git a/src/plugins/platform/config.tests/treeland_test/CMakeLists.txt b/src/plugins/platform/config.tests/treeland_test/CMakeLists.txt
index 4c12e76..4622006 100644
--- a/src/plugins/platform/config.tests/treeland_test/CMakeLists.txt
+++ b/src/plugins/platform/config.tests/treeland_test/CMakeLists.txt
@@ -15,6 +15,9 @@ find_package(Qt${QT_VERSION_MAJOR} ${REQUIRED_QT_VERSION} REQUIRED COMPONENTS Co
 if(${QT_VERSION_MAJOR} STREQUAL "5")
     find_package(Qt5 REQUIRED COMPONENTS XkbCommonSupport)
 endif()
+if(Qt${QT_VERSION_MAJOR}_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt${QT_VERSION_MAJOR} COMPONENTS CorePrivate GuiPrivate WaylandClientPrivate REQUIRED)
+endif()
 
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(WAYLAND_CLIENT REQUIRED wayland-client)
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 0782182..61bbd3a 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -2,6 +2,9 @@ set(BIN_NAME ut-DtkGui)
 
 find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Test)
 find_package(GTest REQUIRED)
+if(Qt${QT_VERSION_MAJOR}_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt${QT_VERSION_MAJOR} COMPONENTS GuiPrivate REQUIRED)
+endif()
 
 file(GLOB test_SRC
     res.qrc
diff --git a/tests/platform-plugin-test/CMakeLists.txt b/tests/platform-plugin-test/CMakeLists.txt
index 2b3b741..138447f 100644
--- a/tests/platform-plugin-test/CMakeLists.txt
+++ b/tests/platform-plugin-test/CMakeLists.txt
@@ -5,6 +5,9 @@ find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Gui)
 if("${QT_VERSION_MAJOR}" STRLESS "6")
     find_package(Qt${QT_VERSION_MAJOR} REQUIRED EventDispatcherSupport ServiceSupport)
 endif()
+if(Qt${QT_VERSION_MAJOR}_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt${QT_VERSION_MAJOR} COMPONENTS CorePrivate GuiPrivate REQUIRED)
+endif()
 
 find_package(Dtk${DTK_VERSION_MAJOR}Core REQUIRED)
 
