diff --git a/treetop.gemspec b/treetop.gemspec
index e2c8a78..1f0e64b 100644
--- a/treetop.gemspec
+++ b/treetop.gemspec
@@ -10,7 +10,7 @@ Gem::Specification.new do |spec|
   spec.authors = ["Nathan Sobo", "Clifford Heath"]
 
   spec.email = "cliffordheath@gmail.com"
-  spec.date = Date.today.strftime("%F")
+  spec.date = Date.parse(`git show -s --format=%ci`)
   spec.summary = "A Ruby-based text parsing and interpretation DSL"
   spec.description = "A Parsing Expression Grammar (PEG) Parser generator DSL for Ruby"
   spec.homepage = "https://github.com/cjheath/treetop"
