--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -274,7 +274,7 @@ elseif(WIN32)
 	install(FILES license.txt DESTINATION .)
 elseif(UNIX)
 	# Install the binary.
-	install(TARGETS EndlessSky CONFIGURATIONS Release RUNTIME DESTINATION games)
+	install(TARGETS EndlessSky CONFIGURATIONS Release RUNTIME)
 
 	# Install the desktop file.
 	install(FILES io.github.endless_sky.endless_sky.desktop DESTINATION share/applications)
