diff --git i/Makefile w/Makefile
index b816bd6..02e8544 100644
--- i/Makefile
+++ w/Makefile
@@ -127,7 +127,7 @@ install:
 	fi
 
 	install -Dm755 packaging/files/usr/libexec/gpclient/vpnc-script $(DESTDIR)/usr/libexec/gpclient/vpnc-script
-	install -Dm755 packaging/files/usr/libexec/gpclient/hipreport.sh $(DESTDIR)/usr/libexec/gpclient/hipreport.sh
+	install -Dm755 packaging/files/usr/libexec/gpclient/hipreport.sh $(DESTDIR)/usr/lib/gpclient/hipreport.sh
 
 	# Install the disconnect hooks
 	install -Dm755 packaging/files/usr/lib/NetworkManager/dispatcher.d/pre-down.d/gpclient.down $(DESTDIR)/usr/lib/NetworkManager/dispatcher.d/pre-down.d/gpclient.down
@@ -150,7 +150,7 @@ uninstall:
 	rm -f $(DESTDIR)/usr/bin/gpgui
 
 	rm -f $(DESTDIR)/usr/libexec/gpclient/vpnc-script
-	rm -f $(DESTDIR)/usr/libexec/gpclient/hipreport.sh
+	rm -f $(DESTDIR)/usr/lib/gpclient/hipreport.sh
 
 	rm -f $(DESTDIR)/usr/lib/NetworkManager/dispatcher.d/pre-down.d/gpclient.down
 	rm -f $(DESTDIR)/usr/lib/NetworkManager/dispatcher.d/gpclient-nm-hook
diff --git i/crates/openconnect/src/vpn_utils.rs w/crates/openconnect/src/vpn_utils.rs
index e5f81c2..e401427 100644
--- i/crates/openconnect/src/vpn_utils.rs
+++ w/crates/openconnect/src/vpn_utils.rs
@@ -19,6 +19,7 @@ const VPNC_SCRIPT_LOCATIONS: &[&str] = &[
 ];
 
 const CSD_WRAPPER_LOCATIONS: &[&str] = &[
+  "/usr/lib/gpclient/hipreport.sh",
   "/usr/libexec/gpclient/hipreport.sh",
   #[cfg(target_arch = "x86_64")]
   "/usr/lib/x86_64-linux-gnu/openconnect/hipreport.sh",
