diff --git i/spyder/api/plugins/tests.py w/spyder/api/plugins/tests.py
index f741c864b2..5d7c577b97 100644
--- i/spyder/api/plugins/tests.py
+++ w/spyder/api/plugins/tests.py
@@ -126,10 +126,10 @@ def register_plugin(main_window_mock):
 
         request._fixturemanager._arg2fixturedefs[fixture_name] = [
             FixtureDef(
+                config=request.config,
                 argname=fixture_name,
                 func=register_plugin_factory(plugin_cls),
                 scope="session",
-                fixturemanager=request._fixturemanager,
                 baseid=request.node.nodeid,
                 params=None,
             )
