diff --git a/src/backend/xrender/xrender.c b/src/backend/xrender/xrender.c
index b2ec3f2..d7b41c8 100644
--- a/src/backend/xrender/xrender.c
+++ b/src/backend/xrender/xrender.c
@@ -624,16 +624,13 @@ static void present(backend_t *base, const region_t *region) {
 		                     XCB_NONE, xd->back[xd->curr_back], orig_x, orig_y, 0,
 		                     0, orig_x, orig_y, region_width, region_height);
 
-		auto xregion = x_create_region(base->c, region);
-
 		// Make sure we got reply from PresentPixmap before waiting for events,
 		// to avoid deadlock
 		auto e = xcb_request_check(
 		    base->c->c, xcb_present_pixmap_checked(
 		                    xd->base.c->c, xd->target_win,
-		                    xd->back_pixmap[xd->curr_back], 0, XCB_NONE, xregion, 0,
+		                    xd->back_pixmap[xd->curr_back], 0, XCB_NONE, XCB_NONE, 0,
 		                    0, XCB_NONE, XCB_NONE, XCB_NONE, 0, 0, 0, 0, 0, NULL));
-		x_destroy_region(base->c, xregion);
 		if (e) {
 			log_error("Failed to present pixmap");
 			free(e);
