--- a/Make.defaults
+++ b/Make.defaults
@@ -260,7 +260,7 @@ CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra -Werror \
            -fno-strict-aliasing \
            -ffreestanding -fno-stack-protector
 else
-CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra -Wno-pointer-sign -Werror \
+CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra -Wno-pointer-sign \
            -fno-strict-aliasing \
            -ffreestanding -fno-stack-protector -fno-stack-check \
            $(if $(findstring 0,$(USING_CLANG)),-Wno-error=maybe-uninitialized,) \
