From 19ba46f3433c14f75724ebc49918d4a6c28d85b9 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sat, 10 Aug 2024 03:30:01 +0200
Subject: [PATCH 02/12] Disable submodule fetching

---
 libraries/cmake/source/modules/utils.cmake | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/libraries/cmake/source/modules/utils.cmake b/libraries/cmake/source/modules/utils.cmake
index d5270b8f9..467cb61c8 100644
--- a/libraries/cmake/source/modules/utils.cmake
+++ b/libraries/cmake/source/modules/utils.cmake
@@ -138,17 +138,6 @@ function(importSourceSubmodule)
 
   set(directory_path "${CMAKE_SOURCE_DIR}/libraries/cmake/source/${ARGS_NAME}")
 
-  foreach(submodule_name ${ARGS_SUBMODULES} ${ARGS_SHALLOW_SUBMODULES})
-    list(FIND ARGS_SHALLOW_SUBMODULES "${submodule_name}" shallow_clone)
-    if(${shallow_clone} EQUAL -1)
-      set(shallow_clone false)
-    else()
-      set(shallow_clone true)
-    endif()
-
-    initializeGitSubmodule("${directory_path}/${submodule_name}" ${ARGS_NO_RECURSIVE} ${shallow_clone})
-  endforeach()
-
   foreach(submodule_to_patch ${ARGS_PATCH})
     set(patched_source_dir "${CMAKE_BINARY_DIR}/libs/src/patched-source/${ARGS_NAME}/${submodule_to_patch}")
 
-- 
2.46.0

