From 34b1f469a782dbc72521c6a659fe7be07b43b697 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sat, 10 Aug 2024 12:13:51 +0200
Subject: [PATCH 03/12] Remove incompatible build flags

---
 cmake/flags.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cmake/flags.cmake b/cmake/flags.cmake
index e0896f72b..1abae4284 100644
--- a/cmake/flags.cmake
+++ b/cmake/flags.cmake
@@ -84,11 +84,9 @@ function(setupBuildFlags)
       -Woverloaded-virtual
       -Wnon-virtual-dtor
       -Weffc++
-      -stdlib=libc++
     )
 
     set(posix_cxx_link_options
-      -stdlib=libc++
       -ldl
     )
 
@@ -179,7 +177,6 @@ function(setupBuildFlags)
       )
 
       set(linux_cxx_link_libraries
-        c++abi.a
         rt
         dl
       )
-- 
2.46.0

