From ac1f8e00f3097984da790260e436bd4cf1c374e9 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sat, 10 Aug 2024 17:56:58 +0200
Subject: [PATCH 10/12] Do not build utils requiring libdpkg

---
 osquery/utils/linux/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/osquery/utils/linux/CMakeLists.txt b/osquery/utils/linux/CMakeLists.txt
index fb2963619..8ba140a15 100644
--- a/osquery/utils/linux/CMakeLists.txt
+++ b/osquery/utils/linux/CMakeLists.txt
@@ -27,7 +27,7 @@ function(generateOsqueryUtilsLinux)
     osquery_cxx_settings
     osquery_utils_expected
     osquery_filesystem
-    thirdparty_libdpkg
+    # thirdparty_libdpkg
   )
 
   generateIncludeNamespace(osquery_utils_linux "osquery/utils/linux" "FILE_ONLY" ${public_header_files})
-- 
2.46.0

