From 82bc9f31955bf6a4ec9f641f7d0084e459bb7b45 Mon Sep 17 00:00:00 2001
From: Tom Poole <tom@juce.com>
Date: Fri, 20 Oct 2023 10:00:57 +0100
Subject: [PATCH] NetworkGraphicsDemo: Fix a complier error

---
 extras/NetworkGraphicsDemo/Source/Demos.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/NetworkGraphicsDemo/Source/Demos.h b/extras/NetworkGraphicsDemo/Source/Demos.h
index c996a364945..839d29961fb 100644
--- a/extras/NetworkGraphicsDemo/Source/Demos.h
+++ b/extras/NetworkGraphicsDemo/Source/Demos.h
@@ -64,7 +64,7 @@ struct GridLines final : public AnimatedContent
 };
 
 //==============================================================================
-struct BackgroundLogo final : public AnimatedContent
+struct BackgroundLogo : public AnimatedContent
 {
     BackgroundLogo()
     {
