diff -ruN libretro-blastem.orig/Makefile libretro-blastem/Makefile
--- libretro-blastem.orig/Makefile	2026-09-04 08:44:07.381267911 +0200
+++ libretro-blastem/Makefile	2026-09-04 08:44:07.381267911 +0200
@@ -12,6 +12,9 @@
 FIXUP:=true
 Z80_DISPATCH:=goto
 
+CFLAGS:=$(CFLAGS)
+LDFLAGS:=$(LDFLAGS)
+
 BUNDLED_LIBZ:=zlib/adler32.o zlib/compress.o zlib/crc32.o zlib/deflate.o zlib/gzclose.o zlib/gzlib.o zlib/gzread.o\
 	zlib/gzwrite.o zlib/infback.o zlib/inffast.o zlib/inflate.o zlib/inftrees.o zlib/trees.o zlib/uncompr.o zlib/zutil.o
 
@@ -46,8 +49,8 @@
 GLUDIR:=x64
 endif
 GLEW32S_LIB:=$(GLEW_PREFIX)/lib/Release/$(GLUDIR)/glew32s.lib
-CFLAGS:=-std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration -Wpointer-arith -Werror=pointer-arith
-LDFLAGS:=-lm -lmingw32 -lws2_32 -lcomdlg32 -lole32 -mwindows
+CFLAGS+= -std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration -Wpointer-arith -Werror=pointer-arith
+LDFLAGS+= -lm -lmingw32 -lws2_32 -lcomdlg32 -lole32 -mwindows
 ifneq ($(MAKECMDGOALS),libblastem.dll)
 CFLAGS+= -I"$(SDL2_PREFIX)/include/$(SDL_UPPER)" -I"$(GLEW_PREFIX)/include" -DGLEW_STATIC
 LDFLAGS+= $(GLEW32S_LIB) -L"$(SDL2_PREFIX)/lib" -l$(SDL_UPPER)main -l$(SDL_UPPER) -lopengl32 -lglu32
@@ -62,7 +65,7 @@
 EXE:=
 
 HAS_PROC:=$(shell if [ -d /proc ]; then /bin/echo -e -DHAS_PROC; fi)
-CFLAGS:=-std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration -Wno-unused-value  -Wpointer-arith -Werror=pointer-arith $(HAS_PROC) -DHAVE_UNISTD_H
+CFLAGS+= -std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration -Wno-unused-value  -Wpointer-arith -Werror=pointer-arith $(HAS_PROC) -DHAVE_UNISTD_H
 
 ifeq ($(OS),Darwin)
 LIBS=$(SDL) glew
@@ -143,10 +146,10 @@
 ifndef GLES_LIB
 GLES_LIB:=$(shell pkg-config --libs glesv2)
 endif
-LDFLAGS:=-lm $(GLES_LIB)
+LDFLAGS+= -lm $(GLES_LIB)
 else #USE_GLES
 CFLAGS+= -DGLEW_STATIC -Iglew/include
-LDFLAGS:=-lm glew/lib/libGLEW.a -lEGL
+LDFLAGS+= -lm glew/lib/libGLEW.a -lEGL
 endif #USE_GLES
 
 ifeq ($(OS),Darwin)
@@ -170,10 +173,10 @@
 
 else #PORTABLE
 ifeq ($(MAKECMDGOALS),libblastem.$(SO))
-LDFLAGS:=-lm
+LDFLAGS+= -lm
 else
 CFLAGS:=$(shell pkg-config --cflags-only-I $(LIBS)) $(CFLAGS)
-LDFLAGS:=-lm $(shell pkg-config --libs $(LIBS))
+LDFLAGS:=-lm $(shell pkg-config --libs $(LIBS)) $(LDFLAGS)
 ifdef USE_FBDEV
 LDFLAGS+= -pthread
 endif
