diff --git a/src/backends/x11/standalone/x11_standalone_backend.h b/src/backends/x11/standalone/x11_standalone_backend.h
index b24ae3f2a..dd0146d4f 100644
--- a/src/backends/x11/standalone/x11_standalone_backend.h
+++ b/src/backends/x11/standalone/x11_standalone_backend.h
@@ -20,6 +20,8 @@
 
 #include <X11/Xlib-xcb.h>
 #include <fixx11h.h>
+#undef True
+#undef False
 
 namespace KWin
 {
diff --git a/src/backends/x11/standalone/x11_standalone_glx_backend.h b/src/backends/x11/standalone/x11_standalone_glx_backend.h
index e42e2486d..0db041610 100644
--- a/src/backends/x11/standalone/x11_standalone_glx_backend.h
+++ b/src/backends/x11/standalone/x11_standalone_glx_backend.h
@@ -16,6 +16,8 @@
 #include <epoxy/glx.h>
 #include <fixx11h.h>
 #include <xcb/glx.h>
+#undef True
+#undef False
 
 #include <kwingltexture.h>
 #include <kwingltexture_p.h>
