diff --git a/src/calibre/headless/headless_integration.cpp b/src/calibre/headless/headless_integration.cpp
index a5dfc869cc..0064bc6722 100644
--- a/src/calibre/headless/headless_integration.cpp
+++ b/src/calibre/headless/headless_integration.cpp
@@ -27,7 +27,7 @@ QT_BEGIN_NAMESPACE
 
 
 #ifndef __APPLE__
-class GenericUnixServices : public QGenericUnixServices {
+class GenericUnixServices : public QDesktopUnixServices {
     /* We must return desktop environment as UNKNOWN otherwise other parts of
      * Qt will try to query the nativeInterface() without checking if it exists
      * leading to a segfault.  For example, defaultHintStyleFromMatch() queries
diff --git a/src/calibre/headless/headless_integration.h b/src/calibre/headless/headless_integration.h
index f8705facff..0e39250997 100644
--- a/src/calibre/headless/headless_integration.h
+++ b/src/calibre/headless/headless_integration.h
@@ -3,7 +3,7 @@
 #include <qpa/qplatformintegration.h>
 #include <qpa/qplatformscreen.h>
 #include <qpa/qplatformservices.h>
-#include <QtGui/private/qgenericunixservices_p.h>
+#include <QtGui/private/qdesktopunixservices_p.h>
 #include <QScopedPointer>
 
 QT_BEGIN_NAMESPACE
