diff --git a/cuda_bindings/setup.py b/cuda_bindings/setup.py
index 776a510c..ee4b4195 100644
--- a/cuda_bindings/setup.py
+++ b/cuda_bindings/setup.py
@@ -237,6 +237,7 @@ include_dirs = [
     os.path.dirname(sysconfig.get_path("include")),
 ] + include_path_list
 library_dirs = [sysconfig.get_path("platlib"), os.path.join(os.sys.prefix, "lib")]
+library_dirs.append("/opt/cuda/targets/x86_64-linux/lib/")

 extra_compile_args = []
 extra_cythonize_kwargs = {}
