diff --git a/libimageviewer/libimageviewer.cmake b/libimageviewer/libimageviewer.cmake
index ad63326..8abc463 100644
--- a/libimageviewer/libimageviewer.cmake
+++ b/libimageviewer/libimageviewer.cmake
@@ -166,18 +166,6 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/translations${IMGE_VERSION_MAJOR}/
     FILES_MATCHING PATTERN "*.qm")
 target_compile_definitions(${TARGET_NAME} PRIVATE -DPLUGIN_QM_TRANS_PATH=${CMAKE_INSTALL_PREFIX}/share/${PROJECT_NAME}/translations)
 
-# Accelerated compilation of optimized parameters
-if(NOT(CMAKE_BUILD_TYPE STREQUAL "Debug"))
-    if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips64")
-        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-        SET(CMAKE_EXE_LINKER_FLAGS "-pie")
-    else()
-        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3  -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-        SET(CMAKE_EXE_LINKER_FLAGS "-pie")
-    endif()
-endif()
 
 if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "sw_64")
     SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mieee")
diff --git a/libimagevisualresult/libimagevisualresult.cmake b/libimagevisualresult/libimagevisualresult.cmake
index 5283200..71f6f9c 100644
--- a/libimagevisualresult/libimagevisualresult.cmake
+++ b/libimagevisualresult/libimagevisualresult.cmake
@@ -61,13 +61,3 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/filter_cube
     DESTINATION ${CMAKE_INSTALL_PREFIX}/share/${PROJECT_NAME}
     FILES_MATCHING PATTERN "*.dat")
 
-# 加速编译优化参数
-if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips64")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-    SET(CMAKE_EXE_LINKER_FLAGS "-pie")
-else()
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3  -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-    SET(CMAKE_EXE_LINKER_FLAGS "-pie")
-endif()
