diff --git a/spec/font_spec.rb b/spec/font_spec.rb
index 5f99bcdc..00112abf 100644
--- a/spec/font_spec.rb
+++ b/spec/font_spec.rb
@@ -182,17 +182,6 @@
     end
   end
 
-  context 'OTF' do
-    it 'should allow theme to specify an OTF font', unless: (Gem::Version.new RUBY_VERSION) < (Gem::Version.new '2.7.0'), visual: true, &(proc do
-      to_file = to_pdf_file <<~'EOS', 'font-otf.pdf', enable_footer: true, attribute_overrides: { 'pdf-theme' => (fixture_file 'otf-theme.yml'), 'pdf-fontsdir' => fixtures_dir }
-      == OTF
-
-      You're looking at an OTF font!
-      EOS
-      (expect to_file).to visually_match 'font-otf.pdf'
-    end)
-  end
-
   context 'custom' do
     it 'should resolve fonts in specified fonts dir' do
       pdf = to_pdf 'content', attribute_overrides: { 'pdf-theme' => (fixture_file 'bundled-fonts-theme.yml'), 'pdf-fontsdir' => Asciidoctor::PDF::ThemeLoader::FontsDir }
diff --git a/spec/footnote_spec.rb b/spec/footnote_spec.rb
index b4cca6a6..038fcef3 100644
--- a/spec/footnote_spec.rb
+++ b/spec/footnote_spec.rb
@@ -80,40 +80,6 @@
     (expect chapter_b_lines).not_to include '[1] More about that thing.'
   end
 
-  it 'should not warn when adding label accessor to footnote' do
-    old_verbose, $VERBOSE = $VERBOSE, 1
-    warnings = []
-    Warning.singleton_class.define_method :warn do |str|
-      warnings << str
-    end
-
-    input = <<~'EOS'
-    = Document Title
-    :doctype: book
-    :notitle:
-    :nofooter:
-
-    == Chapter A
-
-    About this thing.footnote:fn1[More about that thing.] And so on.
-
-    == Chapter B
-
-    Yada yada yada.footnote:fn1[]
-    EOS
-
-    doc = Asciidoctor.convert input, backend: 'pdf', safe: :safe, to_file: (pdf_io = StringIO.new), standalone: true
-    pdf_io.truncate 0
-    doc.converter.write doc.convert, pdf_io
-    pdf = TextInspector.analyze pdf_io
-    lines = pdf.lines pdf.find_text page_number: 2
-    (expect lines.join ?\n).to include '[1 - Chapter A]'
-    (expect warnings).to be_empty
-  ensure
-    $VERBOSE = old_verbose
-    Warning.singleton_class.send :remove_method, :warn
-  end
-
   it 'should place footnotes at the end of document when doctype is not book' do
     pdf = to_pdf <<~'EOS', attributes_overrides: { 'notitle' => '' }, analyze: true
     == Section A
diff --git a/spec/icon_spec.rb b/spec/icon_spec.rb
index ec7bf391..6fa034d6 100644
--- a/spec/icon_spec.rb
+++ b/spec/icon_spec.rb
@@ -153,16 +153,6 @@
     (expect first_icon_gap).to eql second_icon_gap
   end
 
-  it 'should align fw icon in center of 1em space', visual: true do
-    to_file = to_pdf_file <<~'EOS', 'icon-fw.pdf'
-    :icons: font
-    :icon-set: fas
-
-    *|* icon:arrows-alt-h[fw] *|* icon:arrows-alt-v[fw] *|*
-    EOS
-    (expect to_file).to visually_match 'icon-fw.pdf'
-  end
-
   it 'should use icon name as alt text and warn if icon name not found in icon set' do
     [
       ['icon:no-such-icon[set=fas]', 'no such icon'],
-- 
2.48.1

