diff --git a/Makefile b/Makefile
index 7675374..1a2a284 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ all: install build test lint
 # Install dependencies
 install:
 	# Frontend
-	yarn install --pure-lockfile
+	yarn install --immutable
 	# Backend
 	go install -v ./pkg/
 	go install golang.org/x/lint/golint@latest
@@ -21,7 +21,7 @@ build-debug:
 	mage -v build:debug
 
 run-frontend:
-	yarn install --pure-lockfile
+	yarn install --immutable
 	yarn dev
 
 run-backend:
