From 95e0e95d449ba19c4d9a4130b5c1aee9774a3e92 Mon Sep 17 00:00:00 2001
From: hashworks <mail@hashworks.net>
Date: Wed, 5 Feb 2025 15:24:09 +0100
Subject: [PATCH] Fix configuration to point to installed files

---
 settings/env.defaults | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/settings/env.defaults b/settings/env.defaults
index b8c66667..87f139ed 100644
--- a/settings/env.defaults
+++ b/settings/env.defaults
@@ -10,7 +10,7 @@ NOMINATIM_DATABASE_DSN="pgsql:dbname=nominatim"
 
 # Database web user.
 # Nominatim sets up read-only access for this user during installation.
-NOMINATIM_DATABASE_WEBUSER="www-data"
+NOMINATIM_DATABASE_WEBUSER="http"
 
 # Tokenizer used for normalizing and parsing queries and names.
 # The tokenizer is set up during import and cannot be changed afterwards
@@ -57,7 +57,7 @@ NOMINATIM_HTTP_PROXY_PASSWORD=
 
 # Location of the osm2pgsql binary.
 # EXPERT ONLY. You should usually use the supplied osm2pgsql.
-NOMINATIM_OSM2PGSQL_BINARY=
+NOMINATIM_OSM2PGSQL_BINARY=/usr/bin/osm2pgsql
 
 # Directory where to find pre-computed Wikipedia importance files.
 # When unset, the data is expected to be located in the project directory.
-- 
2.48.1

