From 601be5d6d7a951cef6f3b50188bfd1d85942cbe3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 2 Apr 2025 18:53:59 +0000
Subject: [PATCH] Open containing folder for the advanced export dialog too

---
 apostrophe/export_dialog.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/apostrophe/export_dialog.py b/apostrophe/export_dialog.py
index d0c7979..b942bf2 100644
--- a/apostrophe/export_dialog.py
+++ b/apostrophe/export_dialog.py
@@ -345,6 +345,10 @@ class AdvancedExportDialog(Adw.Dialog):
                 self.file.name + '.' +
                 self.formats_list.get_selected_row().item.ext)
 
+        # / is the base path when the current file has not been saved
+        if self.file.base_path != "/":
+            export_dialog.set_current_folder(Gio.File.new_for_path(self.file.base_path))
+
         export_dialog.set_modal(True)
         export_dialog.set_transient_for(self.get_root())
         export_dialog.connect("response", on_response)
-- 
GitLab

