commit 3a29b212f4885182ebb269e4bd3f12c76fc5ebb1
Author: CARBONNEAUX Mathieu <mcarbonneaux@users.noreply.github.com>
Date:   Mon Feb 10 17:54:33 2020 +0100

    Update fcgiapp.c

diff --git a/libfcgi/fcgiapp.c b/libfcgi/fcgiapp.c
index 98fa02f..0f6657c 100644
--- a/libfcgi/fcgiapp.c
+++ b/libfcgi/fcgiapp.c
@@ -468,8 +468,8 @@ int FCGX_VFPrintF(FCGX_Stream *stream, const char *format, va_list arg)
                             fmtBuff[3] = (char) op;
                             fmtBuff[4] = '\0';
                             specifierLength = 4;
-                            break;
                         }
+			break;
 	            case 'L':
                     case 'h':
                         sizeModifier = op;
@@ -570,8 +570,8 @@ int FCGX_VFPrintF(FCGX_Stream *stream, const char *format, va_list arg)
                         if(*p == 'l' && *(p + 1) == 'l') {
                             sizeModifier = 'L';
                             CopyAndAdvance(&fmtBuffPtr, &p, 2);
-                            break;
                         }
+                        break;
                     case 'L':
                     case 'h':
                         sizeModifier = *p;
