diff -ru gerbera-2.6.1.orig/src/web/config_load.cc gerbera-2.6.1/src/web/config_load.cc
--- gerbera-2.6.1.orig/src/web/config_load.cc   2025-10-04 14:25:39.944748329 +0200
+++ gerbera-2.6.1/src/web/config_load.cc        2025-10-04 14:26:02.536710317 +0200
@@ -150,7 +150,7 @@
 template <typename T>
 void Web::ConfigLoad::setValue(Json::Value& item, const T& value)
 {
-    static_assert(fmt::has_formatter<T, fmt::format_context>::value, "T must be formattable");
+    static_assert(fmt::is_formattable<T, fmt::format_context>::value, "T must be formattable");
     item[CONFIG_LOAD_VALUE] = fmt::to_string(value);
 }
 
