From b4f63e4ad9bbb19e952d121c6f3cda2e66c46d19 Mon Sep 17 00:00:00 2001
From: Christian Clason <c.clason@uni-graz.at>
Date: Sat, 12 Jul 2025 17:12:50 +0200
Subject: [PATCH] build(deps): bump tree-sitter-vim to v0.7.0

---
 runtime/queries/vim/highlights.scm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/runtime/queries/vim/highlights.scm b/runtime/queries/vim/highlights.scm
index 20a044a60d..c35986fa31 100644
--- a/runtime/queries/vim/highlights.scm
+++ b/runtime/queries/vim/highlights.scm
@@ -253,6 +253,9 @@
 (heredoc
   (parameter) @keyword)
 
+(script
+  (parameter) @keyword)
+
 [
   (marker_definition)
   (endmarker)
-- 
2.50.1

