diff --git a/test_pycosat.py b/test_pycosat.py
index f1165d8..0a6969c 100644
--- a/test_pycosat.py
+++ b/test_pycosat.py
@@ -258,7 +258,7 @@ def run(verbosity=1, repeat=1):
     suite = unittest.TestSuite()
     for cls in tests:
         for _ in range(repeat):
-            suite.addTest(unittest.makeSuite(cls))
+            suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(cls))
 
     runner = unittest.TextTestRunner(verbosity=verbosity)
     return runner.run(suite)
