diff -aur pysmi-0.3.4.old/docs/source/conf.py pysmi-0.3.4/docs/source/conf.py
--- pysmi-0.3.4.old/docs/source/conf.py	2024-12-06 11:18:38.694903726 +0100
+++ pysmi-0.3.4/docs/source/conf.py	2024-12-06 11:19:15.047714392 +0100
@@ -309,7 +309,7 @@
 
 
 # Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {'https://docs.python.org/': None}
+intersphinx_mapping = {'python': ('https://docs.python.org/', None)}
 
 # this merges constructor docstring with class docstring
 autoclass_content = 'both'
