diff --git a/app/SysTray-X/SysTray-X-app/systrayxlink.cpp b/app/SysTray-X/SysTray-X-app/systrayxlink.cpp
index 9c3adbb..9498496 100644
--- a/app/SysTray-X/SysTray-X-app/systrayxlink.cpp
+++ b/app/SysTray-X/SysTray-X-app/systrayxlink.cpp
@@ -1043,6 +1043,9 @@ void    SysTrayXLink::EncodePreferences( const Preferences& pref )
     prefObject.insert( "startupDelay", QJsonValue::fromVariant( QString::number( pref.getStartupDelay() ) ) );
     prefObject.insert( "apiCountMethod", QJsonValue::fromVariant( QString( pref.getApiCountMethod() ? "true" : "false" ) ) );
 
+    prefObject.insert( "newIndicatorType", QJsonValue::fromVariant( QString::number( pref.getNewIndicatorType() ) ) );
+    prefObject.insert( "newShadeColor", QJsonValue::fromVariant( QString( pref.getNewShadeColor() ) ) );
+
     prefObject.insert( "startApp", QJsonValue::fromVariant( pref.getStartApp() ) );
     prefObject.insert( "startAppArgs", QJsonValue::fromVariant( pref.getStartAppArgs() ) );
     prefObject.insert( "closeApp", QJsonValue::fromVariant( pref.getCloseApp() ) );
diff --git a/webext/js/windowEvent.js b/webext/js/windowEvent.js
index e2393a2..be93150 100644
--- a/webext/js/windowEvent.js
+++ b/webext/js/windowEvent.js
@@ -5,15 +5,32 @@
 // Using a closure to not leak anything but the API to the outside world.
 (function (exports) {
 
+  var ExtensionCommon;
+  var ExtensionSupport;
+  if (typeof ChromeUtils.import === "function") {
+    ExtensionCommon = ChromeUtils.import("resource://gre/modules/ExtensionCommon.jsm").ExtensionCommon;
+    ExtensionSupport = ChromeUtils.import("resource:///modules/ExtensionSupport.jsm").ExtensionSupport;
+  }
+  else
+  {
+    // TB136+
+    ExtensionCommon = ChromeUtils.importESModule("resource://gre/modules/ExtensionCommon.sys.mjs").ExtensionCommon;
+    ExtensionSupport = ChromeUtils.importESModule("resource:///modules/ExtensionSupport.sys.mjs").ExtensionSupport;
+  }
+
+/*
+  The old ways...
+
   // Get various parts of the WebExtension framework that we need.
   var { ExtensionCommon } = ChromeUtils.import("resource://gre/modules/ExtensionCommon.jsm");
 
   // You probably already know what this does.
 //  var { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
+//  var Services = globalThis.Services || ChromeUtils.import("resource://gre/modules/Services.jsm").Services;
 
   // A helpful class for listening to windows opening and closing.
   var { ExtensionSupport } = ChromeUtils.import("resource:///modules/ExtensionSupport.jsm");
-
+*/
 
   /**
    * This object is just what we're using to listen for toolbar clicks. The implementation
diff --git a/webext/manifest.json b/webext/manifest.json
index 9cb2a86..5b51fcb 100644
--- a/webext/manifest.json
+++ b/webext/manifest.json
@@ -10,7 +10,7 @@
     "gecko": {
       "id": "systray-x@Ximi1970",
       "strict_min_version": "91.0",
-      "strict_max_version": "130.*"
+      "strict_max_version": "136.*"
     }
   },
 
