diff --git a/src/greeter/greeterproxy.cpp b/src/greeter/greeterproxy.cpp
index f745c85..ec812cb 100644
--- a/src/greeter/greeterproxy.cpp
+++ b/src/greeter/greeterproxy.cpp
@@ -375,14 +375,6 @@ void GreeterProxy::readyRead()
             // Q_EMIT signal
             Q_EMIT hostNameChanged(d->hostName);
         } break;
-        case DaemonMessages::LoginSucceeded: {
-            QString user;
-            input >> user;
-
-            qCDebug(treelandGreeter) << "Message received from daemon: LoginSucceeded:" << user;
-
-            Q_EMIT loginSucceeded(user);
-        } break;
         case DaemonMessages::LoginFailed: {
             QString user;
             input >> user;
