diff --git a/waylib/src/server/qtquick/wqmlcreator.cpp b/waylib/src/server/qtquick/wqmlcreator.cpp
index a09f0929c..7adba497b 100644
--- a/waylib/src/server/qtquick/wqmlcreator.cpp
+++ b/waylib/src/server/qtquick/wqmlcreator.cpp
@@ -238,7 +238,7 @@ void WQmlCreatorComponent::create(QSharedPointer<WQmlCreatorDelegateData> data,
 
     auto context = new QQmlContext(qmlContext(this), this);
     context->setContextProperties(m_contextProperties);
-    data->object = d->createWithProperties(parent, tmp, context);
+    data->object = d->createWithProperties(parent, tmp, context, QQmlComponentPrivate::CreateBehavior::Qml);
     context->setParent(data->object);
 
     if (data->object) {
