diff --git a/pyproject.toml b/pyproject.toml
index 34b1d7c..d48a246 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["hatchling >=1.5", "virtualenv<21.0"]
+requires = ["hatchling >=1.5"]
 build-backend = "hatchling.build"

 [project]
@@ -84,10 +84,8 @@ exclude = [
 artifacts = ["metakernel/py.typed"]

 [tool.pytest.ini_options]
-minversion = "9.0"
 addopts= "-raXs  --durations 10 --color=yes --doctest-modules"
 testpaths = ["tests", "tests/magics"]
-strict = true
 log_level = "INFO"
 timeout = 300
 # Restore this setting to debug failures
