--- a/platformio/dependencies.py
+++ b/platformio/dependencies.py
@@ -28,23 +28,23 @@ def get_core_dependencies():
 def get_pip_dependencies():
     core = [
         "pip",
-        "bottle == 0.13.*",
-        "click >=8.0.4, <8.6, !=8.3",  # click 9.0 removes 'protected_args' attribute
+        "bottle",
+        "click",
         "colorama",
-        "marshmallow >=3, <5",
-        "pyelftools >=0.27, <1",
-        "pyserial == 3.5.*",  # keep in sync "device/monitor/terminal.py"
-        "requests%s == 2.*" % ("[socks]" if is_proxy_set(socks=True) else ""),
-        "semantic_version == 2.10.*",
-        "tabulate == 0.*",
+        "marshmallow",
+        "pyelftools",
+        "pyserial",  # keep in sync "device/monitor/terminal.py"
+        "requests%s" % ("[socks]" if is_proxy_set(socks=True) else ""),
+        "semantic_version",
+        "tabulate",
     ]
 
     home = [
         # PIO Home requirements
-        "ajsonrpc == 1.2.*",
-        "starlette < 2",
-        "uvicorn < 1",
-        "wsproto == 1.*",
+        "ajsonrpc",
+        "starlette",
+        "uvicorn",
+        "wsproto",
     ]
 
     extra = []
