From 7e48bab2616ee9688dd1159d1d143f7ca641f917 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 11 Nov 2024 22:15:23 +0100
Subject: [PATCH] application: Translate app name on about dialog

---
 src/application.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/application.rs b/src/application.rs
index 2fb564b..c7edbea 100644
--- a/src/application.rs
+++ b/src/application.rs
@@ -160,7 +160,7 @@ impl Application {
     fn show_about_dialog(&self) {
         let window = self.active_window().unwrap().downcast::<Window>().unwrap();
         adw::AboutWindow::builder()
-            .application_name("App Icon Preview")
+            .application_name(gettext("App Icon Preview"))
             .application_icon(config::APP_ID)
             .license_type(gtk::License::Gpl30)
             .website("https://gitlab.gnome.org/World/design/app-icon-preview/")
-- 
GitLab

