From 1a2617cb2e1b63fd484141ea6f11dccb73c53d45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 11 Nov 2024 22:52:01 +0100
Subject: [PATCH] Fix crash on second activate of the application

---
 src/application.rs | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/application.rs b/src/application.rs
index 2fb564b..4c684c2 100644
--- a/src/application.rs
+++ b/src/application.rs
@@ -87,11 +87,12 @@ mod imp {
             let window = application.create_window();
             window.present();
             // Setup action channel
-            let receiver = self.receiver.borrow_mut().take().unwrap();
-            receiver.attach(
-                None,
-                clone!(@strong application => move |action| application.do_action(action)),
-            );
+            if let Some(receiver) = self.receiver.borrow_mut().take() {
+                receiver.attach(
+                    None,
+                    clone!(@strong application => move |action| application.do_action(action)),
+                );
+            }
         }
 
         fn open(&self, files: &[gio::File], _hint: &str) {
-- 
2.47.0

