diff -ru CyLP-0.92.2.orig/cylp/cpp/IClpSimplex.cpp CyLP-0.92.2/cylp/cpp/IClpSimplex.cpp
--- CyLP-0.92.2.orig/cylp/cpp/IClpSimplex.cpp	2024-06-22 23:03:22.358210636 +0200
+++ CyLP-0.92.2/cylp/cpp/IClpSimplex.cpp	2024-06-22 23:30:48.335977434 +0200
@@ -20,7 +20,7 @@
 int IClpSimplex::argWeightedMax(PyObject* arr, PyObject* arr_ind, PyObject* w, PyObject* w_ind){
     //_import_array();
 
-    npy_intp w_ind_len = PyArray_DIM(w_ind, 0);
+    npy_intp w_ind_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(w_ind), 0);
     if (w_ind_len == 0)
         return -1; //return PyArray_ArgMax(reinterpret_cast<PyArrayObject*>(arr));
 
@@ -57,7 +57,7 @@
 
     PyObject* arr_it = PyArray_IterNew(arr);
 
-    npy_intp arr_len = PyArray_DIM(arr, 0);
+    npy_intp arr_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(arr), 0);
 
     if (arr_len == 0)
         return 0;
@@ -107,8 +107,8 @@
             }
         }
         else{ //look in whole array, weights array
-            npy_intp w_len = PyArray_DIM(w, 0);
-            npy_intp w_ind_len = PyArray_DIM(w_ind, 0);
+            npy_intp w_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(w), 0);
+            npy_intp w_ind_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(w_ind), 0);
             if (w_ind_len != w_len){
                 PyErr_SetString(PyExc_ValueError,
                         "If w is a numpy array, w_ind should be a numpy array of the same size.");
@@ -148,8 +148,8 @@
     }
     else{  //only indices specified in arr_ind
 
-        npy_intp arr_ind_len = PyArray_DIM(arr_ind, 0);
-        npy_intp arr_len = PyArray_DIM(arr, 0);
+        npy_intp arr_ind_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(arr_ind), 0);
+        npy_intp arr_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(arr), 0);
 
         if (arr_ind_len != arr_len){
             PyErr_SetString(PyExc_ValueError,
@@ -206,8 +206,8 @@
         }
         else{  //just elements specified in arr_ind, weight's an array
 
-            npy_intp arr_ind_len = PyArray_DIM(arr_ind, 0);
-            npy_intp arr_len = PyArray_DIM(arr, 0);
+            npy_intp arr_ind_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(arr_ind), 0);
+            npy_intp arr_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(arr), 0);
 
             if (arr_ind_len != arr_len){
                 PyErr_SetString(PyExc_ValueError,
@@ -216,8 +216,8 @@
             }
 
 
-            npy_intp w_len = PyArray_DIM(w, 0);
-            npy_intp w_ind_len = PyArray_DIM(w_ind, 0);
+            npy_intp w_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(w), 0);
+            npy_intp w_ind_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(w_ind), 0);
             if (w_ind_len != w_len){
                 PyErr_SetString(PyExc_ValueError,
                         "If w is a numpy array, w_ind should be a numpy array of the same size.");
@@ -291,7 +291,7 @@
     PyObject* arr_it = PyArray_IterNew(arr);
     PyObject* whr_it = PyArray_IterNew(whr);
 
-    npy_intp arr_len = PyArray_DIM(arr, 0);
+    npy_intp arr_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(arr), 0);
 
     if (arr_len == 0)
         return 0;
@@ -1397,7 +1397,7 @@
         return NULL;
     }
 
-    npy_intp inds_len = PyArray_DIM(inds, 0);
+    npy_intp inds_len = PyArray_DIM(reinterpret_cast<PyArrayObject*>(inds), 0);
     if (inds_len == 0){
         npy_intp dims = 0;
         PyObject *Arr = PyArray_SimpleNewFromData( 1, &dims, NPY_INT, tempIntArray );
diff -ru CyLP-0.92.2.orig/cylp/cpp/ICoinIndexedVector.cpp CyLP-0.92.2/cylp/cpp/ICoinIndexedVector.cpp
--- CyLP-0.92.2.orig/cylp/cpp/ICoinIndexedVector.cpp	2024-06-22 23:03:22.358210636 +0200
+++ CyLP-0.92.2/cylp/cpp/ICoinIndexedVector.cpp	2024-06-22 23:12:49.922277803 +0200
@@ -126,7 +126,7 @@
                         PyFloat_AsDouble(PyList_GetItem(other, (Py_ssize_t)i)));
             }
         }else if (other_is_array){
-            npy_intp other_length = PyArray_DIM(other, 0);
+            npy_intp other_length = PyArray_DIM(reinterpret_cast<PyArrayObject*>(other), 0);
             PyObject* iterator = PyArray_IterNew(other);
             if (ind_length != other_length){
                 PyErr_SetString(PyExc_ValueError, 
@@ -149,7 +149,7 @@
 
         // If the index set is a numpy array 3 possibilities for the rhs: number, list, array 
         else if (ind_is_array){
-            npy_intp ind_length = PyArray_DIM(ind, 0);
+            npy_intp ind_length = PyArray_DIM(reinterpret_cast<PyArrayObject*>(ind), 0);
             PyObject* ind_iterator = PyArray_IterNew(ind);
 
             if (other_is_num){
@@ -174,7 +174,7 @@
                         PyArray_ITER_NEXT(ind_iterator);
                     }
                 }else if (other_is_array){
-                    npy_intp other_length = PyArray_DIM(other, 0);
+                    npy_intp other_length = PyArray_DIM(reinterpret_cast<PyArrayObject*>(other), 0);
                     PyObject* iterator = PyArray_IterNew(other);
                     if (ind_length != other_length){
                         PyErr_SetString(PyExc_ValueError, 
