diff --git a/lib/cli/build.rs b/lib/cli/build.rs
index 28df3c148f1..2339cb1bd91 100644
--- a/lib/cli/build.rs
+++ b/lib/cli/build.rs
@@ -1,7 +1,7 @@
 use chrono::prelude::*;
 
 pub fn main() {
-    let utc: DateTime<Utc> = Utc::now();
+    let utc: DateTime<Utc> = DateTime::from_timestamp(0, 0).unwrap();
     let date = utc.format("%Y-%m-%d").to_string();
     println!("cargo:rustc-env=WASMER_BUILD_DATE={date}");
 }
