--- a/src/SDCCmain.c
+++ b/src/SDCCmain.c
@@ -2915,7 +2915,7 @@ main (int argc, char **argv, char **envp)
 #endif
           preamble = fdopen (p[1], "w");
           wassert (preamble);
-          fprintf (preamble, port->c_preamble);
+          fprintf (preamble, "%s", port->c_preamble);
           fclose (preamble);
           yyin = fdopen (p[0], "r");
           wassert (yyin);
--- a/support/cpp/gcc/diagnostic.cc
+++ b/support/cpp/gcc/diagnostic.cc
@@ -2254,7 +2254,7 @@ fancy_abort (const char *file, int line, const char *function)
   if (global_dc->printer == NULL)
     {
       /* Print the error message.  */
-      fnotice (stderr, diagnostic_kind_text[DK_ICE]);
+      fnotice (stderr, "%s", diagnostic_kind_text[DK_ICE]);
       fnotice (stderr, "in %s, at %s:%d", function, trim_filename (file), line);
       fputc ('\n', stderr);
 
