diff --git a/jieba/_compat.py b/jieba/_compat.py
index 4ea3f7a..8e23319 100644
--- a/jieba/_compat.py
+++ b/jieba/_compat.py
@@ -15,11 +15,10 @@ def setLogLevel(log_level):
 check_paddle_install = {'is_paddle_installed': False}
 
 try:
-    import pkg_resources
+    from importlib.resources import files
 
-    get_module_res = lambda *res: pkg_resources.resource_stream(__name__,
-                                                                os.path.join(*res))
-except ImportError:
+    get_module_res = lambda *res: files(__name__).joinpath(*res).open('rb')
+except (ImportError, AttributeError):
     get_module_res = lambda *res: open(os.path.normpath(os.path.join(
         os.getcwd(), os.path.dirname(__file__), *res)), 'rb')
 
