diff --unified --recursive --text --new-file ldapdomaindump-0.10.0.orig/setup.py ldapdomaindump-0.10.0/setup.py
--- ldapdomaindump-0.10.0.orig/setup.py	2025-06-19 19:08:32.244697492 +0200
+++ ldapdomaindump-0.10.0/setup.py	2025-06-19 19:08:40.707224933 +0200
@@ -10,6 +10,5 @@
       install_requires=['dnspython', 'ldap3>=2.5,!=2.5.2,!=2.5.0,!=2.6'],
       package_data={'ldapdomaindump': ['style.css']},
       include_package_data=True,
-      scripts=['bin/ldapdomaindump', 'bin/ldd2bloodhound', 'bin/ldd2pretty'],
       license="MIT",
       )
