diff -aur remoto-1.2.1.old/remoto/tests/test_log.py remoto-1.2.1/remoto/tests/test_log.py
--- remoto-1.2.1.old/remoto/tests/test_log.py	2024-02-10 14:58:30.019922611 +0100
+++ remoto-1.2.1/remoto/tests/test_log.py	2024-02-10 14:59:20.017630110 +0100
@@ -1,7 +1,7 @@
 from pytest import raises
 from remoto import log
 from remoto.exc import TimeoutError
-from mock import Mock
+from unittest.mock import Mock
 
 
 class TestReporting(object):
diff -aur remoto-1.2.1.old/remoto/tests/test_process.py remoto-1.2.1/remoto/tests/test_process.py
--- remoto-1.2.1.old/remoto/tests/test_process.py	2024-02-10 14:58:30.019922611 +0100
+++ remoto-1.2.1/remoto/tests/test_process.py	2024-02-10 14:59:17.057568476 +0100
@@ -1,4 +1,4 @@
-from mock import Mock
+from unittest.mock import Mock
 from remoto import process
 
 
diff -aur remoto-1.2.1.old/remoto/tests/test_rsync.py remoto-1.2.1/remoto/tests/test_rsync.py
--- remoto-1.2.1.old/remoto/tests/test_rsync.py	2024-02-10 14:58:30.019922611 +0100
+++ remoto-1.2.1/remoto/tests/test_rsync.py	2024-02-10 14:59:23.144361882 +0100
@@ -1,4 +1,4 @@
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 from remoto import file_sync
 
 
