diff --unified --recursive --text --new-file fabric-3.2.2.orig/tests/auth.py fabric-3.2.2/tests/auth.py
--- fabric-3.2.2.orig/tests/auth.py	2026-04-06 11:07:50.183099028 +0200
+++ fabric-3.2.2/tests/auth.py	2026-04-06 11:17:51.828978282 +0200
@@ -2,7 +2,7 @@
 from pathlib import Path
 from unittest.mock import Mock, patch
 
-from invoke.vendor.lexicon import Lexicon
+from lexicon import Lexicon
 from pytest import raises, fixture
 from paramiko import (
     AgentKey,
diff --unified --recursive --text --new-file fabric-3.2.2.orig/tests/config.py fabric-3.2.2/tests/config.py
--- fabric-3.2.2.orig/tests/config.py	2026-04-06 11:07:50.183145298 +0200
+++ fabric-3.2.2/tests/config.py	2026-04-06 11:17:51.634977823 +0200
@@ -3,7 +3,7 @@
 
 from paramiko.config import SSHConfig
 from invoke import Local
-from invoke.vendor.lexicon import Lexicon
+from lexicon import Lexicon
 
 from fabric import Config, Remote, RemoteShell
 from fabric.util import get_local_user
diff --unified --recursive --text --new-file fabric-3.2.2.orig/tests/connection.py fabric-3.2.2/tests/connection.py
--- fabric-3.2.2.orig/tests/connection.py	2026-04-06 11:07:50.183230301 +0200
+++ fabric-3.2.2/tests/connection.py	2026-04-06 11:17:51.727978043 +0200
@@ -13,7 +13,7 @@
 import pytest  # for mark, internal raises
 from pytest import skip, param
 from pytest_relaxed import raises
-from invoke.vendor.lexicon import Lexicon
+from lexicon import Lexicon
 
 from invoke.config import Config as InvokeConfig
 from invoke.exceptions import ThreadException
diff --unified --recursive --text --new-file fabric-3.2.2.orig/tests/_util.py fabric-3.2.2/tests/_util.py
--- fabric-3.2.2.orig/tests/_util.py	2026-04-06 11:07:50.183061061 +0200
+++ fabric-3.2.2/tests/_util.py	2026-04-06 11:17:51.779978166 +0200
@@ -3,7 +3,7 @@
 import re
 import sys
 
-from invoke.vendor.lexicon import Lexicon
+from lexicon import Lexicon
 from pytest_relaxed import trap
 
 from fabric.main import make_program
