--- a/Makefile
+++ b/Makefile
@@ -36,9 +36,10 @@ UPB_DIR := vendor/upb
 UPB_INCLUDES := -I./$(UPB_DIR) -I./$(UPB_DIR)/third_party/utf8_range
 UPB_SRC_FILES := $(UPB_DIR)/upb.c $(UPB_DIR)/third_party/utf8_range/utf8_range.c protobuf/pg_query.upb_minitable.c
 
-SRC_FILES := $(wildcard src/*.c src/postgres/*.c) vendor/xxhash/xxhash.c $(UPB_SRC_FILES)
+SRC_FILES := $(wildcard src/*.c src/postgres/*.c) $(UPB_SRC_FILES)
 OBJ_FILES := $(SRC_FILES:.c=.o)
 
+LIBS = -lxxhash
 override CFLAGS += -g -I. -I./vendor $(UPB_INCLUDES) -I./src/include -I./src/postgres/include -Wall -Wno-unused-function -Wno-unused-value -Wno-unused-variable -fno-strict-aliasing -fwrapv -fPIC
 
 ifeq ($(OS),Windows_NT)
@@ -49,7 +50,7 @@ endif
 override PG_CONFIGURE_FLAGS += -q --without-readline --without-zlib --without-icu
 
 override TEST_CFLAGS += -g -I. -I./vendor $(UPB_INCLUDES) -Wall
-override TEST_LDFLAGS += -pthread
+override TEST_LDFLAGS += -pthread $(LIBS)
 
 CFLAGS_OPT_LEVEL = -O3
 ifeq ($(DEBUG),1)
--- a/src/pg_query_fingerprint.c
+++ b/src/pg_query_fingerprint.c
@@ -5,7 +5,7 @@
 #include "pg_query_fingerprint.h"
 
 #include "postgres.h"
-#include "xxhash/xxhash.h"
+#include <xxhash.h>
 #include "lib/ilist.h"
 
 #include "parser/parser.h"
