diff --git a/Gemfile b/Gemfile
index 081f0c1dc82..d5cd789b82f 100644
--- a/Gemfile
+++ b/Gemfile
@@ -143,7 +143,7 @@ gem 'browser', '~> 5.3.1', feature_category: :shared
 gem 'ohai', '~> 18.1', feature_category: :product_analytics
 
 # GPG
-gem 'gpgme', '~> 2.0.24', feature_category: :source_code_management
+gem 'gpgme', '~> 2.0.24', git: '../ruby-gpgme', branch: 'makepkg', feature_category: :source_code_management
 
 # LDAP Auth
 # GitLab fork with several improvements to original library. For full list of changes
