diff --git a/Gemfile b/Gemfile
index c0eadbe75fa..d2bfd0e5966 100644
--- a/Gemfile
+++ b/Gemfile
@@ -760,3 +760,7 @@ gem 'paper_trail', '~> 16.0', feature_category: :shared
 gem "i18n_data", "~> 0.13.1", feature_category: :system_access
 
 gem "gitlab-cloud-connector", "~> 1.11", require: 'gitlab/cloud_connector', feature_category: :cloud_connector
+
+gem "erb", "~> 4.0"
+gem "irb", "~> 1.15"
+gem "rdoc", "~> 6.13"
diff --git a/Gemfile.checksum b/Gemfile.checksum
index f39ebe39db8..c2f508f0483 100644
--- a/Gemfile.checksum
+++ b/Gemfile.checksum
@@ -70,6 +70,8 @@
 {"name":"capybara-screenshot","version":"1.0.26","platform":"ruby","checksum":"816b9370a07752097c82a05f568aaf5d3b7f45c3db5d3aab2014071e1b3c0c77"},
 {"name":"carrierwave","version":"1.3.4","platform":"ruby","checksum":"81772dabd1830edbd7f4526d2ae2c79f974f1d48900c3f03f7ecb7c657463a21"},
 {"name":"cbor","version":"0.5.9.8","platform":"ruby","checksum":"9ee097fc58d9bc5e406d112cd2d4e112c7354ec16f8b6ff34e4732c1e44b4eb7"},
+{"name":"cgi","version":"0.4.2","platform":"java","checksum":"009a20f179f86e5df89771fc079a08dda1d7f891c3b5909f5801834de8965859"},
+{"name":"cgi","version":"0.4.2","platform":"ruby","checksum":"a3cb190d46a820ca01a3e28bd5b64e67003ff99d7884b70512448566f35347e6"},
 {"name":"character_set","version":"1.8.0","platform":"java","checksum":"2d94ac33d6569434cf1ba464012b5e98010f5dafbd7b750e8d7db79f4c8eb8f7"},
 {"name":"character_set","version":"1.8.0","platform":"ruby","checksum":"2b7317462adaedff0bd1576ae86d71bc5efe133a5d0b7c257021b00fe3153f51"},
 {"name":"charlock_holmes","version":"0.7.9","platform":"ruby","checksum":"b49e8a11ce1921e2c5b65511bb864ae51720ce9bd1c336ccf0e89e6c8ae62db0"},
@@ -148,6 +150,8 @@
 {"name":"email_spec","version":"2.3.0","platform":"ruby","checksum":"df23be7a131186f7a3d5be3b35eaac9196f9ac13bd26c9c3d59341e13d852d11"},
 {"name":"email_validator","version":"2.2.4","platform":"ruby","checksum":"5ab238095bec7aef9389f230e9e0c64c5081cdf91f19d6c5cecee0a93af20604"},
 {"name":"encryptor","version":"3.0.0","platform":"ruby","checksum":"abf23f94ab4d864b8cea85b43f3432044a60001982cda7c33c1cd90da8db1969"},
+{"name":"erb","version":"4.0.4","platform":"java","checksum":"3767088f230194a84316a5879f0c3dc5f2a96fb1c9a198c3e27d3f35d416c597"},
+{"name":"erb","version":"4.0.4","platform":"ruby","checksum":"de116e106205c46bc01918789b611aaad1328dcc6e9f12cf8cd2cc60ef619717"},
 {"name":"erubi","version":"1.12.0","platform":"ruby","checksum":"27bedb74dfb1e04ff60674975e182d8ca787f2224f2e8143268c7696f42e4723"},
 {"name":"escape_utils","version":"1.3.0","platform":"ruby","checksum":"dffb7010922880ace6ceed642156c64e2a64620f27e0849f43bc4f68fd3c2c09"},
 {"name":"et-orbi","version":"1.2.11","platform":"ruby","checksum":"d26e868cc21db88280a9ec1a50aa3da5d267eb9b2037ba7b831d6c2731f5df64"},
diff --git a/Gemfile.lock b/Gemfile.lock
index 114ed753996..9bc86c84ad2 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -436,6 +436,7 @@ GEM
       mime-types (>= 1.16)
       ssrf_filter (~> 1.0, < 1.1.0)
     cbor (0.5.9.8)
+    cgi (0.4.2)
     character_set (1.8.0)
     charlock_holmes (0.7.9)
     chef-config (18.3.0)
@@ -609,6 +610,8 @@ GEM
     email_validator (2.2.4)
       activemodel
     encryptor (3.0.0)
+    erb (4.0.4)
+      cgi (>= 0.3.3)
     erubi (1.12.0)
     escape_utils (1.3.0)
     et-orbi (1.2.11)
@@ -2136,6 +2139,7 @@ DEPENDENCIES
   elasticsearch-rails (~> 7.2)
   email_reply_trimmer (~> 0.1)
   email_spec (~> 2.3.0)
+  erb
   error_tracking_open_api!
   factory_bot_rails (~> 6.4.3)
   faraday (~> 2)
@@ -2235,6 +2239,7 @@ DEPENDENCIES
   invisible_captcha (~> 2.1.0)
   ipaddress (~> 0.8.3)
   ipynbdiff!
+  irb
   jira-ruby (~> 2.3.0)
   js_regex (~> 3.8)
   json (~> 2.11.0)
@@ -2345,6 +2350,7 @@ DEPENDENCIES
   rails-i18n (~> 7.0, >= 7.0.9)
   rainbow (~> 3.0)
   rbtrace (~> 0.4)
+  rdoc
   re2 (~> 2.15)
   recaptcha (~> 5.12)
   redis (~> 5.4.0)
