diff -ru ruby-gpgme-2.0.24.orig/ext/gpgme/gpgme_n.c ruby-gpgme-2.0.24/ext/gpgme/gpgme_n.c
--- ruby-gpgme-2.0.24.orig/ext/gpgme/gpgme_n.c	2025-06-04 13:10:38.870341132 +0200
+++ ruby-gpgme-2.0.24/ext/gpgme/gpgme_n.c	2025-06-04 13:24:37.213160847 +0200
@@ -1634,70 +1634,6 @@
 }
 
 static VALUE
-rb_s_gpgme_op_trustlist_start (VALUE dummy, VALUE vctx, VALUE vpattern,
-                               VALUE vmax_level)
-{
-  gpgme_ctx_t ctx;
-  gpgme_error_t err;
-
-  CHECK_KEYLIST_NOT_IN_PROGRESS(vctx);
-
-  UNWRAP_GPGME_CTX(vctx, ctx);
-  if (!ctx)
-    rb_raise (rb_eArgError, "released ctx");
-  err = gpgme_op_trustlist_start (ctx, StringValueCStr(vpattern),
-                                  NUM2INT(vmax_level));
-  return LONG2NUM(err);
-}
-
-static VALUE
-rb_s_gpgme_op_trustlist_next (VALUE dummy, VALUE vctx, VALUE ritem)
-{
-  gpgme_ctx_t ctx;
-  gpgme_trust_item_t item;
-  gpgme_error_t err;
-  VALUE vitem;
-
-  CHECK_KEYLIST_NOT_IN_PROGRESS(vctx);
-
-  UNWRAP_GPGME_CTX(vctx, ctx);
-  if (!ctx)
-    rb_raise (rb_eArgError, "released ctx");
-
-  err = gpgme_op_trustlist_next (ctx, &item);
-  if (gpgme_err_code(err) == GPG_ERR_NO_ERROR)
-    {
-      vitem = WRAP_GPGME_TRUST_ITEM(item);
-      rb_iv_set (vitem, "@keyid", rb_str_new2 (item->keyid));
-      rb_iv_set (vitem, "@type", INT2FIX(item->type));
-      rb_iv_set (vitem, "@level", INT2FIX(item->level));
-      if (item->owner_trust)
-        rb_iv_set (vitem, "@owner_trust", rb_str_new2 (item->owner_trust));
-      rb_iv_set (vitem, "@validity", rb_str_new2 (item->validity));
-      if (item->name)
-        rb_iv_set (vitem, "@name", rb_str_new2 (item->name));
-      rb_ary_store (ritem, 0, vitem);
-    }
-  return LONG2NUM(err);
-}
-
-static VALUE
-rb_s_gpgme_op_trustlist_end (VALUE dummy, VALUE vctx)
-{
-  gpgme_ctx_t ctx;
-  gpgme_error_t err;
-
-  CHECK_KEYLIST_NOT_IN_PROGRESS(vctx);
-
-  UNWRAP_GPGME_CTX(vctx, ctx);
-  if (!ctx)
-    rb_raise (rb_eArgError, "released ctx");
-
-  err = gpgme_op_trustlist_end (ctx);
-  return LONG2NUM(err);
-}
-
-static VALUE
 rb_s_gpgme_op_decrypt (VALUE dummy, VALUE vctx, VALUE vcipher, VALUE vplain)
 {
   gpgme_ctx_t ctx;
@@ -2558,14 +2494,6 @@
   rb_define_module_function (mGPGME, "gpgme_op_card_edit_start",
                              rb_s_gpgme_op_card_edit_start, 5);
 
-  /* Trust Item Management */
-  rb_define_module_function (mGPGME, "gpgme_op_trustlist_start",
-                             rb_s_gpgme_op_trustlist_start, 3);
-  rb_define_module_function (mGPGME, "gpgme_op_trustlist_next",
-                             rb_s_gpgme_op_trustlist_next, 2);
-  rb_define_module_function (mGPGME, "gpgme_op_trustlist_end",
-                             rb_s_gpgme_op_trustlist_end, 1);
-
   /* Decrypt */
   rb_define_module_function (mGPGME, "gpgme_op_decrypt",
                              rb_s_gpgme_op_decrypt, 3);
@@ -2805,70 +2733,6 @@
   rb_define_const (mGPGME, "GPGME_SIG_MODE_CLEAR",
                    INT2FIX(GPGME_SIG_MODE_CLEAR));
 
-  /* gpgme_attr_t */
-  rb_define_const (mGPGME, "GPGME_ATTR_KEYID",
-                   INT2FIX(GPGME_ATTR_KEYID));
-  rb_define_const (mGPGME, "GPGME_ATTR_FPR",
-                   INT2FIX(GPGME_ATTR_FPR));
-  rb_define_const (mGPGME, "GPGME_ATTR_ALGO",
-                   INT2FIX(GPGME_ATTR_ALGO));
-  rb_define_const (mGPGME, "GPGME_ATTR_LEN",
-                   INT2FIX(GPGME_ATTR_LEN));
-  rb_define_const (mGPGME, "GPGME_ATTR_CREATED",
-                   INT2FIX(GPGME_ATTR_CREATED));
-  rb_define_const (mGPGME, "GPGME_ATTR_EXPIRE",
-                   INT2FIX(GPGME_ATTR_EXPIRE));
-  rb_define_const (mGPGME, "GPGME_ATTR_OTRUST",
-                   INT2FIX(GPGME_ATTR_OTRUST));
-  rb_define_const (mGPGME, "GPGME_ATTR_USERID",
-                   INT2FIX(GPGME_ATTR_USERID));
-  rb_define_const (mGPGME, "GPGME_ATTR_NAME",
-                   INT2FIX(GPGME_ATTR_NAME));
-  rb_define_const (mGPGME, "GPGME_ATTR_EMAIL",
-                   INT2FIX(GPGME_ATTR_EMAIL));
-  rb_define_const (mGPGME, "GPGME_ATTR_COMMENT",
-                   INT2FIX(GPGME_ATTR_COMMENT));
-  rb_define_const (mGPGME, "GPGME_ATTR_VALIDITY",
-                   INT2FIX(GPGME_ATTR_VALIDITY));
-  rb_define_const (mGPGME, "GPGME_ATTR_LEVEL",
-                   INT2FIX(GPGME_ATTR_LEVEL));
-  rb_define_const (mGPGME, "GPGME_ATTR_TYPE",
-                   INT2FIX(GPGME_ATTR_TYPE));
-  rb_define_const (mGPGME, "GPGME_ATTR_IS_SECRET",
-                   INT2FIX(GPGME_ATTR_IS_SECRET));
-  rb_define_const (mGPGME, "GPGME_ATTR_KEY_REVOKED",
-                   INT2FIX(GPGME_ATTR_KEY_REVOKED));
-  rb_define_const (mGPGME, "GPGME_ATTR_KEY_INVALID",
-                   INT2FIX(GPGME_ATTR_KEY_INVALID));
-  rb_define_const (mGPGME, "GPGME_ATTR_UID_REVOKED",
-                   INT2FIX(GPGME_ATTR_UID_REVOKED));
-  rb_define_const (mGPGME, "GPGME_ATTR_UID_INVALID",
-                   INT2FIX(GPGME_ATTR_UID_INVALID));
-  rb_define_const (mGPGME, "GPGME_ATTR_KEY_CAPS",
-                   INT2FIX(GPGME_ATTR_KEY_CAPS));
-  rb_define_const (mGPGME, "GPGME_ATTR_CAN_ENCRYPT",
-                   INT2FIX(GPGME_ATTR_CAN_ENCRYPT));
-  rb_define_const (mGPGME, "GPGME_ATTR_CAN_SIGN",
-                   INT2FIX(GPGME_ATTR_CAN_SIGN));
-  rb_define_const (mGPGME, "GPGME_ATTR_CAN_CERTIFY",
-                   INT2FIX(GPGME_ATTR_CAN_CERTIFY));
-  rb_define_const (mGPGME, "GPGME_ATTR_KEY_EXPIRED",
-                   INT2FIX(GPGME_ATTR_KEY_EXPIRED));
-  rb_define_const (mGPGME, "GPGME_ATTR_KEY_DISABLED",
-                   INT2FIX(GPGME_ATTR_KEY_DISABLED));
-  rb_define_const (mGPGME, "GPGME_ATTR_SERIAL",
-                   INT2FIX(GPGME_ATTR_SERIAL));
-  rb_define_const (mGPGME, "GPGME_ATTR_ISSUER",
-                   INT2FIX(GPGME_ATTR_ISSUER));
-  rb_define_const (mGPGME, "GPGME_ATTR_CHAINID",
-                   INT2FIX(GPGME_ATTR_CHAINID));
-  rb_define_const (mGPGME, "GPGME_ATTR_SIG_STATUS",
-                   INT2FIX(GPGME_ATTR_SIG_STATUS));
-  rb_define_const (mGPGME, "GPGME_ATTR_ERRTOK",
-                   INT2FIX(GPGME_ATTR_ERRTOK));
-  rb_define_const (mGPGME, "GPGME_ATTR_SIG_SUMMARY",
-                   INT2FIX(GPGME_ATTR_SIG_SUMMARY));
-
   /* gpgme_validity_t */
   rb_define_const (mGPGME, "GPGME_VALIDITY_UNKNOWN",
                    INT2FIX(GPGME_VALIDITY_UNKNOWN));
diff -ru ruby-gpgme-2.0.24.orig/lib/gpgme/constants.rb ruby-gpgme-2.0.24/lib/gpgme/constants.rb
--- ruby-gpgme-2.0.24.orig/lib/gpgme/constants.rb	2025-06-04 13:10:38.871084490 +0200
+++ ruby-gpgme-2.0.24/lib/gpgme/constants.rb	2025-06-04 13:14:08.345075542 +0200
@@ -1,36 +1,5 @@
 module GPGME
 
-  ATTR_ALGO = GPGME_ATTR_ALGO
-  ATTR_CAN_CERTIFY = GPGME_ATTR_CAN_CERTIFY
-  ATTR_CAN_ENCRYPT = GPGME_ATTR_CAN_ENCRYPT
-  ATTR_CAN_SIGN = GPGME_ATTR_CAN_SIGN
-  ATTR_CHAINID = GPGME_ATTR_CHAINID
-  ATTR_COMMENT = GPGME_ATTR_COMMENT
-  ATTR_CREATED = GPGME_ATTR_CREATED
-  ATTR_EMAIL = GPGME_ATTR_EMAIL
-  ATTR_ERRTOK = GPGME_ATTR_ERRTOK
-  ATTR_EXPIRE = GPGME_ATTR_EXPIRE
-  ATTR_FPR = GPGME_ATTR_FPR
-  ATTR_ISSUER = GPGME_ATTR_ISSUER
-  ATTR_IS_SECRET = GPGME_ATTR_IS_SECRET
-  ATTR_KEYID = GPGME_ATTR_KEYID
-  ATTR_KEY_CAPS = GPGME_ATTR_KEY_CAPS
-  ATTR_KEY_DISABLED = GPGME_ATTR_KEY_DISABLED
-  ATTR_KEY_EXPIRED = GPGME_ATTR_KEY_EXPIRED
-  ATTR_KEY_INVALID = GPGME_ATTR_KEY_INVALID
-  ATTR_KEY_REVOKED = GPGME_ATTR_KEY_REVOKED
-  ATTR_LEN = GPGME_ATTR_LEN
-  ATTR_LEVEL = GPGME_ATTR_LEVEL
-  ATTR_NAME = GPGME_ATTR_NAME
-  ATTR_OTRUST = GPGME_ATTR_OTRUST
-  ATTR_SERIAL = GPGME_ATTR_SERIAL
-  ATTR_SIG_STATUS = GPGME_ATTR_SIG_STATUS
-  ATTR_SIG_SUMMARY = GPGME_ATTR_SIG_SUMMARY
-  ATTR_TYPE = GPGME_ATTR_TYPE
-  ATTR_UID_INVALID = GPGME_ATTR_UID_INVALID
-  ATTR_UID_REVOKED = GPGME_ATTR_UID_REVOKED
-  ATTR_USERID = GPGME_ATTR_USERID
-  ATTR_VALIDITY = GPGME_ATTR_VALIDITY
   DATA_ENCODING_ARMOR = GPGME_DATA_ENCODING_ARMOR
   DATA_ENCODING_BASE64 = GPGME_DATA_ENCODING_BASE64
   DATA_ENCODING_BINARY = GPGME_DATA_ENCODING_BINARY
