diff --unified --recursive --text --new-file nzbget-25.3.orig/cmake/par2-turbo.cmake nzbget-25.3/cmake/par2-turbo.cmake
--- nzbget-25.3.orig/cmake/par2-turbo.cmake	2025-09-01 20:50:58.443363400 +0200
+++ nzbget-25.3/cmake/par2-turbo.cmake	2025-09-01 20:51:31.722598503 +0200
@@ -51,7 +51,7 @@
 ExternalProject_add(
 	par2-turbo
 	PREFIX			par2-turbo
-	GIT_REPOSITORY	https://github.com/nzbgetcom/par2cmdline-turbo.git
+	GIT_REPOSITORY	file://${CMAKE_SOURCE_DIR}/../par2cmdline-turbo
 	GIT_TAG			v1.3.0-20250808
 	TLS_VERIFY		TRUE
 	GIT_SHALLOW		TRUE
