diff --git a/src/cysignals/signals.pxd b/src/cysignals/signals.pxd
index 76320f7..531a9a8 100644
--- a/src/cysignals/signals.pxd
+++ b/src/cysignals/signals.pxd
@@ -1,3 +1,4 @@
+# cython: preliminary_late_includes_cy28=True
 #*****************************************************************************
 #  cysignals is free software: you can redistribute it and/or modify it
 #  under the terms of the GNU Lesser General Public License as published
