--- geeqie-2.5/meson.build	2024-09-21 17:11:19.000000000 +0200
+++ geeqie-2.5/meson.build.new	2024-09-22 20:28:00.882740820 +0200
@@ -845,10 +845,10 @@
 endif
 
 # Ancillary files test
-test_ancillary_files_sh = find_program('test-ancillary-files.sh', dirs : scriptsdir, required : true)
-test('Ancillary files', test_ancillary_files_sh, args: [meson.current_source_dir()], timeout: 100, suite : 'analysis')
+#test_ancillary_files_sh = find_program('test-ancillary-files.sh', dirs : scriptsdir, required : true)
+#test('Ancillary files', test_ancillary_files_sh, args: [meson.current_source_dir()], timeout: 100, suite : 'analysis')
 
-summary({'Ancillary files' : ['Test runs:', true]}, section : 'Testing', bool_yn : true)
+#summary({'Ancillary files' : ['Test runs:', true]}, section : 'Testing', bool_yn : true)
 
 # Unit tests
 if conf_data.get('ENABLE_UNIT_TESTS', 0) == 1
