diff --git a/lib/CL/devices/cuda/pocl-cuda.c b/lib/CL/devices/cuda/pocl-cuda.c
index 2cd290d4d..dce6283e4 100644
--- a/lib/CL/devices/cuda/pocl-cuda.c
+++ b/lib/CL/devices/cuda/pocl-cuda.c
@@ -753,7 +753,11 @@ pocl_cuda_init (unsigned j, cl_device_id dev, const char *parameters)
   /* Create context */
   if (ret != CL_INVALID_DEVICE)
     {
+#if CUDA_VERSION >= 13000
+      result = cuCtxCreate (&data->context, NULL, CU_CTX_MAP_HOST, data->device);
+#else
       result = cuCtxCreate (&data->context, CU_CTX_MAP_HOST, data->device);
+#endif
       if (CUDA_CHECK_ERROR (result, "cuCtxCreate"))
         ret = CL_INVALID_DEVICE;
     }
