--- a/crates/common/src/manager/mod.rs
+++ b/crates/common/src/manager/mod.rs
@@ -26,8 +26,7 @@ pub const SPAM_CLASSIFIER_KEY: &[u8] = "STALWART_SPAM_CLASSIFIER_MODEL.lz4".as_b
 
 
 #[cfg(not(feature = "enterprise"))]
-const DEFAULT_WEBADMIN_URL: &str =
-    "https://github.com/stalwartlabs/webadmin/releases/latest/download/webadmin-oss.zip";
+const DEFAULT_WEBADMIN_URL: &str = "file:///usr/share/stalwart-webadmin/webadmin.zip";
 
 impl ConfigManager {
     pub async fn fetch_resource(&self, resource_id: &str) -> Result<Vec<u8>, String> {
