--- a/slime.el
+++ b/slime.el
@@ -4548,7 +4548,7 @@ With prefix argument include internal symbols."
   (let ((file (seq-some (lambda (name)
                           (let ((f (expand-file-name name slime-path)))
                             (and (file-exists-p f) f)))
-                        '("doc/slime.info" "slime.info"))))
+                        '("/usr/share/info/slime.info.gz"))))
     (if file
         (info file)
       (message "No slime.info, run `make slime.info' in %s"
