diff -ru e-antic-2.0.2.orig/libeantic/test/fmpq_poly_extra/t-evaluate.c e-antic-2.0.2/libeantic/test/fmpq_poly_extra/t-evaluate.c
--- e-antic-2.0.2.orig/libeantic/test/fmpq_poly_extra/t-evaluate.c	2025-03-16 08:58:09.200375767 +0100
+++ e-antic-2.0.2/libeantic/test/fmpq_poly_extra/t-evaluate.c	2025-03-16 08:59:12.694170757 +0100
@@ -201,7 +201,7 @@
         arf_clear(d);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c e-antic-2.0.2/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c
--- e-antic-2.0.2.orig/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c	2025-03-16 08:58:09.200375767 +0100
+++ e-antic-2.0.2/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c	2025-03-16 08:59:12.694577591 +0100
@@ -148,7 +148,7 @@
 
         }
 
-        FLINT_TEST_CLEANUP(state)
+        FLINT_TEST_CLEAR(state)
     }
 
     return 0;
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-has_real_root.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-has_real_root.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-has_real_root.c	2025-03-16 08:58:09.199999734 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-has_real_root.c	2025-03-16 08:59:12.694985960 +0100
@@ -70,7 +70,7 @@
     }
 
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c	2025-03-16 08:58:09.199999734 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c	2025-03-16 08:59:12.695172789 +0100
@@ -392,7 +392,7 @@
             }
         }
 
-        FLINT_TEST_CLEANUP(state)
+        FLINT_TEST_CLEAR(state)
 
         printf("PASS\n");
     }
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c	2025-03-16 08:59:12.695436166 +0100
@@ -44,7 +44,7 @@
         fmpz_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots.c	2025-03-16 08:58:09.199999734 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots.c	2025-03-16 08:59:12.695589400 +0100
@@ -47,7 +47,7 @@
         fmpz_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c	2025-03-16 08:59:12.695755766 +0100
@@ -59,7 +59,7 @@
         fmpz_poly_clear(q);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c	2025-03-16 08:59:12.695914658 +0100
@@ -80,7 +80,7 @@
         fmpz_poly_clear(q);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c	2025-03-16 08:59:12.696098274 +0100
@@ -103,7 +103,7 @@
         fmpz_poly_clear(q);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c	2025-03-16 08:59:12.696259820 +0100
@@ -48,6 +48,6 @@
         fmpz_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c	2025-03-16 08:59:12.696415010 +0100
@@ -89,7 +89,7 @@
         _fmpq_vec_clear(v2, n);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c	2025-03-16 08:59:12.696587451 +0100
@@ -146,7 +146,7 @@
             fmpz_poly_clear(b);
         }
 
-        FLINT_TEST_CLEANUP(state)
+        FLINT_TEST_CLEAR(state)
     }
 
     return 0;
diff -ru e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-squarefree_part.c e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-squarefree_part.c
--- e-antic-2.0.2.orig/libeantic/test/fmpz_poly_extra/t-squarefree_part.c	2025-03-16 08:58:09.199361303 +0100
+++ e-antic-2.0.2/libeantic/test/fmpz_poly_extra/t-squarefree_part.c	2025-03-16 08:59:12.696770788 +0100
@@ -67,6 +67,6 @@
 
     printf("PASS\n");
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf/t-init.c e-antic-2.0.2/libeantic/test/renf/t-init.c
--- e-antic-2.0.2.orig/libeantic/test/renf/t-init.c	2025-03-16 08:58:09.199026407 +0100
+++ e-antic-2.0.2/libeantic/test/renf/t-init.c	2025-03-16 08:59:12.699052686 +0100
@@ -127,7 +127,7 @@
     check_init(p, -0.48296797364451214901, 1e-15, 3, 100, state);
 
     fmpq_poly_clear(p);
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf/t-init_nth_root_fmpq.c e-antic-2.0.2/libeantic/test/renf/t-init_nth_root_fmpq.c
--- e-antic-2.0.2.orig/libeantic/test/renf/t-init_nth_root_fmpq.c	2025-03-16 08:58:09.199026407 +0100
+++ e-antic-2.0.2/libeantic/test/renf/t-init_nth_root_fmpq.c	2025-03-16 08:59:12.699212486 +0100
@@ -52,7 +52,7 @@
         fmpq_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff -ru e-antic-2.0.2.orig/libeantic/test/renf/t-randtest.c e-antic-2.0.2/libeantic/test/renf/t-randtest.c
--- e-antic-2.0.2.orig/libeantic/test/renf/t-randtest.c	2025-03-16 08:58:09.199026407 +0100
+++ e-antic-2.0.2/libeantic/test/renf/t-randtest.c	2025-03-16 08:59:12.699355454 +0100
@@ -51,6 +51,6 @@
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf/t-real_embeddings.c e-antic-2.0.2/libeantic/test/renf/t-real_embeddings.c
--- e-antic-2.0.2.orig/libeantic/test/renf/t-real_embeddings.c	2025-03-16 08:58:09.199026407 +0100
+++ e-antic-2.0.2/libeantic/test/renf/t-real_embeddings.c	2025-03-16 08:59:12.699504987 +0100
@@ -37,7 +37,7 @@
         for (i = 0; i < n; i++) renf_clear(nfemb + i);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-ceil.c e-antic-2.0.2/libeantic/test/renf_elem/t-ceil.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-ceil.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-ceil.c	2025-03-16 08:59:12.696982132 +0100
@@ -219,7 +219,7 @@
         arb_clear(e);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-cmp.c e-antic-2.0.2/libeantic/test/renf_elem/t-cmp.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-cmp.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-cmp.c	2025-03-16 08:59:12.697221483 +0100
@@ -292,6 +292,6 @@
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-cmp_fmpq.c e-antic-2.0.2/libeantic/test/renf_elem/t-cmp_fmpq.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-cmp_fmpq.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-cmp_fmpq.c	2025-03-16 08:59:12.697441417 +0100
@@ -94,6 +94,6 @@
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-equal_fmpq.c e-antic-2.0.2/libeantic/test/renf_elem/t-equal_fmpq.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-equal_fmpq.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-equal_fmpq.c	2025-03-16 08:59:12.697600239 +0100
@@ -49,6 +49,6 @@
     fmpq_clear(d);
     renf_clear(nf);
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-floor.c e-antic-2.0.2/libeantic/test/renf_elem/t-floor.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-floor.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-floor.c	2025-03-16 08:59:12.697761506 +0100
@@ -264,7 +264,7 @@
     }
 
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-gen.c e-antic-2.0.2/libeantic/test/renf_elem/t-gen.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-gen.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-gen.c	2025-03-16 08:59:12.697944703 +0100
@@ -60,7 +60,7 @@
         fmpq_clear(d);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     printf("PASS\n");
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-get_arb.c e-antic-2.0.2/libeantic/test/renf_elem/t-get_arb.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-get_arb.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-get_arb.c	2025-03-16 08:59:12.698108344 +0100
@@ -76,7 +76,7 @@
         renf_elem_clear(a, nf);
     }
     renf_clear(nf);
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-get_cfrac.c e-antic-2.0.2/libeantic/test/renf_elem/t-get_cfrac.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-get_cfrac.c	2025-03-16 08:58:09.197999720 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-get_cfrac.c	2025-03-16 08:59:12.698260252 +0100
@@ -116,6 +116,6 @@
 
     fmpq_clear(d);
     
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-get_str_pretty.c e-antic-2.0.2/libeantic/test/renf_elem/t-get_str_pretty.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-get_str_pretty.c	2025-03-16 08:58:09.197523952 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-get_str_pretty.c	2025-03-16 08:59:12.698419074 +0100
@@ -113,7 +113,7 @@
     renf_elem_clear(a, nf);
     renf_clear(nf);
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-is_rational.c e-antic-2.0.2/libeantic/test/renf_elem/t-is_rational.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-is_rational.c	2025-03-16 08:58:09.197523952 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-is_rational.c	2025-03-16 08:59:12.698567070 +0100
@@ -67,7 +67,7 @@
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff -ru e-antic-2.0.2.orig/libeantic/test/renf_elem/t-pow.c e-antic-2.0.2/libeantic/test/renf_elem/t-pow.c
--- e-antic-2.0.2.orig/libeantic/test/renf_elem/t-pow.c	2025-03-16 08:58:09.197523952 +0100
+++ e-antic-2.0.2/libeantic/test/renf_elem/t-pow.c	2025-03-16 08:59:12.698709898 +0100
@@ -52,6 +52,6 @@
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff -ru e-antic-2.0.2.orig/libeantic/test/renfxx/t-ceil.cpp e-antic-2.0.2/libeantic/test/renfxx/t-ceil.cpp
--- e-antic-2.0.2.orig/libeantic/test/renfxx/t-ceil.cpp	2025-03-16 08:58:09.196999713 +0100
+++ e-antic-2.0.2/libeantic/test/renfxx/t-ceil.cpp	2025-03-16 08:59:14.133426115 +0100
@@ -76,7 +76,7 @@
         fmpq_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff -ru e-antic-2.0.2.orig/libeantic/test/renfxx/t-get.cpp e-antic-2.0.2/libeantic/test/renfxx/t-get.cpp
--- e-antic-2.0.2.orig/libeantic/test/renfxx/t-get.cpp	2025-03-16 08:58:09.195999706 +0100
+++ e-antic-2.0.2/libeantic/test/renfxx/t-get.cpp	2025-03-16 08:59:14.134402514 +0100
@@ -37,7 +37,7 @@
         a.get_renf_elem();
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
