diff --git a/application/CMakeLists.txt b/application/CMakeLists.txt
index f0a44d8..35d9219 100644
--- a/application/CMakeLists.txt
+++ b/application/CMakeLists.txt
@@ -86,9 +86,11 @@ set(f3d_compile_options_private "")
 set(f3d_compile_options_public "")
 set(f3d_link_options_public "")
 
-# Headless EGL build
-if (VTK_OPENGL_HAS_EGL)
-  target_compile_definitions(f3d PRIVATE F3D_HEADLESS_BUILD)
+if(VTK_VERSION VERSION_LESS 9.3.20240914)
+  # Headless EGL build
+  if (VTK_OPENGL_HAS_EGL)
+    target_compile_definitions(f3d PRIVATE F3D_HEADLESS_BUILD)
+  endif ()
 endif ()
 
 # F3D_STRICT_BUILD
