diff --git a/configure.in b/configure.in
index f4f09ef..1dd6b4d 100644
--- a/configure.in
+++ b/configure.in
@@ -138,7 +138,7 @@ AC_LANG_RESTORE
 
 AC_CHECK_HEADERS(ndbm.h db1/ndbm.h)
 
-AC_SEARCH_LIBS(dbm_open, db1 db, , AC_MSG_ERROR("Cannot find library containing dbm_open"))
+AC_SEARCH_LIBS(dbm_open, db1 db gdbm_compat, , AC_MSG_ERROR("Cannot find library containing dbm_open"))
 
 x11xkbdir=/usr/share/X11/xkb
 test -d $x11xkbdir || AC_MSG_ERROR("Directory $x11xkbdir does not exist")
