diff -wbBur freehdl-0.0.8/freehdl/kernel-db.hh freehdl-0.0.8.my/freehdl/kernel-db.hh
--- freehdl-0.0.8/freehdl/kernel-db.hh	2025-09-28 23:56:00.588823147 +0300
+++ freehdl-0.0.8.my/freehdl/kernel-db.hh	2025-09-28 23:51:11.342159847 +0300
@@ -608,7 +608,7 @@
     return *d;
   }
 
-  typename kind::data_type &get (db_basic_key key) { return get (this->get_key (key)); }
+//  typename kind::data_type &get (db_basic_key key) { return get (this->get_key (key)); }
 };
 
 
diff -wbBur freehdl-0.0.8/freehdl/std-vhdl-types.hh freehdl-0.0.8.my/freehdl/std-vhdl-types.hh
--- freehdl-0.0.8/freehdl/std-vhdl-types.hh	2025-09-28 23:56:00.612156480 +0300
+++ freehdl-0.0.8.my/freehdl/std-vhdl-types.hh	2025-09-28 23:50:44.838826821 +0300
@@ -1494,7 +1494,7 @@
   new_array.info->add_ref();
 
   /* Allocate memory for the data */
-  const int mem_size = length * ainfo->element_type->size();
+  const int mem_size = length * ainfo->element_type->size;
   new_array.data = (char*)internal_dynamic_alloc(mem_size);
 
   /* Note, scalar(...) can be evaluated at compile time. */
