diff --git a/src/code/incidencewrapper.cpp b/src/code/incidencewrapper.cpp
index f5232f7..55c70f2 100644
--- a/src/code/incidencewrapper.cpp
+++ b/src/code/incidencewrapper.cpp
@@ -36,7 +36,6 @@ IncidenceWrapper::IncidenceWrapper(QObject *parent)
     Akonadi::ItemFetchScope scope;
     scope.fetchFullPayload();
     scope.fetchAllAttributes();
-    scope.setFetchRelations(true);
     scope.setAncestorRetrieval(Akonadi::ItemFetchScope::Parent);
     setFetchScope(scope);
     
