diff -ru radcli-1.4.0.orig/lib/nettle-hmac.c radcli-1.4.0/lib/nettle-hmac.c
--- radcli-1.4.0.orig/lib/nettle-hmac.c	2026-05-01 13:22:06.167520228 +0200
+++ radcli-1.4.0/lib/nettle-hmac.c	2026-05-01 13:25:28.639205017 +0200
@@ -37,6 +37,6 @@
     memset(digest, 0, MD5_DIGEST_SIZE);
     hmac_md5_set_key(&md5, key_len, key);
     hmac_md5_update(&md5, data_len, data);
-    hmac_md5_digest(&md5, MD5_DIGEST_SIZE, digest);
+    hmac_md5_digest(&md5, digest);
 }
 
diff -ru radcli-1.4.0.orig/lib/rc-md5.c radcli-1.4.0/lib/rc-md5.c
--- radcli-1.4.0.orig/lib/rc-md5.c	2026-05-01 13:22:06.166520225 +0200
+++ radcli-1.4.0/lib/rc-md5.c	2026-05-01 13:38:01.948189973 +0200
@@ -20,10 +20,10 @@
 void rc_md5_calc(unsigned char *output, unsigned char const *input,
 		 size_t inlen)
 {
-	MD5_CTX	context;
+	struct md5_ctx context;
 
-	MD5Init(&context);
-	MD5Update(&context, input, inlen);
-	MD5Final(output, &context);
+	md5_init(&context);
+	md5_update(&context, inlen, input);
+	md5_digest(&context, output);
 }
 
diff -ru radcli-1.4.0.orig/lib/rc-md5.h radcli-1.4.0/lib/rc-md5.h
--- radcli-1.4.0.orig/lib/rc-md5.h	2026-05-01 13:22:06.166520225 +0200
+++ radcli-1.4.0/lib/rc-md5.h	2026-05-01 13:22:52.743912505 +0200
@@ -13,7 +13,7 @@
 
 #ifdef HAVE_NETTLE
 
-#include <nettle/md5-compat.h>
+#include <nettle/md5.h>
 
 #else
 
