diff --git a/poetry.lock b/poetry.lock
index 267872aa..d9faf1a7 100644
--- a/poetry.lock
+++ b/poetry.lock
@@ -2355,7 +2355,7 @@ test = ["pyfakefs", "pytest (>=6,!=8.1.*)"]
 type = ["pygobject-stubs", "pytest-mypy", "shtab", "types-pywin32"]
 
 [[package]]
-name = "lia-web"
+name = "cross-web"
 version = "0.2.3"
 description = "A library for working with web frameworks"
 optional = false
diff --git a/pyproject.toml b/pyproject.toml
index ae735cc9..50df9bb4 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,7 +19,7 @@ dependencies = [
   "typing-extensions>=4.5.0",
   "python-dateutil>=2.7",
   "packaging>=23",
-  "lia-web (>=0.2.1)",
+  "cross-web (>=0.2.1)",
 ]
 
 [project.urls]
diff --git a/strawberry/aiohttp/views.py b/strawberry/aiohttp/views.py
index 370ceb31..5cded93a 100644
--- a/strawberry/aiohttp/views.py
+++ b/strawberry/aiohttp/views.py
@@ -10,7 +10,7 @@ from typing import (
 )
 
 from aiohttp import ClientConnectionResetError, http, web
-from lia import AiohttpHTTPRequestAdapter, HTTPException
+from cross_web import AiohttpHTTPRequestAdapter, HTTPException
 
 from strawberry.http.async_base_view import (
     AsyncBaseHTTPView,
diff --git a/strawberry/asgi/__init__.py b/strawberry/asgi/__init__.py
index 2ee77394..0bc27eec 100644
--- a/strawberry/asgi/__init__.py
+++ b/strawberry/asgi/__init__.py
@@ -8,7 +8,7 @@ from typing import (
     TypeGuard,
 )
 
-from lia import HTTPException, StarletteRequestAdapter
+from cross_web import HTTPException, StarletteRequestAdapter
 from starlette import status
 from starlette.requests import Request
 from starlette.responses import (
diff --git a/strawberry/chalice/views.py b/strawberry/chalice/views.py
index 98eca7ac..33be5ae8 100644
--- a/strawberry/chalice/views.py
+++ b/strawberry/chalice/views.py
@@ -4,7 +4,7 @@ import warnings
 from typing import TYPE_CHECKING
 
 from chalice.app import Request, Response
-from lia import ChaliceHTTPRequestAdapter, HTTPException
+from cross_web import ChaliceHTTPRequestAdapter, HTTPException
 
 from strawberry.http.sync_base_view import SyncBaseHTTPView
 from strawberry.http.temporal_response import TemporalResponse
diff --git a/strawberry/channels/handlers/http_handler.py b/strawberry/channels/handlers/http_handler.py
index 7e3382b8..8b90660d 100644
--- a/strawberry/channels/handlers/http_handler.py
+++ b/strawberry/channels/handlers/http_handler.py
@@ -14,7 +14,7 @@ from channels.generic.http import AsyncHttpConsumer
 from django.conf import settings
 from django.core.files import uploadhandler
 from django.http.multipartparser import MultiPartParser
-from lia import AsyncHTTPRequestAdapter, FormData, HTTPException, SyncHTTPRequestAdapter
+from cross_web import AsyncHTTPRequestAdapter, FormData, HTTPException, SyncHTTPRequestAdapter
 
 from strawberry.http.async_base_view import AsyncBaseHTTPView
 from strawberry.http.sync_base_view import SyncBaseHTTPView
diff --git a/strawberry/django/views.py b/strawberry/django/views.py
index ce57d0d7..0cd74a43 100644
--- a/strawberry/django/views.py
+++ b/strawberry/django/views.py
@@ -22,7 +22,7 @@ from django.template.exceptions import TemplateDoesNotExist
 from django.template.loader import render_to_string
 from django.utils.decorators import classonlymethod
 from django.views.generic import View
-from lia import AsyncDjangoHTTPRequestAdapter, DjangoHTTPRequestAdapter, HTTPException
+from cross_web import AsyncDjangoHTTPRequestAdapter, DjangoHTTPRequestAdapter, HTTPException
 
 from strawberry.http.async_base_view import AsyncBaseHTTPView
 from strawberry.http.sync_base_view import SyncBaseHTTPView
diff --git a/strawberry/fastapi/router.py b/strawberry/fastapi/router.py
index f90bbe0d..8a153f13 100644
--- a/strawberry/fastapi/router.py
+++ b/strawberry/fastapi/router.py
@@ -14,7 +14,7 @@ from fastapi import APIRouter, Depends, params
 from fastapi.datastructures import Default
 from fastapi.routing import APIRoute
 from fastapi.utils import generate_unique_id
-from lia import HTTPException, StarletteRequestAdapter
+from cross_web import HTTPException, StarletteRequestAdapter
 from starlette import status
 from starlette.background import BackgroundTasks  # noqa: TC002
 from starlette.requests import HTTPConnection, Request
diff --git a/strawberry/flask/views.py b/strawberry/flask/views.py
index 30cecf95..bf72698d 100644
--- a/strawberry/flask/views.py
+++ b/strawberry/flask/views.py
@@ -7,7 +7,7 @@ from typing import (
     TypeGuard,
 )
 
-from lia import AsyncFlaskHTTPRequestAdapter, FlaskHTTPRequestAdapter, HTTPException
+from cross_web import AsyncFlaskHTTPRequestAdapter, FlaskHTTPRequestAdapter, HTTPException
 
 from flask import Request, Response, render_template_string, request
 from flask.views import View
diff --git a/strawberry/http/async_base_view.py b/strawberry/http/async_base_view.py
index 4ef71381..b790b2ca 100644
--- a/strawberry/http/async_base_view.py
+++ b/strawberry/http/async_base_view.py
@@ -14,7 +14,7 @@ from typing import (
 )
 
 from graphql import GraphQLError
-from lia import AsyncHTTPRequestAdapter, HTTPException
+from cross_web import AsyncHTTPRequestAdapter, HTTPException
 
 from strawberry.exceptions import MissingQueryError
 from strawberry.file_uploads.utils import replace_placeholders_with_files
diff --git a/strawberry/http/base.py b/strawberry/http/base.py
index 2bba5b8b..91459776 100644
--- a/strawberry/http/base.py
+++ b/strawberry/http/base.py
@@ -3,7 +3,7 @@ from collections.abc import Mapping
 from typing import Any, Generic
 from typing_extensions import Protocol
 
-from lia import HTTPException
+from cross_web import HTTPException
 
 from strawberry.http import GraphQLRequestData
 from strawberry.http.ides import GraphQL_IDE, get_graphql_ide_html
diff --git a/strawberry/http/sync_base_view.py b/strawberry/http/sync_base_view.py
index 1d4529ed..4473639e 100644
--- a/strawberry/http/sync_base_view.py
+++ b/strawberry/http/sync_base_view.py
@@ -7,7 +7,7 @@ from typing import (
 )
 
 from graphql import GraphQLError
-from lia import HTTPException, SyncHTTPRequestAdapter
+from cross_web import HTTPException, SyncHTTPRequestAdapter
 
 from strawberry.exceptions import MissingQueryError
 from strawberry.file_uploads.utils import replace_placeholders_with_files
diff --git a/strawberry/litestar/controller.py b/strawberry/litestar/controller.py
index 5384493e..4e469e46 100644
--- a/strawberry/litestar/controller.py
+++ b/strawberry/litestar/controller.py
@@ -14,7 +14,7 @@ from typing import (
     TypeGuard,
 )
 
-from lia import HTTPException, LitestarRequestAdapter
+from cross_web import HTTPException, LitestarRequestAdapter
 from litestar import (
     Controller,
     MediaType,
diff --git a/strawberry/quart/views.py b/strawberry/quart/views.py
index 7f8bf8ca..bd46994e 100644
--- a/strawberry/quart/views.py
+++ b/strawberry/quart/views.py
@@ -5,7 +5,7 @@ from datetime import timedelta
 from json.decoder import JSONDecodeError
 from typing import TYPE_CHECKING, ClassVar, TypeGuard, Union
 
-from lia import HTTPException, QuartHTTPRequestAdapter
+from cross_web import HTTPException, QuartHTTPRequestAdapter
 from quart import Request, Response, Websocket, request, websocket
 from quart.ctx import has_websocket_context
 from quart.views import View
diff --git a/strawberry/sanic/views.py b/strawberry/sanic/views.py
index 0788265e..d30392df 100644
--- a/strawberry/sanic/views.py
+++ b/strawberry/sanic/views.py
@@ -8,7 +8,7 @@ from typing import (
     TypeGuard,
 )
 
-from lia import HTTPException, SanicHTTPRequestAdapter
+from cross_web import HTTPException, SanicHTTPRequestAdapter
 from sanic.request import Request
 from sanic.response import HTTPResponse, html
 from sanic.views import HTTPMethodView
