diff --git a/Modules/Core/Common/include/itkFloatingPointExceptions.h b/Modules/Core/Common/include/itkFloatingPointExceptions.h
index fd2f0ce256..440f8dd940 100644
--- a/Modules/Core/Common/include/itkFloatingPointExceptions.h
+++ b/Modules/Core/Common/include/itkFloatingPointExceptions.h
@@ -18,6 +18,8 @@
 #ifndef itkFloatingPointExceptions_h
 #define itkFloatingPointExceptions_h

+#include <cstdint>
+
 #include "itkMacro.h" // for ITKCommon_EXPORT
 #include "itkSingletonMacro.h"

