From 47a3b214bfa4ca5ff3b2ed77bd30f2f705eaad25 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 25 Mar 2025 03:05:59 +0100
Subject: [PATCH] google: fix getting phase

---
 webfontkitgenerator/google.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/webfontkitgenerator/google.py b/webfontkitgenerator/google.py
index 29a8764..597a559 100644
--- a/webfontkitgenerator/google.py
+++ b/webfontkitgenerator/google.py
@@ -122,7 +122,7 @@ def on_google_response(
             response_headers = message.get_response_headers()
 
             print(
-                f'Google Fonts Response Status {status.get_phrase(status_code)}'
+                f'Google Fonts Response Status {status.get_phrase()}'
             )
 
             if status == Soup.Status.OK and 'kind' in data:
