diff --git a/amd_openvx_extensions/amd_media/encoder.cpp b/amd_openvx_extensions/amd_media/encoder.cpp
index d112580f..97bccab0 100644
--- a/amd_openvx_extensions/amd_media/encoder.cpp
+++ b/amd_openvx_extensions/amd_media/encoder.cpp
@@ -29,6 +29,7 @@ THE SOFTWARE.
 #include <string>
 #include <vector>
 #include <sstream>
+#include <iostream>
 
 // GPU configuration
 #define ENCODE_ENABLE_GPU       1         // enable use of GPU buffers
