diff --git a/primecountpy/defs.pxd b/primecountpy/defs.pxd
index fe98b1c..148b25a 100644
--- a/primecountpy/defs.pxd
+++ b/primecountpy/defs.pxd
@@ -1,13 +1,17 @@
 # distutils: libraries = primesieve primecount
 # distutils: language = c++
 
-from libc.stdint cimport int64_t
+from libc.stdint cimport int64_t, uint64_t
 from libcpp.string cimport string as cppstring
 
 cdef extern from "primecount.hpp" namespace "primecount":
+    ctypedef struct pc_int128_t:
+        uint64_t lo
+        int64_t hi
+
     int64_t pi(int64_t x)
 
-    cppstring pi(const cppstring& x)
+    pc_int128_t pi(const pc_int128_t x)
 
     int64_t nth_prime(int64_t n)
 
@@ -16,7 +20,4 @@ cdef extern from "primecount.hpp" namespace "primecount":
     void set_num_threads(int num_threads)
     int get_num_threads()
 
-    cppstring get_max_x()
-    cppstring get_max_x(double alpha)
-
     cppstring primecount_version()
diff --git a/primecountpy/primecount.pyx b/primecountpy/primecount.pyx
index c930de0..8750eaa 100644
--- a/primecountpy/primecount.pyx
+++ b/primecountpy/primecount.pyx
@@ -11,9 +11,8 @@ Interface to the `primecount <https://github.com/kimwalisch/primecount>`_ C++ li
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
 
-from libc.stdint cimport int64_t
+from libc.stdint cimport int64_t, uint64_t
 from libcpp.string cimport string as cppstring
-from cpython.long cimport PyLong_FromString
 
 from cysignals.signals cimport sig_on, sig_off
 cimport defs as pcount
@@ -62,12 +61,14 @@ cpdef prime_pi_128(n):
         >>> prime_pi_128(10**10)
         455052511
     """
-    cdef cppstring s = str(n).encode('ascii')
-    cdef bytes ans
+    cdef pcount.pc_int128_t s
+    s.lo = (n % 2**64)
+    s.hi = (n >> 64)
+    cdef pcount.pc_int128_t ans
     sig_on()
     ans = pcount.pi(s)
     sig_off()
-    return PyLong_FromString(ans, NULL, 10)
+    return ans.lo | (ans.hi << 64)
 
 cpdef int64_t nth_prime(int64_t n) except -1:
     r"""
