diff --git a/src/fs/file.rs b/src/fs/file.rs
index 306d33e6..5cafcad2 100644
--- a/src/fs/file.rs
+++ b/src/fs/file.rs
@@ -334,7 +334,7 @@ impl<'dir> File<'dir> {
     #[cfg(unix)]
     pub fn is_executable_file(&self) -> bool {
         let bit = modes::USER_EXECUTE;
-        if self.is_file() {
+        if !self.is_file() {
             return false;
         }
         let Ok(md) = self.metadata() else {
