diff --git a/libgnucash/app-utils/gnc-quotes.cpp b/libgnucash/app-utils/gnc-quotes.cpp
index 1fdf64a385..5ad2a81289 100644
--- a/libgnucash/app-utils/gnc-quotes.cpp
+++ b/libgnucash/app-utils/gnc-quotes.cpp
@@ -20,7 +20,13 @@
  * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
 \ *******************************************************************/
 
-#include <boost/process/environment.hpp>
+#define BOOST_PROCESS_VERSION 1
+#include <boost/process/v1/env.hpp>
+#include <boost/process/v1/async.hpp>
+#include <boost/process/v1/child.hpp>
+#include <boost/process/v1/environment.hpp>
+#include <boost/process/v1/io.hpp>
+#include <boost/process/v1/search_path.hpp>
 #include <config.h>
 #include <qoflog.h>
 
@@ -39,7 +45,6 @@
 #ifdef BOOST_WINDOWS_API
 #include <boost/process/windows.hpp>
 #endif
-#include <boost/process.hpp>
 #include <boost/regex.hpp>
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/json_parser.hpp>
