diff -ru gnucash-5.13.orig/CMakeLists.txt gnucash-5.13/CMakeLists.txt
--- gnucash-5.13.orig/CMakeLists.txt	2025-10-15 20:19:14.534961905 +0200
+++ gnucash-5.13/CMakeLists.txt	2025-10-15 20:19:41.504852530 +0200
@@ -599,7 +599,7 @@
 if (NOT DEFINED ${BOOST_ROOT})
   set(BOOST_ROOT $ENV{BOOST_ROOT})
 endif()
-find_package (Boost 1.67.0 COMPONENTS date_time filesystem locale program_options regex system)
+find_package (Boost 1.67.0 COMPONENTS date_time filesystem locale program_options regex)
 
 if (Boost_FOUND)
   include_directories(${Boost_INCLUDE_DIRS})
