diff --git i/CMakeLists.txt w/CMakeLists.txt
index b26fa3ddb1..6685f38f8c 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -608,7 +608,7 @@ if (Boost_FOUND)
   if (Boost_VERSION VERSION_LESS "1.90.0")
     find_package(Boost 1.67.0  COMPONENTS date_time filesystem locale program_options regex)
   else()
-    find_package(Boost 1.67.0  COMPONENTS date_time locale program_options regex)
+    find_package(Boost 1.67.0  COMPONENTS date_time filesystem locale program_options regex)
   endif()
  endif()
 endif()
