diff --git a/src/libponyc/pkg/package.c b/src/libponyc/pkg/package.c
index 2ee105ac0..d438cce05 100644
--- a/src/libponyc/pkg/package.c
+++ b/src/libponyc/pkg/package.c
@@ -854,10 +854,6 @@ bool package_init(pass_opt_t* opt)
   package_add_paths(getenv("PONYPATH"), opt);
 
   // Finally we add OS specific paths.
-#ifdef PLATFORM_IS_POSIX_BASED
-  add_path("/usr/local/lib", opt);
-  add_path("/opt/local/lib", opt);
-#endif
 #ifdef PLATFORM_IS_DRAGONFLY
   add_path("/usr/local/cxx_atomics", opt);
 #endif
diff --git a/src/libponyrt/CMakeLists.txt b/src/libponyrt/CMakeLists.txt
index f8f13fd10..78b8d9ca4 100644
--- a/src/libponyrt/CMakeLists.txt
+++ b/src/libponyrt/CMakeLists.txt
@@ -174,9 +174,6 @@ if (MSVC)
         COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:libponyrt> ${PONY_OUTPUT_DIR}/libponyrt.lib
     )
 else()
-    target_include_directories(libponyrt
-        PRIVATE /usr/local/include
-    )
 
     if(${CMAKE_HOST_SYSTEM_NAME} MATCHES "Darwin")
         set(LIBPONYRT_ARCHIVE libponyrt.a)
