--- a/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu	2026-06-11 03:20:25.369162071 +0000
+++ b/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu	2026-06-11 03:20:50.536240145 +0000
@@ -43,7 +43,11 @@
 #include <cub/device/device_radix_sort.cuh>
 #include <cub/util_type.cuh>
 
+#if DUMP_TENSOR_LEVEL > 0
 #include "contrib_ops/cuda/utils/dump_cuda_tensor.h"
+#else
+#include "contrib_ops/cpu/utils/debug_macros.h"
+#endif
 
 namespace ort_fastertransformer {
 static constexpr int WARP_SIZE = 32;
