diff --git a/src/lightdm-deepin-greeter/greeterworker.h b/src/lightdm-deepin-greeter/greeterworker.h
index dbc5a6d5..9a61c118 100644
--- a/src/lightdm-deepin-greeter/greeterworker.h
+++ b/src/lightdm-deepin-greeter/greeterworker.h
@@ -14,2 +14,7 @@
+#include <QtCore/qglobal.h>
+#pragma push_macro("Q_DECL_EXPORT")
+#undef Q_DECL_EXPORT
+#define Q_DECL_EXPORT Q_DECL_IMPORT
 #include <QLightDM/Greeter>
 #include <QLightDM/SessionsModel>
+#pragma pop_macro("Q_DECL_EXPORT")
diff --git a/src/lightdm-deepin-greeter/sessionwidget.h b/src/lightdm-deepin-greeter/sessionwidget.h
index 2e5beb16..c3beee06 100644
--- a/src/lightdm-deepin-greeter/sessionwidget.h
+++ b/src/lightdm-deepin-greeter/sessionwidget.h
@@ -12,2 +12,7 @@
+#include <QtCore/qglobal.h>
+#pragma push_macro("Q_DECL_EXPORT")
+#undef Q_DECL_EXPORT
+#define Q_DECL_EXPORT Q_DECL_IMPORT
 #include <QLightDM/SessionsModel>
 #include <QLightDM/UsersModel>
+#pragma pop_macro("Q_DECL_EXPORT")
