diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b767c696..aae64f01a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -328,7 +328,7 @@ endif()
 
 ## OpenCV
 # Ensure we have at least version 4.6.0
-find_package(OpenCV 4.6.0 REQUIRED COMPONENTS core imgcodecs calib3d photo)
+find_package(OpenCV 5 REQUIRED COMPONENTS core imgcodecs photo)
 set_package_properties(OpenCV PROPERTIES DESCRIPTION "OpenCV library" URL "https://opencv.org" TYPE REQUIRED PURPOSE "Standard library of image processing functions.")
 include_directories(SYSTEM ${OpenCV_INCLUDE_DIRS})
 
