diff -Nur dde-session-1.2.9/src/dde-session/environmentsmanager.cpp dde-session-1.2.9-new/src/dde-session/environmentsmanager.cpp
--- dde-session-1.2.9/src/dde-session/environmentsmanager.cpp	2024-05-16 17:21:37.000000000 +0800
+++ dde-session-1.2.9-new/src/dde-session/environmentsmanager.cpp	2024-07-20 20:51:47.125907148 +0800
@@ -85,6 +85,15 @@
     m_envMap.insert("XDG_CURRENT_DESKTOP", "DDE");
     m_envMap.insert("QT_DBL_CLICK_DIST", QString::number(15 * scaleFactor));
     m_envMap.insert("QT_LINUX_ACCESSIBILITY_ALWAYS_ON", "1");
+    
+    QByteArray sessionType = qgetenv("XDG_SESSION_TYPE");
+    if (sessionType == "x11") {
+        m_envMap.insert("QT_QPA_PLATFORM", "dxcb:xcb");
+    } else if (sessionType == "wayland") {
+        m_envMap.insert("QT_QPA_PLATFORM", "wayland");
+        m_envMap.insert("QT_WAYLAND_SHELL_INTEGRATION", "xdg-shell;wl-shell;ivi-shell;qt-shell;");
+    }
+    
 
     // 浮点数相等
     if (qFuzzyIsNull(scaleFactor - 1.0)) {
