diff --git a/j2cli/__init__.py b/j2cli/__init__.py
index bc39e44..148d479 100644
--- a/j2cli/__init__.py
+++ b/j2cli/__init__.py
@@ -1,11 +1,14 @@
 #! /usr/bin/env python
 
 """ j2cli main file """
-import pkg_resources
+from importlib.metadata import version
 
 __author__ = "Mark Vartanyan"
 __email__ = "kolypto@gmail.com"
-__version__ = pkg_resources.get_distribution('j2cli').version
+try:
+    __version__ = version("j2cli")
+except Exception:
+    __version__ = "unknown"
 
 from j2cli.cli import main
 
