diff --git a/Makefile b/Makefile
index c0284bb..8d7d1d8 100644
--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,8 @@ objects = \
 	3rdparty/hmac_sha/hmac_sha2.o \
 	3rdparty/hmac_sha/sha2.o
 
-CHECKFLAGS = -Wall -Werror -Wuninitialized -Wundef
+GCC_INCLUDE_DIR := $(shell $(CC) -print-file-name=include)
+CHECKFLAGS = -I$(GCC_INCLUDE_DIR) -Wall -Wuninitialized -Wundef
 
 DEPFLAGS = -Wp,-MMD,$(@D)/.$(@F).d,-MT,$@
 
