--- a/whipper/test/test_common_accurip.py
+++ b/whipper/test/test_common_accurip.py
@@ -42,8 +42,8 @@
         self.assertEqual(responses[1].discId1, '0000f21c')
         self.assertEqual(responses[1].discId2, '00027ef8')
         self.assertEqual(responses[1].cddbDiscId, '05021002')
-        self.assertEqual(responses[1].confidences[0], 7)
-        self.assertEqual(responses[1].confidences[1], 7)
+        self.assertEqual(responses[1].confidences[0], 9)
+        self.assertEqual(responses[1].confidences[1], 9)
         self.assertEqual(responses[1].checksums[0], 'dc77f9ab')
         self.assertEqual(responses[1].checksums[1], 'dd97d2c3')
 
