------------------------------------------------------------------------
r74 | bdt-rob | 2024-06-20 19:36:36 +0300 (Thu, 20 Jun 2024) | 2 lines

Bug - gentoo #904366


Index: trunk/log.h
===================================================================
--- trunk/log.h	(revision 73)
+++ trunk/log.h	(revision 74)
@@ -15,6 +15,14 @@
 
 #include <fstream>
 
+// some of your ostream  -  bugs gentoo 904366
+#ifdef USE_CONSOLEKIT
+#include "Ck.h"
+#endif
+#ifdef USE_PAM
+#include "PAM.h"
+#endif
+
 using namespace std;
 
 class LogUnit

------------------------------------------------------------------------
