From 50f46fe2b446229417ef47807b0b849579b4f1e0 Mon Sep 17 00:00:00 2001
From: alex <alex@sinus.fritz.box>
Date: Sat, 5 Aug 2023 22:29:58 +0200
Subject: [PATCH] fixed bug #727 (can't open user configuration file)

---
 src/Config.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Config.cpp b/src/Config.cpp
index 2fe0d3c..9403da6 100644
--- a/src/Config.cpp
+++ b/src/Config.cpp
@@ -12,6 +12,7 @@
 #include <wxVillaLib/utils.h>
 #include <wx/fileconf.h>
 #include <wx/stdpaths.h>
+#include <wx/dir.h>
 
 Config s_config;
 
@@ -37,6 +38,10 @@ void Config::Init() {
 		wxMkdir(dataDir);
 		wxRenameFile(dataDir + ".tmp", dataDir + wxFILE_SEP_PATH + "dvdstyler");
 	}
+	if (!wxDir::Exists(dataDir)) {
+		wxMkdir(dataDir);
+	}
+
 	cfg = new wxFileConfig("", "", dataDir + wxFILE_SEP_PATH + "dvdstyler");
 	wxConfig::Set(cfg);
 #endif
-- 
2.55.0

