diff -ru openmw-openmw-0.49.0.orig/components/files/qtconversion.cpp openmw-openmw-0.49.0/components/files/qtconversion.cpp
--- openmw-openmw-0.49.0.orig/components/files/qtconversion.cpp	2025-10-15 22:45:05.525008496 +0200
+++ openmw-openmw-0.49.0/components/files/qtconversion.cpp	2025-10-17 22:33:42.669183665 +0200
@@ -18,11 +18,11 @@
 std::filesystem::path Files::pathFromQString(QStringView path)
 {
     const auto tmp = path.toUtf8();
-    return std::filesystem::path{ Misc::StringUtils::stringToU8String(tmp) };
+    return std::filesystem::path{ Misc::StringUtils::stringToU8String(tmp.data()) };
 }
 
 std::filesystem::path Files::pathFromQString(QString&& path)
 {
     const auto tmp = path.toUtf8();
-    return std::filesystem::path{ Misc::StringUtils::stringToU8String(tmp) };
+    return std::filesystem::path{ Misc::StringUtils::stringToU8String(tmp.data()) };
 }
diff -ru openmw-openmw-0.49.0.orig/components/vfs/qtconversion.cpp openmw-openmw-0.49.0/components/vfs/qtconversion.cpp
--- openmw-openmw-0.49.0.orig/components/vfs/qtconversion.cpp	2025-10-15 22:45:05.545008595 +0200
+++ openmw-openmw-0.49.0/components/vfs/qtconversion.cpp	2025-10-17 22:29:24.481846343 +0200
@@ -16,11 +16,11 @@
 VFS::Path::Normalized VFS::Path::normalizedFromQString(QStringView path)
 {
     const auto tmp = path.toUtf8();
-    return Normalized{ tmp };
+    return Normalized{ tmp.data() };
 }
 
 VFS::Path::Normalized VFS::Path::normalizedFromQString(QString&& path)
 {
     const auto tmp = path.toUtf8();
-    return Normalized{ tmp };
+    return Normalized{ tmp.data() };
 }
