diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 900fa39..b8a8c32 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,7 +2,6 @@ cmake_minimum_required(VERSION 3.1)
 
 #玲珑构建依赖目录
 include_directories(${CMAKE_INSTALL_PREFIX}/include)
-include_directories(${CMAKE_INSTALL_PREFIX}/include/deepin-ocr-plugin-manager)
 
 set(APP_BIN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/bin/)
 set(BIN_NAME ${CMAKE_PROJECT_NAME})
@@ -18,16 +17,17 @@ file(GLOB_RECURSE SRCS
     ./src/*.h
     ./src/*.cpp
     )
+list(FILTER SRCS EXCLUDE REGEX ".*/src/ocr/.*")
 
 file(GLOB_RECURSE QMLSRC
     ./*.qml
     )
+list(FILTER QMLSRC EXCLUDE REGEX ".*/qml/Live.*")
 
 # Find the Qt5Quick library
 find_package(Qt5Quick CONFIG REQUIRED)
 find_package(Qt5QuickCompiler REQUIRED)
 find_package(Qt5DBus CONFIG REQUIRED)
-find_package(Qt5Concurrent CONFIG REQUIRED)
 find_package(Qt5Svg CONFIG REQUIRED)
 find_package(Qt5PrintSupport CONFIG REQUIRED)
 find_package(Qt5LinguistTools CONFIG REQUIRED)
@@ -39,9 +39,6 @@ pkg_check_modules(3rd_lib REQUIRED
     dtkwidget
     )
 
-# OCR
-pkg_search_module(OCR_PLUGIN REQUIRED deepin-ocr-plugin-manager)
-include_directories(${OCR_PLUGIN_INCLUDE_DIRS})
 
 # 保证 src 目录下头文件全局可见
 include_directories(src)
@@ -72,10 +69,9 @@ add_executable(${BIN_NAME}
 target_include_directories(${BIN_NAME} PUBLIC ${3rd_lib_INCLUDE_DIRS})
 target_link_libraries(${BIN_NAME}
     ${DtkDeclarative_LIBRARIES}
-    Qt5::Quick Qt5::PrintSupport Qt5::Gui Qt5::Qml Qt5::Core Qt5::DBus Qt5::Concurrent Qt5::Svg
+    Qt5::Quick Qt5::PrintSupport Qt5::Gui Qt5::Qml Qt5::Core Qt5::DBus Qt5::Svg
     GL pthread
     ${3rd_lib_LIBRARIES}
-    ${OCR_PLUGIN_LIBRARIES}
     )
 
 if (${CMAKE_BUILD_TYPE} MATCHES "Debug")
diff --git a/src/deepin-image-viewer.qrc b/src/deepin-image-viewer.qrc
index bf63ea6..3fdd58d 100644
--- a/src/deepin-image-viewer.qrc
+++ b/src/deepin-image-viewer.qrc
@@ -27,10 +27,6 @@
         <file>qml/ThumbnailDelegate/NormalThumbnailDelegate.qml</file>
         <file>qml/ThumbnailDelegate/MultiThumnailDelegate.qml</file>
         <file>qml/ThumbnailDelegate/ThumbnailImage.qml</file>
-        <file>qml/LiveText/LiveBlock.qml</file>
-        <file>qml/LiveText/LiveBlockController.js</file>
-        <file>qml/LiveText/LiveBlockRubberBand.qml</file>
-        <file>qml/LiveText/LiveTextWidget.qml</file>
         <file>qml/Utils/FloatingNotice.qml</file>
         <file>qml/Utils/ImageInputHandler.qml</file>
         <file>qml/Utils/RightMenuItem.qml</file>
diff --git a/src/main.cpp b/src/main.cpp
index febd05e..dabe74d 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -4,7 +4,6 @@
 
 #include "src/filecontrol.h"
 #include "src/cursortool.h"
-#include "src/ocr/livetextanalyzer.h"
 #include "src/dbus/applicationadpator.h"
 #include "src/declarative/mousetrackitem.h"
 #include "src/globalcontrol.h"
@@ -103,11 +102,6 @@ int main(int argc, char *argv[])
         providerCache->removeImageCache(fileName);
     });
 
-    // OCR分析工具
-    auto liveTextAnalyzer = new LiveTextAnalyzer;
-    engine.rootContext()->setContextProperty("liveTextAnalyzer", liveTextAnalyzer);
-    engine.addImageProvider(QLatin1String("liveTextAnalyzer"), liveTextAnalyzer);
-
     // 判断命令行数据，在 QML 前优先加载
     QString cliParam = fileControl.parseCommandlineGetPath();
     if (!cliParam.isEmpty()) {
diff --git a/src/qml/ImageViewer.qml b/src/qml/ImageViewer.qml
index 66f9cd1..0a88a13 100644
--- a/src/qml/ImageViewer.qml
+++ b/src/qml/ImageViewer.qml
@@ -11,8 +11,6 @@ import org.deepin.dtk 1.0
 import org.deepin.image.viewer 1.0 as IV
 
 import "./ImageDelegate"
-import "./LiveText"
-
 Item {
     id: imageViewer
 
@@ -76,20 +74,12 @@ Item {
     }
 
     function recalculateLiveText() {
-        if (targetImageReady
-                && IV.Types.DynamicImage !== currentImageInfo.type) {
-            exitLiveText()
-            startLiveText()
-        }
     }
 
     function startLiveText() {
-        console.debug("function startLiveText()")
-        view.startLiveTextAnalyze()
     }
 
     function exitLiveText() {
-        view.exitLiveText()
     }
 
     function fitImage() {
@@ -114,7 +104,7 @@ Item {
 
     // 触发全屏展示图片
     function showPanelFullScreen() {
-        view.exitLiveText()
+        exitLiveText()
         GStatus.showImageInfo = false
 
         showFullScreen()
@@ -424,10 +414,6 @@ Item {
             GStatus.viewFlicking = false
         }
 
-        Component.onCompleted: {
-            liveTextAnalyzer.analyzeFinished.connect(runLiveText)
-        }
-
         BusyIndicator {
             anchors.centerIn: parent
             width: 48
@@ -443,81 +429,12 @@ Item {
             }
         }
 
-        // 实况文本背景遮罩
-        Rectangle {
-            id: liveTextBackground
-
-            anchors.centerIn: parent
-            color: "#000000" // live text高亮阴影
-            opacity: 0.5
-            visible: highlightTextButton.checked && highlightTextButton.visible
-
-            onVisibleChanged: {
-                var target = view.currentImage
-                if (null !== target) {
-                    width = target.paintedWidth * target.scale
-                    height = target.paintedHeight * target.scale
-                }
-            }
-        }
-
-        //live text分析函数
-        //缩放和切换需要重新执行此函数
-        function liveTextAnalyze() {
-            console.debug("Live Text analyze start")
-            viewBackground.grabToImage(function (result) {
-                //截取当前控件显示
-                liveTextAnalyzer.setImage(result.image) //设置分析图片
-                liveTextAnalyzer.analyze(currentIndex) //执行分析（异步执行，函数会立即返回）
-                // result.saveToFile("/home/user/Desktop/viewer.png") //保存截取的图片，debug用
-            })
-        }
-
-        //live text执行函数
-        function runLiveText(resultCanUse, token) {
-            console.debug("function runLiveText", token, currentIndex)
-            if (resultCanUse && token == currentIndex) {
-                //这里无视警告，就是需要js的==来进行自动类型转换
-                console.debug("run live start")
-                ltw.drawRect(liveTextAnalyzer.liveBlock())
-                ltw.visible = true
-            }
-        }
-
-        //live text退出函数
         function exitLiveText() {
-            console.debug("live exit")
-            liveTextAnalyzer.breakAnalyze()
-            ltw.clearLive()
-            liveTextTimer.stop()
         }
 
-        //live text分析启动控制
         function startLiveTextAnalyze() {
-            if (targetImageReady
-                    && IV.Types.DynamicImage !== currentImageInfo.type) {
-                liveTextTimer.restart()
-            }
         }
 
-        //live text分析启动延迟
-        Timer {
-            id: liveTextTimer
-
-            interval: 1500
-            running: false
-            repeat: false
-
-            onTriggered: {
-                if (fileControl.isCanSupportOcr(GControl.currentSource)
-                        && targetImageReady
-                        && IV.Types.DynamicImage !== currentImageInfo.type) {
-                    // 执行条件和OCR按钮使能条件一致
-                    view.liveTextAnalyze()
-                    running = false
-                }
-            }
-        }
     }
 
     IV.ImageInfo {
@@ -527,64 +444,6 @@ Item {
         source: GControl.currentSource
     }
 
-    Connections {
-        target: GStatus
-        onViewFlickingChanged: {
-            if (GStatus.viewFlicking) {
-                view.exitLiveText()
-            } else {
-                view.startLiveTextAnalyze()
-            }
-        }
-    }
-
-    LiveTextWidget {
-        //live text主控件
-        id: ltw
-
-        anchors.fill: imageViewerArea
-        parent: imageViewerArea
-    }
-
-    FloatingButton {
-        id: highlightTextButton
-
-        property bool isHighlight: false
-
-        checked: isHighlight
-        width: 50
-        height: 50
-        visible: false
-        parent: imageViewerArea
-        z: parent.z + 100
-        anchors {
-            right: parent.right
-            rightMargin: 100
-            bottom: parent.bottom
-            bottomMargin: thumbnailViewBackGround.height + 20
-        }
-
-        DciIcon {
-            width: 45
-            height: 45
-            Layout.alignment: Qt.AlignHCenter | Qt.AlignBottom
-            name: "icon_recognition_highlight"
-        }
-
-        onClicked: {
-            isHighlight = !isHighlight
-        }
-
-        // 高亮时不弹出工具栏栏以方便选取
-        onCheckedChanged: {
-            GStatus.animationBlock = checked
-        }
-        onVisibleChanged: {
-            if (!visible) {
-                GStatus.animationBlock = false
-            }
-        }
-    }
 
     //rename窗口
     ReName {
diff --git a/src/qml/ThumbnailListView.qml b/src/qml/ThumbnailListView.qml
index c267dc6..a571eac 100644
--- a/src/qml/ThumbnailListView.qml
+++ b/src/qml/ThumbnailListView.qml
@@ -356,34 +356,14 @@ Item {
 
     Row {
         id: rightRowLayout
+        visible: false
+        width: 0
+        height: 0
 
         anchors {
             right: deleteButton.left
             verticalCenter: parent.verticalCenter
         }
-        spacing: 10
-        leftPadding: 20
-        rightPadding: 20
-
-        IconButton {
-            id: ocrButton
-
-            width: 50
-            height: 50
-            enabled: fileControl.isCanSupportOcr(GControl.currentSource)
-                     && !imageIsNull
-            icon.name: "icon_character_recognition"
-            ToolTip.delay: 500
-            ToolTip.timeout: 5000
-            ToolTip.visible: hovered
-            ToolTip.text: qsTr("Extract text")
-
-            onClicked: {
-                GControl.submitImageChangeImmediately()
-                fileControl.ocrImage(GControl.currentSource,
-                                     GControl.currentFrameIndex)
-            }
-        }
     }
 
     IconButton {
diff --git a/src/qml/ViewRightMenu.qml b/src/qml/ViewRightMenu.qml
index 40003ee..7f716f5 100644
--- a/src/qml/ViewRightMenu.qml
+++ b/src/qml/ViewRightMenu.qml
@@ -20,7 +20,6 @@ Menu {
     property bool renamable: !isNullImage && fileControl.isCanRename(imageSource)
     property bool rotatable: !isNullImage && fileControl.isRotatable(imageSource)
     property bool deletable: !isNullImage && fileControl.isCanDelete(imageSource)
-    property bool supportOcr: !isNullImage && fileControl.isCanSupportOcr(imageSource)
     property bool supportWallpaper: !isNullImage && fileControl.isSupportSetWallpaper(imageSource)
 
     x: 250
@@ -68,25 +67,6 @@ Menu {
         }
     }
 
-    RightMenuItem {
-        text: qsTr("Extract text")
-        visible: supportOcr
-
-        onTriggered: {
-            GControl.submitImageChangeImmediately()
-            fileControl.ocrImage(imageSource, GControl.currentFrameIndex)
-        }
-
-        Shortcut {
-            sequence: "Alt+O"
-            enabled: supportOcr
-            onActivated: {
-                GControl.submitImageChangeImmediately()
-                fileControl.ocrImage(imageSource, GControl.currentFrameIndex)
-            }
-        }
-    }
-
     RightMenuItem {
         text: qsTr("Slide show")
 
diff --git a/src/src/filecontrol.cpp b/src/src/filecontrol.cpp
index d19700c..f742897 100644
--- a/src/src/filecontrol.cpp
+++ b/src/src/filecontrol.cpp
@@ -7,7 +7,6 @@
 #include "unionimage/unionimage_global.h"
 #include "unionimage/unionimage.h"
 #include "printdialog/printhelper.h"
-#include "ocr/ocrinterface.h"
 #include "imagedata/imageinfo.h"
 
 #include <DSysInfo>
@@ -89,7 +88,6 @@ QUrl UrlInfo(QString path)
 FileControl::FileControl(QObject *parent)
     : QObject(parent)
 {
-    m_ocrInterface = new OcrInterface("com.deepin.Ocr", "/com/deepin/Ocr", QDBusConnection::sessionBus(), this);
     m_shortcutViewProcess = new QProcess(this);
     m_config = LibConfigSetter::instance();
     imageFileWatcher = new ImageFileWatcher(this);
@@ -411,24 +409,6 @@ bool FileControl::isCanDelete(const QString &path)
     return bRet;
 }
 
-void FileControl::ocrImage(const QString &path, int index)
-{
-    QString localPath = QUrl(path).toLocalFile();
-    // 此处借用已取得的缓存信息，一般状态下，调用OCR前已完成图像的加载
-    ImageInfo info(path);
-
-    if (Types::MultiImage != info.type()) {  // 非多页图使用路径直接进行识别
-        m_ocrInterface->openFile(localPath);
-    } else {  // 多页图需要确定识别哪一页
-        QImageReader imageReader(localPath);
-        imageReader.jumpToImage(index);
-        auto image = imageReader.read();
-        auto tempFileName = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation) + QDir::separator() + "rec.png";
-        image.save(tempFileName);
-        m_ocrInterface->openFile(tempFileName);
-    }
-}
-
 QString FileControl::parseCommandlineGetPath()
 {
     QString filepath = "";
@@ -719,18 +699,6 @@ bool FileControl::isCheckOnly()
     return true;
 }
 
-bool FileControl::isCanSupportOcr(const QString &path)
-{
-    bool bRet = false;
-    QString localPath = QUrl(path).toLocalFile();
-    QFileInfo info(localPath);
-    imageViewerSpace::ImageType type = LibUnionImage_NameSpace::getImageType(localPath);
-    if (imageViewerSpace::ImageTypeDynamic != type && info.isReadable()) {
-        bRet = true;
-    }
-    return bRet;
-}
-
 bool FileControl::isCanRename(const QString &path)
 {
     bool bRet = false;
@@ -811,9 +779,6 @@ QString FileControl::createShortcutString()
     shortcut2.insert("name", tr("Exit fullscreen"));
     shortcut2.insert("value", "Esc");
 
-    QJsonObject shortcut3;
-    shortcut3.insert("name", tr("Extract text"));
-    shortcut3.insert("value", "Alt + O");
 
     QJsonObject shortcut4;
     shortcut4.insert("name", tr("Slide show"));
@@ -878,7 +843,6 @@ QString FileControl::createShortcutString()
     QJsonArray shortcutArray1;
     shortcutArray1.append(shortcut1);
     shortcutArray1.append(shortcut2);
-    shortcutArray1.append(shortcut3);
     shortcutArray1.append(shortcut4);
     shortcutArray1.append(shortcut5);
     shortcutArray1.append(shortcut6);
@@ -915,25 +879,9 @@ QString FileControl::createShortcutString()
     shortcut_group2.insert("groupName", tr("Settings"));
     shortcut_group2.insert("groupItems", shortcutArray2);
 
-    QJsonObject shortcut21;
-    shortcut21.insert("name", tr("Copy"));
-    shortcut21.insert("value", "Ctrl + C");
-
-    QJsonObject shortcut22;
-    shortcut22.insert("name", tr("Select all"));
-    shortcut22.insert("value", "Ctrl + A");
-
-    QJsonArray shortcutArray3;
-    shortcutArray3.append(shortcut21);
-    shortcutArray3.append(shortcut22);
-
-    QJsonObject shortcut_group3;
-    shortcut_group3.insert("groupName", tr("Live Text"));
-    shortcut_group3.insert("groupItems", shortcutArray3);
 
     QJsonArray shortcutArrayall;
     shortcutArrayall.append(shortcut_group1);
-    shortcutArrayall.append(shortcut_group3);
     shortcutArrayall.append(shortcut_group2);
 
     QJsonObject main_shortcut;
diff --git a/src/src/filecontrol.h b/src/src/filecontrol.h
index a032539..6d859df 100644
--- a/src/src/filecontrol.h
+++ b/src/src/filecontrol.h
@@ -17,7 +17,6 @@
 
 #include "imagedata/imagefilewatcher.h"
 
-class OcrInterface;
 class QProcess;
 
 class FileControl : public QObject
@@ -41,14 +40,12 @@ public:
     Q_INVOKABLE bool displayinFileManager(const QString &path);   // 在文件目录中显示
     Q_INVOKABLE void copyImage(const QString &path);              // 复制图片
     Q_INVOKABLE void copyText(const QString &str);                // 复制文字
-    Q_INVOKABLE void ocrImage(const QString &path, int index);    // 进行ocr识别
     Q_INVOKABLE void showPrintDialog(const QString &path);        // 调用打印接口
     Q_INVOKABLE QString parseCommandlineGetPath();                // 解析命令行
     Q_SLOT void rotateImageFile(const QString &path, int angle);  // 旋转图片文件
 
     Q_INVOKABLE bool isCheckOnly();                               // 判断当前进程是否为唯一看图实例
     Q_INVOKABLE bool isSupportSetWallpaper(const QString &path);  // 是否支持设置壁纸
-    Q_INVOKABLE bool isCanSupportOcr(const QString &path);
     Q_INVOKABLE bool isCanRename(const QString &path);
     Q_INVOKABLE bool isCanReadable(const QString &path);
     Q_INVOKABLE bool isRotatable(const QString &path);  // 是否可以被选旋转
@@ -87,8 +84,6 @@ private:
     QString createShortcutString();
 
 private:
-    OcrInterface *m_ocrInterface;
-
     QString m_shortcutString;  // 快捷键字符串，将采用懒加载模式，需要通过createShortcutString()函数使用
     QProcess *m_shortcutViewProcess;               // 快捷键面板进程
     QStringList listsupportWallPaper;              // 支持设置壁纸的图片后缀类型
