diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 377057fa..e3385951 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,7 +2,6 @@ cmake_minimum_required(VERSION 3.13)
 
 # 玲珑构建依赖目录
 include_directories(${CMAKE_INSTALL_PREFIX}/include)
-include_directories(${CMAKE_INSTALL_PREFIX}/include/deepin-ocr-plugin-manager)
 
 set(APP_BIN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/bin/)
 set(BIN_NAME ${CMAKE_PROJECT_NAME})
@@ -22,17 +21,14 @@ find_package(Dtk${DTK_VERSION_MAJOR} REQUIRED COMPONENTS
     Declarative
 )
 
-# OCR
-find_package(PkgConfig REQUIRED)
-pkg_search_module(OCR_PLUGIN REQUIRED deepin-ocr-plugin-manager)
-include_directories(${OCR_PLUGIN_INCLUDE_DIRS})
-
 # 保证 src 目录下头文件全局可见
 include_directories(src)
 
 # 源文件
 file(GLOB_RECURSE SRCS CONFIGURE_DEPENDS "./*.h" "./*.cpp")
+list(FILTER SRCS EXCLUDE REGEX ".*/src/ocr/.*")
 file(GLOB_RECURSE QMLSRC ./*.qml)
+list(FILTER QMLSRC EXCLUDE REGEX ".*/qml/Live.*")
 
 if(NOT BUILD_WITH_QT6)
     find_package(Qt5QuickCompiler CONFIG REQUIRED)
@@ -74,7 +70,6 @@ target_link_libraries(${BIN_NAME}
     Dtk${DTK_VERSION_MAJOR}::Declarative
     GL
     pthread
-    ${OCR_PLUGIN_LIBRARIES}
 )
 
 if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
diff --git a/src/deepin-image-viewer.qrc b/src/deepin-image-viewer.qrc
index f178fd4e..0faf0def 100644
--- a/src/deepin-image-viewer.qrc
+++ b/src/deepin-image-viewer.qrc
@@ -27,10 +27,6 @@
         <file>qml/ThumbnailDelegate/NormalThumbnailDelegate.qml</file>
         <file>qml/ThumbnailDelegate/MultiThumnailDelegate.qml</file>
         <file>qml/ThumbnailDelegate/ThumbnailImage.qml</file>
-        <file>qml/LiveText/LiveBlock.qml</file>
-        <file>qml/LiveText/LiveBlockController.js</file>
-        <file>qml/LiveText/LiveBlockRubberBand.qml</file>
-        <file>qml/LiveText/LiveTextWidget.qml</file>
         <file>qml/Utils/FloatingNotice.qml</file>
         <file>qml/Utils/ImageInputHandler.qml</file>
         <file>qml/Utils/RightMenuItem.qml</file>
diff --git a/src/main.cpp b/src/main.cpp
index ac5eee89..69b66274 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -4,7 +4,6 @@
 
 #include "src/filecontrol.h"
 #include "src/cursortool.h"
-#include "src/ocr/livetextanalyzer.h"
 #include "src/dbus/applicationadpator.h"
 #include "src/declarative/mousetrackitem.h"
 #include "src/declarative/pathviewrangehandler.h"
@@ -138,11 +137,6 @@ int main(int argc, char *argv[])
         providerCache->removeImageCache(fileName);
     });
 
-    // OCR分析工具
-    auto liveTextAnalyzer = new LiveTextAnalyzer;
-    engine.rootContext()->setContextProperty("liveTextAnalyzer", liveTextAnalyzer);
-    engine.addImageProvider(QLatin1String("liveTextAnalyzer"), liveTextAnalyzer);
-
     // 判断命令行数据，在 QML 前优先加载
     if (!cliParam.isEmpty()) {
         QStringList filePaths = fileControl.getDirImagePath(cliParam);
diff --git a/src/qml/ImageViewer.qml b/src/qml/ImageViewer.qml
index 43a1a93c..e320be9a 100644
--- a/src/qml/ImageViewer.qml
+++ b/src/qml/ImageViewer.qml
@@ -10,7 +10,6 @@ import QtQuick.Shapes
 import org.deepin.dtk 1.0
 import org.deepin.image.viewer 1.0 as IV
 import "./ImageDelegate"
-import "./LiveText"
 import "./InformationDialog"
 import "./Utils"
 
@@ -20,8 +19,6 @@ Item {
     // 记录图像缩放，用于在窗口缩放时，根据前后窗口变化保持图片缩放比例
     property bool enableChangeDisplay: true
     property real lastDisplayScaleWidth: 0
-    // current LiveText select text
-    property bool liveTextSelected: false
     // Image 类型的对象，空图片、错误图片、消失图片等异常为 null
     property alias targetImage: view.currentImage
 
@@ -519,154 +516,15 @@ Item {
         id: ltwLoader
 
         function recalculateLiveText() {
-            if (Loader.Ready === status && (targetImageReady && IV.Types.DynamicImage !== currentImageInfo.type)) {
-                item.exitLiveText();
-                item.startLiveTextAnalyze();
-            }
         }
 
         function startLiveText() {
-            if (Loader.Ready === status) {
-                item.startLiveTextAnalyze();
-            }
         }
 
         function stopLiveText() {
-            if (Loader.Ready === status) {
-                item.exitLiveText();
-            }
         }
 
         active: false
-        anchors.fill: parent
-        asynchronous: true
-
-        // 实况文本蒙版
-        sourceComponent: LiveTextWidget {
-            //live text主控件
-            id: ltw
-
-            //live text退出函数
-            function exitLiveText() {
-                liveTextAnalyzer.breakAnalyze();
-                ltw.clearLive();
-                liveTextTimer.stop();
-            }
-
-            //live text分析函数
-            //缩放和切换需要重新执行此函数
-            function liveTextAnalyze() {
-                viewBackground.grabToImage(function (result) {
-                    //截取当前控件显示
-                    liveTextAnalyzer.setImage(result.image); //设置分析图片
-                    liveTextAnalyzer.analyze(view.currentIndex); //执行分析（异步执行，函数会立即返回）
-                    // result.saveToFile("/home/user/Desktop/viewer.png") //保存截取的图片，debug用
-                });
-            }
-
-            //live text执行函数
-            function runLiveText(resultCanUse, token) {
-                if (resultCanUse && token == view.currentIndex) {
-                    //这里无视警告，就是需要js的==来进行自动类型转换
-                    console.debug("run live start");
-                    ltw.drawRect(liveTextAnalyzer.liveBlock());
-                    ltw.visible = true;
-                }
-            }
-
-            //live text分析启动控制
-            function startLiveTextAnalyze() {
-                if (targetImageReady && IV.Types.DynamicImage !== currentImageInfo.type) {
-                    liveTextTimer.restart();
-                }
-            }
-
-            anchors.fill: parent
-
-            Component.onCompleted: {
-                liveTextAnalyzer.analyzeFinished.connect(runLiveText);
-            }
-            onCurrentHasSelectChanged: {
-                imageViewer.liveTextSelected = currentHasSelect;
-            }
-
-            // 实况文本背景遮罩
-            Rectangle {
-                id: liveTextBackground
-
-                anchors.centerIn: parent
-                color: "#000000" // live text高亮阴影
-                opacity: 0.5
-                visible: highlightTextButton.checked && highlightTextButton.visible
-
-                onVisibleChanged: {
-                    var target = view.currentImage;
-                    if (null !== target) {
-                        width = target.paintedWidth * target.scale;
-                        height = target.paintedHeight * target.scale;
-                    }
-                }
-            }
-
-            //live text分析启动延迟
-            Timer {
-                id: liveTextTimer
-
-                interval: 1500
-                repeat: false
-                running: false
-
-                onTriggered: {
-                    var supportOcr = IV.FileControl.isCanSupportOcr(IV.GControl.currentSource);
-                    if (supportOcr && targetImageReady && IV.Types.DynamicImage !== currentImageInfo.type) {
-                        // 执行条件和OCR按钮使能条件一致
-                        ltw.liveTextAnalyze();
-                        running = false;
-                    }
-                }
-            }
-
-            FloatingButton {
-                id: highlightTextButton
-
-                property bool isHighlight: false
-
-                checked: isHighlight
-                height: 50
-                parent: imageViewerArea
-                visible: false
-                width: 50
-                z: ltw.z + 100
-
-                // 高亮时不弹出工具栏栏以方便选取
-                onCheckedChanged: {
-                    IV.GStatus.animationBlock = checked;
-                }
-                onClicked: {
-                    isHighlight = !isHighlight;
-                }
-                onVisibleChanged: {
-                    if (!visible) {
-                        IV.GStatus.animationBlock = false;
-                    }
-                }
-
-                anchors {
-                    bottom: parent.bottom
-                    bottomMargin: thumbnailViewBackGround.height + 20
-                    right: parent.right
-                    rightMargin: 100
-                }
-
-                DciIcon {
-                    anchors.centerIn: parent
-                    height: 45
-                    name: "icon_recognition_highlight"
-                    palette: DTK.makeIconPalette(parent.palette)
-                    width: 45
-                }
-            }
-        }
     }
 
     Loader {
@@ -703,9 +561,6 @@ Item {
         sourceComponent: ViewRightMenu {
             id: rightMenu
 
-            // 拷贝快捷键冲突：选中实况文本时，屏蔽拷贝图片的快捷键
-            copyableConfig: !imageViewer.liveTextSelected
-
             // 菜单销毁后也需要发送信号，否则可能未正常送达
             Component.onDestruction: {
                 IV.GStatus.showRightMenu = false;
diff --git a/src/qml/ThumbnailListView.qml b/src/qml/ThumbnailListView.qml
index 095ee864..b9150a90 100644
--- a/src/qml/ThumbnailListView.qml
+++ b/src/qml/ThumbnailListView.qml
@@ -501,32 +501,14 @@ Control {
     Row {
         id: rightRowLayout
 
-        leftPadding: 20
-        rightPadding: 20
-        spacing: 10
+        height: 0
+        visible: false
+        width: 0
 
         anchors {
             right: deleteButton.left
             verticalCenter: parent.verticalCenter
         }
-
-        IconButton {
-            id: ocrButton
-
-            ToolTip.delay: 500
-            ToolTip.text: qsTr("Extract text")
-            ToolTip.timeout: 5000
-            ToolTip.visible: hovered
-            enabled: IV.FileControl.isCanSupportOcr(IV.GControl.currentSource) && !imageIsNull
-            height: 50
-            icon.name: "icon_character_recognition"
-            width: 50
-
-            onClicked: {
-                IV.GControl.submitImageChangeImmediately();
-                IV.FileControl.ocrImage(IV.GControl.currentSource, IV.GControl.currentFrameIndex);
-            }
-        }
     }
 
     IconButton {
diff --git a/src/qml/ViewRightMenu.qml b/src/qml/ViewRightMenu.qml
index 577c0a39..41f74d38 100644
--- a/src/qml/ViewRightMenu.qml
+++ b/src/qml/ViewRightMenu.qml
@@ -21,7 +21,6 @@ Menu {
     property bool readable: !isNullImage && IV.FileControl.isCanReadable(imageSource)
     property bool renamable: !isNullImage && IV.FileControl.isCanRename(imageSource)
     property bool rotatable: !isNullImage && IV.FileControl.isRotatable(imageSource)
-    property bool supportOcr: !isNullImage && IV.FileControl.isCanSupportOcr(imageSource)
     property bool supportWallpaper: !isNullImage && IV.FileControl.isSupportSetWallpaper(imageSource)
 
     maxVisibleItems: 20
@@ -72,26 +71,6 @@ Menu {
         }
     }
 
-    RightMenuItem {
-        text: qsTr("Extract text")
-        visible: supportOcr
-
-        onTriggered: {
-            IV.GControl.submitImageChangeImmediately();
-            IV.FileControl.ocrImage(imageSource, IV.GControl.currentFrameIndex);
-        }
-
-        Shortcut {
-            enabled: supportOcr
-            sequence: "Alt+O"
-
-            onActivated: {
-                IV.GControl.submitImageChangeImmediately();
-                IV.FileControl.ocrImage(imageSource, IV.GControl.currentFrameIndex);
-            }
-        }
-    }
-
     RightMenuItem {
         text: qsTr("Slide show")
 
diff --git a/src/src/filecontrol.cpp b/src/src/filecontrol.cpp
index 4bacf1fb..60f07ed8 100644
--- a/src/src/filecontrol.cpp
+++ b/src/src/filecontrol.cpp
@@ -7,7 +7,6 @@
 #include "unionimage/unionimage_global.h"
 #include "unionimage/unionimage.h"
 #include "printdialog/printhelper.h"
-#include "ocr/ocrinterface.h"
 #include "imagedata/imageinfo.h"
 
 #include <DSysInfo>
@@ -84,7 +83,6 @@ QUrl UrlInfo(QString path)
 FileControl::FileControl(QObject *parent)
     : QObject(parent)
 {
-    m_ocrInterface = new OcrInterface("com.deepin.Ocr", "/com/deepin/Ocr", QDBusConnection::sessionBus(), this);
     m_shortcutViewProcess = new QProcess(this);
     m_config = LibConfigSetter::instance();
     imageFileWatcher = ImageFileWatcher::instance();
@@ -411,30 +409,6 @@ bool FileControl::isCanDelete(const QString &path)
     return bRet;
 }
 
-void FileControl::ocrImage(const QString &path, int index)
-{
-    QString localPath = QUrl(path).toLocalFile();
-    // 此处借用已取得的缓存信息，一般状态下，调用OCR前已完成图像的加载
-    ImageInfo info(path);
-
-    if (Types::MultiImage != info.type()) {  // 非多页图使用路径直接进行识别
-        m_ocrInterface->openFile(localPath);
-    } else {  // 多页图需要确定识别哪一页
-        QImageReader imageReader(localPath);
-        imageReader.jumpToImage(index);
-        auto image = imageReader.read();
-        auto tempDir = QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
-        QDir dir(tempDir);
-        if (!dir.exists()) {
-            dir.mkpath(".");
-        }
-        auto tempFileName = tempDir + QDir::separator() + "rec.png";
-
-        image.save(tempFileName);
-        m_ocrInterface->openFile(tempFileName);
-    }
-}
-
 QString FileControl::parseCommandlineGetPath()
 {
     QString filepath = "";
@@ -697,18 +671,6 @@ bool FileControl::isCheckOnly()
     return true;
 }
 
-bool FileControl::isCanSupportOcr(const QString &path)
-{
-    bool bRet = false;
-    QString localPath = QUrl(path).toLocalFile();
-    QFileInfo info(localPath);
-    imageViewerSpace::ImageType type = LibUnionImage_NameSpace::getImageType(localPath);
-    if (imageViewerSpace::ImageTypeDynamic != type && info.isReadable()) {
-        bRet = true;
-    }
-    return bRet;
-}
-
 bool FileControl::isCanRename(const QString &path)
 {
     bool bRet = false;
@@ -789,10 +751,6 @@ QString FileControl::createShortcutString()
     shortcut2.insert("name", tr("Exit fullscreen"));
     shortcut2.insert("value", "Esc");
 
-    QJsonObject shortcut3;
-    shortcut3.insert("name", tr("Extract text"));
-    shortcut3.insert("value", "Alt + O");
-
     QJsonObject shortcut4;
     shortcut4.insert("name", tr("Slide show"));
     shortcut4.insert("value", "F5");
@@ -856,7 +814,6 @@ QString FileControl::createShortcutString()
     QJsonArray shortcutArray1;
     shortcutArray1.append(shortcut1);
     shortcutArray1.append(shortcut2);
-    shortcutArray1.append(shortcut3);
     shortcutArray1.append(shortcut4);
     shortcutArray1.append(shortcut5);
     shortcutArray1.append(shortcut6);
@@ -893,25 +850,8 @@ QString FileControl::createShortcutString()
     shortcut_group2.insert("groupName", tr("Settings"));
     shortcut_group2.insert("groupItems", shortcutArray2);
 
-    QJsonObject shortcut21;
-    shortcut21.insert("name", tr("Copy"));
-    shortcut21.insert("value", "Ctrl + C");
-
-    QJsonObject shortcut22;
-    shortcut22.insert("name", tr("Select all"));
-    shortcut22.insert("value", "Ctrl + A");
-
-    QJsonArray shortcutArray3;
-    shortcutArray3.append(shortcut21);
-    shortcutArray3.append(shortcut22);
-
-    QJsonObject shortcut_group3;
-    shortcut_group3.insert("groupName", tr("Live Text"));
-    shortcut_group3.insert("groupItems", shortcutArray3);
-
     QJsonArray shortcutArrayall;
     shortcutArrayall.append(shortcut_group1);
-    shortcutArrayall.append(shortcut_group3);
     shortcutArrayall.append(shortcut_group2);
 
     QJsonObject main_shortcut;
diff --git a/src/src/filecontrol.h b/src/src/filecontrol.h
index 3edfec46..c001a90a 100644
--- a/src/src/filecontrol.h
+++ b/src/src/filecontrol.h
@@ -17,7 +17,6 @@
 
 #include "imagedata/imagefilewatcher.h"
 
-class OcrInterface;
 class QProcess;
 
 class FileControl : public QObject
@@ -43,13 +42,11 @@ public:
     Q_INVOKABLE bool displayinFileManager(const QString &path);  // 在文件目录中显示
     Q_INVOKABLE void copyImage(const QString &path);             // 复制图片
     Q_INVOKABLE void copyText(const QString &str);               // 复制文字
-    Q_INVOKABLE void ocrImage(const QString &path, int index);   // 进行ocr识别
     Q_INVOKABLE void showPrintDialog(const QString &path);       // 调用打印接口
     Q_INVOKABLE QString parseCommandlineGetPath();               // 解析命令行
 
     Q_INVOKABLE bool isCheckOnly();                               // 判断当前进程是否为唯一看图实例
     Q_INVOKABLE bool isSupportSetWallpaper(const QString &path);  // 是否支持设置壁纸
-    Q_INVOKABLE bool isCanSupportOcr(const QString &path);
     Q_INVOKABLE bool isCanRename(const QString &path);
     Q_INVOKABLE bool isCanReadable(const QString &path);
     Q_INVOKABLE bool isRotatable(const QString &path);  // 是否可以被选旋转
@@ -88,8 +85,6 @@ private:
     QString createShortcutString();
 
 private:
-    OcrInterface *m_ocrInterface;
-
     QString m_shortcutString;  // 快捷键字符串，将采用懒加载模式，需要通过createShortcutString()函数使用
     QProcess *m_shortcutViewProcess;               // 快捷键面板进程
     QStringList listsupportWallPaper;              // 支持设置壁纸的图片后缀类型
