--- a/Makefile.in
+++ b/Makefile.in
@@ -115,21 +115,16 @@ uninstall: ## Uninstall the Chrome version of curl-impersonate after 'make insta
 
 clean:
 	## Remove all build artifacts, including dependencies
-	rm -Rf brotli-$(BROTLI_VERSION).tar.gz brotli-$(BROTLI_VERSION)
+	rm -Rf brotli-$(BROTLI_VERSION)
 	rm -Rf boringssl-$(BORING_SSL_COMMIT).zip boringssl-$(BORING_SSL_COMMIT)
-	rm -Rf nghttp2-$(NGHTTP2_VERSION).tar.bz2 nghttp2-$(NGHTTP2_VERSION)
-	rm -Rf ngtcp2-$(NGTCP2_VERSION).tar.bz2 ngtcp2-$(NGTCP2_VERSION)
-	rm -Rf nghttp3-$(NGHTTP3_VERSION).tar.bz2 nghttp3-$(NGHTTP3_VERSION)
-	rm -Rf $(CURL_VERSION).tar.gz $(CURL_VERSION)
+	rm -Rf nghttp2-$(NGHTTP2_VERSION)
+	rm -Rf ngtcp2-$(NGTCP2_VERSION)
+	rm -Rf nghttp3-$(NGHTTP3_VERSION)
+	rm -Rf $(CURL_VERSION)
 .PHONY: clean
 
 
-brotli-$(BROTLI_VERSION).tar.gz:
-	curl -L "https://github.com/google/brotli/archive/refs/tags/v${BROTLI_VERSION}.tar.gz" \
-		-o "brotli-${BROTLI_VERSION}.tar.gz"
-
-$(brotli_static_libs): brotli-$(BROTLI_VERSION).tar.gz
-	tar xf brotli-$(BROTLI_VERSION).tar.gz
+$(brotli_static_libs):
 	cd brotli-$(BROTLI_VERSION)
 	mkdir -p out
 	cd out
@@ -161,19 +156,13 @@ $(brotli_static_libs): brotli-$(BROTLI_VERSION).tar.gz
 	@cmake@ --build . --config Release --target install --parallel $(SUBJOBS)
 
 
-boringssl-$(BORING_SSL_COMMIT).zip:
-	curl -L https://github.com/google/boringssl/archive/$(BORING_SSL_COMMIT).zip \
-		-o boringssl-$(BORING_SSL_COMMIT).zip
-
 # Patch boringssl and use a dummy '.patched' file to mark it patched
 boringssl-$(BORING_SSL_COMMIT)/.patched: $(srcdir)/patches/boringssl.patch
-	unzip -q -o boringssl-$(BORING_SSL_COMMIT).zip
-	# mv boringssl boringssl-$(BORING_SSL_COMMIT)
 	cd boringssl-$(BORING_SSL_COMMIT)/
 	for p in $^; do patch -p1 < $$p; done
 	touch .patched
 
-$(boringssl_static_libs): boringssl-$(BORING_SSL_COMMIT).zip boringssl-$(BORING_SSL_COMMIT)/.patched
+$(boringssl_static_libs): boringssl-$(BORING_SSL_COMMIT)/.patched
 	mkdir -p $(boringssl_install_dir)
 	cd $(boringssl_install_dir)
 
@@ -213,11 +202,7 @@ $(boringssl_static_libs): boringssl-$(BORING_SSL_COMMIT).zip boringssl-$(BORING_
 	cp crypto/libcrypto.a $(boringssl_dir)/lib
 
 
-nghttp2-$(NGHTTP2_VERSION).tar.bz2:
-	curl -L $(NGHTTP2_URL) -o nghttp2-$(NGHTTP2_VERSION).tar.bz2
-
-$(nghttp2_static_libs): nghttp2-$(NGHTTP2_VERSION).tar.bz2
-	tar -xf nghttp2-$(NGHTTP2_VERSION).tar.bz2
+$(nghttp2_static_libs):
 	cd nghttp2-$(NGHTTP2_VERSION)
 
 	# Set up the configure flags to nghttp2.
@@ -236,11 +221,7 @@ $(nghttp2_static_libs): nghttp2-$(NGHTTP2_VERSION).tar.bz2
 	$(MAKE) MAKEFLAGS=-j$(SUBJOBS)
 	$(MAKE) install MAKEFLAGS=
 
-ngtcp2-$(NGTCP2_VERSION).tar.bz2:
-	curl -L $(NGTCP2_URL) -o ngtcp2-$(NGTCP2_VERSION).tar.bz2
-
-$(ngtcp2_static_libs): ngtcp2-$(NGTCP2_VERSION).tar.bz2 $(boringssl_static_libs)
-	tar -xf ngtcp2-$(NGTCP2_VERSION).tar.bz2
+$(ngtcp2_static_libs): $(boringssl_static_libs)
 	cd ngtcp2-$(NGTCP2_VERSION)
 	autoreconf -i
 
@@ -265,11 +246,7 @@ $(ngtcp2_static_libs): ngtcp2-$(NGTCP2_VERSION).tar.bz2 $(boringssl_static_libs)
 	$(MAKE) MAKEFLAGS=-j$(SUBJOBS)
 	$(MAKE) install MAKEFLAGS=
 
-nghttp3-$(NGHTTP3_VERSION).tar.bz2:
-	curl -L $(NGHTTP3_URL) -o nghttp3-$(NGHTTP3_VERSION).tar.bz2
-
-$(nghttp3_static_libs): nghttp3-$(NGHTTP3_VERSION).tar.bz2
-	tar -xf nghttp3-$(NGHTTP3_VERSION).tar.bz2
+$(nghttp3_static_libs):
 	cd nghttp3-$(NGHTTP3_VERSION)
 
 	# Set up the configure flags to nghttp2.
@@ -288,15 +265,8 @@ $(nghttp3_static_libs): nghttp3-$(NGHTTP3_VERSION).tar.bz2
 	$(MAKE) MAKEFLAGS=-j$(SUBJOBS)
 	$(MAKE) install MAKEFLAGS=
 
-$(CURL_VERSION).tar.gz:
-	curl -L "https://github.com/curl/curl/archive/$(CURL_VERSION).tar.gz" \
-		-o "$(CURL_VERSION).tar.gz"
-
 # Apply the "Chorme version" patches and mark using a dummy file
 $(CURL_VERSION)/.patched: $(srcdir)/patches/curl.patch
-	rm -Rf $(CURL_VERSION)
-	tar -xf $(CURL_VERSION).tar.gz
-	mv curl-$(CURL_VERSION) $(CURL_VERSION)  # fix directory name
 	cd $(CURL_VERSION)
 	for p in $^; do patch -p1 < $$p; done
 	# Re-generate the configure script
@@ -304,7 +274,7 @@ $(CURL_VERSION)/.patched: $(srcdir)/patches/curl.patch
 	touch .patched
 
 # This is a small hack that flags that curl was patched and configured in the "chrome" version
-$(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION).tar.gz $(CURL_VERSION)/.patched
+$(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION)/.patched
 	cd $(CURL_VERSION)
 
 	# Set up the configure flags to curl.
