--- a/src/plugins/plugins.cmake
+++ b/src/plugins/plugins.cmake
@@ -18,7 +18,7 @@ target_sources(${LIB_NAME} PRIVATE
 )
 
 set(PRIVATE_HEADER_FILES ${CMAKE_CURRENT_SOURCE_DIR}/private/dplatforminterface_p.h)
-set(PRIVATE_INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_INCLUDEDIR}/dtk${PROJECT_VERSION_MAJOR}/DGui/Private")
+set(PRIVATE_INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_INCLUDEDIR}/dtk${DTK_VERSION_MAJOR}/DGui/Private")
 
 # XCB
 if(NOT DTK_DISABLE_XCB)
