diff --git a/core/Services/Calendar/Util/ICalComponent.vala b/core/Services/Calendar/Util/ICalComponent.vala
index 8d10dd08..474bfcd3 100644
--- a/core/Services/Calendar/Util/ICalComponent.vala
+++ b/core/Services/Calendar/Util/ICalComponent.vala
@@ -34,7 +34,7 @@ namespace Calendar.Util {
         if (dt_end.is_null_time ()) {
             if (!component.get_duration ().is_null_duration ()) {
                 // Given duration
-                dt_end = dt_start.add (component.get_duration ());
+                dt_end = ICal.Duration.extend (dt_start, component.get_duration ());
             } else if (dt_start.is_date ()) {
                 // Implicit duration for DATE-type: 1 day
                 dt_end = dt_start.clone ();
diff --git a/src/EventEdition/RepeatPanel.vala b/src/EventEdition/RepeatPanel.vala
index 116635e0..7caa4af5 100644
--- a/src/EventEdition/RepeatPanel.vala
+++ b/src/EventEdition/RepeatPanel.vala
@@ -272,12 +272,8 @@ public class Maya.View.EventEdition.RepeatPanel : Gtk.Grid {
 
     private void load_weekly_recurrence (ICal.Recurrence rrule) {
         repeat_combobox.active = 1;
-        var by_day = rrule.get_by_day_array ();
-        for (
-            uint i = 0;
-            i < by_day.length && (by_day.index (i) != ICal.RecurrenceArrayMaxValues.RECURRENCE_ARRAY_MAX);
-            i++
-        ) {
+        var by_day = rrule.get_by_array (ICal.RecurrenceByRule.BY_DAY);
+        for (uint i = 0; i < by_day.length; i++) {
             switch (ICal.Recurrence.day_day_of_week (by_day.index (i))) {
                 case ICal.RecurrenceWeekday.SUNDAY_WEEKDAY:
                     sun_button.active = true;
@@ -309,14 +305,12 @@ public class Maya.View.EventEdition.RepeatPanel : Gtk.Grid {
 
     private void load_monthly_recurrence (ICal.Recurrence rrule) {
         repeat_combobox.active = 2;
-        if (rrule.get_by_month_day (0) != ICal.RecurrenceArrayMaxValues.RECURRENCE_ARRAY_MAX) {
+        if (rrule.get_by_array_size (ICal.RecurrenceByRule.BY_MONTH_DAY) > 0) {
             same_radiobutton.active = true;
-        } else {
-            var by_day = rrule.get_by_day (0);
-            if (by_day != ICal.RecurrenceArrayMaxValues.RECURRENCE_ARRAY_MAX) {
-                set_every_day (by_day);
-                every_radiobutton.active = true;
-            }
+        }  else if (rrule.get_by_array_size (ICal.RecurrenceByRule.BY_DAY) > 0) {
+            var by_day = rrule.get_by (ICal.RecurrenceByRule.BY_DAY, 0);
+            set_every_day (by_day);
+            every_radiobutton.active = true;
         }
     }
 
@@ -682,7 +676,7 @@ public class Maya.View.EventEdition.RepeatPanel : Gtk.Grid {
                     array.append_val (day);
                 }
 
-                rrule.set_by_day_array (array);
+                rrule.set_by_array (ICal.RecurrenceByRule.BY_DAY, array);
                 break;
             case 2:
                 rrule.set_freq (ICal.RecurrenceFrequency.MONTHLY_RECURRENCE);
@@ -715,12 +709,12 @@ public class Maya.View.EventEdition.RepeatPanel : Gtk.Grid {
 
                     short day = encode_day (weekday, (int) Math.ceil ((double)parent_dialog.date_time.get_day_of_month () / (double)7));
                     array.append_val (day);
-                    rrule.set_by_day_array (array);
+                    rrule.set_by_array (ICal.RecurrenceByRule.BY_DAY, array);
                 } else {
                     var array = new GLib.Array<short>.sized (false, false, sizeof (short), 1);
                     var day_of_month = (short)parent_dialog.date_time.get_day_of_month ();
                     array.append_val (day_of_month);
-                    rrule.set_by_month_day_array (array);
+                    rrule.set_by_array (ICal.RecurrenceByRule.BY_MONTH_DAY, array);
                 }
                 break;
             case 3:

