--- aotriton-0.10b/CMakeLists.txt.bak	2025-08-19 11:21:19.571669252 +0200
+++ aotriton-0.10b/CMakeLists.txt	2025-08-19 11:30:47.895809078 +0200
@@ -65,10 +65,7 @@
 # Early failure for python binding
 if(NOT AOTRITON_NO_PYTHON)
   find_package(Python3 ${AOTRITON_MIN_PYTHON} COMPONENTS Development REQUIRED)
-  add_subdirectory(third_party/pybind11)
-  if(NOT EXISTS ${CMAKE_SOURCE_DIR}/third_party/pybind11/CMakeLists.txt)
-    message(FATAL_ERROR "third_party/pybind11 is not initialized\nRun `git submodule sync && git submodule update --init --recursive --force` to fix it.")
-  endif()
+  find_package(pybind11)
 endif()
 
 # Resolve name conflicts with suffix
@@ -102,34 +99,13 @@
 
 set(Python_ARTIFACTS_INTERACTIVE TRUE)
 
-# Not a target, we need to override Python3_EXECUTABLE later
-execute_process(COMMAND "${Python3_EXECUTABLE}" -m venv "${VENV_DIR}")
-
-set(ENV{VIRTUAL_ENV} "${VENV_DIR}")
+set(VENV_DIR "/usr")
+execute_process(COMMAND /usr/bin/python -c "import site; print(site.getsitepackages()[0], end='')" OUTPUT_VARIABLE VENV_SITE)
 message("VENV_DIR ${VENV_DIR}")
-# set(Python3_FIND_VIRTUALENV FIRST)
-# unset(Python3_EXECUTABLE)
-# find_package(Python3 COMPONENTS Interpreter REQUIRED)
-
-execute_process(COMMAND ${CMAKE_COMMAND} -E env VIRTUAL_ENV=${VENV_DIR} "${VENV_DIR}/bin/python" -c "import site; print(site.getsitepackages()[0], end='')" OUTPUT_VARIABLE VENV_SITE)
-# string(STRIP "${VENV_SITE}" VENV_SITE)
 message("VENV_SITE ${VENV_SITE}")
-
-execute_process(COMMAND ${CMAKE_COMMAND} -E env VIRTUAL_ENV=${VENV_DIR} "${VENV_DIR}/bin/python" -m pip install -r "${CMAKE_CURRENT_LIST_DIR}/requirements.txt")
-
 # AOTRITON_NOIMAGE_MODE does not need Triton
 if(NOT AOTRITON_NOIMAGE_MODE)
-  set(TRITON_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/triton_build")
-  execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory "${TRITON_BUILD_DIR}")
-  # set(AOTRITON_TRITON_EGGLINK "${VENV_SITE}/triton.egg-link")
-  # message("AOTRITON_TRITON_EGGLINK ${AOTRITON_TRITON_EGGLINK}")
   set(AOTRITON_TRITON_SO "${VENV_SITE}/triton/_C/libtriton.so")
-
-  add_custom_command(OUTPUT "${AOTRITON_TRITON_SO}"
-    COMMAND ${CMAKE_COMMAND} -E env TRITON_BUILD_PROTON=OFF VIRTUAL_ENV=${VENV_DIR} TRITON_BUILD_DIR=${TRITON_BUILD_DIR} "${VENV_DIR}/bin/python" -m pip install .
-    # COMMAND ${CMAKE_COMMAND} -E env VIRTUAL_ENV=${VENV_DIR} python -m pip show triton
-    WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/third_party/triton/"
-  )
   add_custom_target(aotriton_venv_triton ALL DEPENDS "${AOTRITON_TRITON_SO}")
 endif(NOT AOTRITON_NOIMAGE_MODE)
 
