From 8adcddad78e8f41ded5286b0430223d2d6f9cb5f Mon Sep 17 00:00:00 2001
From: Lubosz Sarnecki <lubosz@gmail.com>
Date: Tue, 18 Nov 2025 01:02:49 +0100
Subject: [PATCH 2/2] Don't create a virtualenv and build triton.

---
 CMakeLists.txt | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e33df5..28b30d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,11 +148,7 @@ endif()
 set(Python_ARTIFACTS_INTERACTIVE TRUE)
 
 # Not a target, we need to override Python3_EXECUTABLE later
-if(AOTRITON_INHERIT_SYSTEM_SITE_TRITON)
-  execute_process(COMMAND "${Python3_EXECUTABLE}" -m venv --system-site-packages "${VENV_DIR}")
-else()
-  execute_process(COMMAND "${Python3_EXECUTABLE}" -m venv "${VENV_DIR}")
-endif()
+set(VENV_DIR "/usr")
 
 set(ENV{VIRTUAL_ENV} "${VENV_DIR}")
 message("VENV_DIR ${VENV_DIR}")
@@ -168,12 +164,10 @@ set(VENV_BIN_PYTHON "${VENV_DIR}/${VENV_REL_PYTHON}")
 # unset(Python3_EXECUTABLE)
 # find_package(Python3 COMPONENTS Interpreter REQUIRED)
 
-execute_process(COMMAND ${CMAKE_COMMAND} -E env VIRTUAL_ENV=${VENV_DIR} "${VENV_BIN_PYTHON}" -c "import site; print(site.getsitepackages()[0], end='')" OUTPUT_VARIABLE VENV_SITE)
+execute_process(COMMAND /usr/bin/python -c "import site; print(site.getsitepackages()[0], end='')" OUTPUT_VARIABLE VENV_SITE)
 # string(STRIP "${VENV_SITE}" VENV_SITE)
 message("VENV_SITE ${VENV_SITE}")
 
-execute_process(COMMAND ${CMAKE_COMMAND} -E env VIRTUAL_ENV=${VENV_DIR} "${VENV_BIN_PYTHON}" -m pip install -r "${CMAKE_CURRENT_LIST_DIR}/requirements.txt")
-
 set(ALL_TRITON_SOS "")
 if(AOTRITON_NOIMAGE_MODE)
   message(STATUS "[AOTriton] Skipping triton due to AOTRITON_NOIMAGE_MODE")
@@ -194,8 +188,6 @@ elseif(AOTRITON_INHERIT_SYSTEM_SITE_TRITON)
     message(FATAL_ERROR "[AOTriton] AOTRITON_INHERIT_SYSTEM_SITE_TRITON is enabled but triton is not available in the virtual environment. Please install triton or disable this option. ${TRITON_OUTPUT}")
   endif()
 else()
-  set(TRITON_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/triton_build")
-  execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory "${TRITON_BUILD_DIR}")
   # set(AOTRITON_TRITON_EGGLINK "${VENV_SITE}/triton.egg-link")
   # message("AOTRITON_TRITON_EGGLINK ${AOTRITON_TRITON_EGGLINK}")
 
@@ -205,11 +197,6 @@ else()
     add_custom_command(OUTPUT "${AOTRITON_TRITON_SO}"
       COMMAND ${CMAKE_COMMAND} -E env VIRTUAL_ENV=${VENV_DIR} "${VENV_BIN_PYTHON}" -m pip install ${AOTRITON_USE_LOCAL_TRITON_WHEEL}
     )
-  else()
-    add_custom_command(OUTPUT "${AOTRITON_TRITON_SO}"
-      COMMAND ${CMAKE_COMMAND} -E env TRITON_BUILD_PROTON=OFF VIRTUAL_ENV=${VENV_DIR} TRITON_BUILD_DIR=${TRITON_BUILD_DIR} "${VENV_BIN_PYTHON}" -m pip install .
-      WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/third_party/triton/"
-    )
   endif()
   LIST(APPEND ALL_TRITON_SOS "${AOTRITON_TRITON_SO}")
 endif()
-- 
2.51.2

