From 978d68f9e898e57e5cda961f026d09b8831b98b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 15 Apr 2025 09:22:41 +0200
Subject: [PATCH] Port to Gaphas 5

---
 gaphor/ui/modelmerge/editor.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/gaphor/ui/modelmerge/editor.py b/gaphor/ui/modelmerge/editor.py
index c38c08c0b..ae7d63e48 100644
--- a/gaphor/ui/modelmerge/editor.py
+++ b/gaphor/ui/modelmerge/editor.py
@@ -1,6 +1,5 @@
 from __future__ import annotations
 
-from gaphas.decorators import nonrecursive
 from gi.repository import Gio, Gtk
 
 from gaphor.core import event_handler
@@ -65,7 +64,6 @@ class ModelMerge:
         self.event_manager.unsubscribe(self.on_model_loaded)
         self.event_manager.unsubscribe(self.on_model_updated)
 
-    @nonrecursive
     def apply(self, change_node: Node | None):
         def do_apply(node):
             for element in node.elements:
-- 
2.49.0

