From 62a2268e4099cd694d509966d83b2c06db730a2c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 20 Aug 2026 11:03:28 +0200
Subject: [PATCH] Fix incompatible pointer type

---
 src/gui.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui.c b/src/gui.c
index 8e34d076..1b4924ed 100644
--- a/src/gui.c
+++ b/src/gui.c
@@ -7244,7 +7244,7 @@ void player_media_state_changed_callback(GtkButton * button, GmtkMediaPlayerMedi
             dbus_disable_screensaver();
         gmtk_media_tracker_set_text(GMTK_MEDIA_TRACKER(tracker), _("Playing"));
         gmtk_media_player_set_attribute_boolean(GMTK_MEDIA_PLAYER(media), ATTRIBUTE_SUB_VISIBLE,
-                                                gtk_check_menu_item_get_active(menuitem_view_subtitles));
+                                                gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(menuitem_view_subtitles)));
         dbus_send_event("MediaPlaying", 0);
         g_idle_add(set_media_label, idledata);
         if (gmtk_media_player_get_attribute_string(GMTK_MEDIA_PLAYER(media), ATTRIBUTE_TITLE) != NULL) {
-- 
2.55.0

