diff --git a/platformthemeplugin/qdeepinfiledialoghelper.cpp b/platformthemeplugin/qdeepinfiledialoghelper.cpp
index be51637..ec50c0c 100644
--- a/platformthemeplugin/qdeepinfiledialoghelper.cpp
+++ b/platformthemeplugin/qdeepinfiledialoghelper.cpp
@@ -26,6 +26,7 @@
 #include <private/qguiapplication_p.h>
 
 #include <X11/Xlib.h>
+#include <unistd.h>
 
 #include <DPlatformHandle>
 DGUI_USE_NAMESPACE
