diff --git a/CMakeLists.txt b/CMakeLists.txt
index 91cc9da..2b2eac2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,6 +21,7 @@ option(ENABLE_COV "Enable code coverage" OFF)
 if(PROJECT_VERSION_MAJOR EQUAL 6)
   set(VERSION_SUFFIX 6)
   find_package(QT NAMES Qt6 REQUIRED COMPONENTS Core)
+  find_package(Qt6 COMPONENTS CorePrivate GuiPrivate WidgetsPrivate REQUIRED)
 else()
   set(VERSION_SUFFIX)
   find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core)
diff --git a/platformthemeplugin/qdeepintheme.h b/platformthemeplugin/qdeepintheme.h
index 9c6fcc4..34b6553 100644
--- a/platformthemeplugin/qdeepintheme.h
+++ b/platformthemeplugin/qdeepintheme.h
@@ -7,7 +7,11 @@
 
 #include <QMimeDatabase>
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 10, 0)
+#include <private/qgenericunixtheme_p.h>
+#else
 #include <private/qgenericunixthemes_p.h>
+#endif
 #include <qpa/qplatformwindow.h>
 #include <qpa/qplatformnativeinterface.h>
 
