diff -aur flask-principal-0.4.0.new/docs/conf.py flask-principal-0.4.0/docs/conf.py
--- flask-principal-0.4.0.new/docs/conf.py	2024-11-19 14:24:30.633855412 +0100
+++ flask-principal-0.4.0/docs/conf.py	2024-11-19 14:24:58.907373548 +0100
@@ -290,7 +290,7 @@
 
 
 # Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {'http://docs.python.org/': None}
+intersphinx_mapping = {'python': ('https://docs.python.org/3', None)}
 
 pygments_style = 'flask_theme_support.FlaskyStyle'
 
