--- a/bin/brat
+++ b/bin/brat
@@ -26,7 +26,7 @@ OLDENV="$(export -p)"
 SELF="$(realpath "$0")"
 NAME="${0##*/}"
 PROGRAM_NAME="${SELF##*/}"
-ROOT="${SELF%/*/*}"
+ROOT="/usr"
 
 
 # Set up the environment
@@ -35,7 +35,7 @@ LIBEXEC="$ROOT/libexec"
 SHARE="$ROOT/share/$PROGRAM_NAME"
 VENDOR="$SHARE/vendor"
 
-PATH="$LIB:$LIBEXEC:$VENDOR:$PATH"
+PATH="$LIB:$VENDOR:$PATH"
 export SELF
 
 USAGE="<COMMAND>"
