--- a/mididings/extra/osc.py
+++ b/mididings/extra/osc.py
@@ -18,7 +18,7 @@ import mididings.misc as _misc
 
 import mididings.extra.panic as _panic
 
-import liblo as _liblo
+import pyliblo3 as _liblo
 
 
 class OSCInterface(object):
--- a/mididings/live/osc_control.py
+++ b/mididings/live/osc_control.py
@@ -10,7 +10,7 @@
 # (at your option) any later version.
 #
 
-import liblo
+import pyliblo3 as liblo
 
 
 class LiveOSC(liblo.ServerThread):
