diff --git a/iotas/default_note_exporter.py b/iotas/default_note_exporter.py
index 563bf82..ce9a239 100644
--- a/iotas/default_note_exporter.py
+++ b/iotas/default_note_exporter.py
@@ -10,7 +10,10 @@ import shutil
 from threading import Thread
 from typing import Optional
 
-import pypandoc
+try:
+    import pypandoc
+except ModuleNotFoundError:
+    pypandoc = None
 
 from iotas.attachment_helpers import (
     AttachmentsCopyOutcome,
@@ -123,20 +126,27 @@ class DefaultNoteExporter(NoteExporter):
         working_dir = os.path.abspath(os.path.join(get_attachments_dir(), os.pardir))
 
         def thread_do():
-            try:
-                pypandoc.convert_text(
-                    content,
-                    out_format,
-                    format="gfm+attributes+implicit_figures",
-                    outputfile=self.__location,
-                    cworkdir=working_dir,
-                )
-            except (RuntimeError, OSError) as e:
-                logging.warning(f"Failed to export {out_format} to {self.__location}: {e}")
-                GLib.idle_add(self.emit, "failed", str(e))
+            if pypandoc:
+                try:
+                    pypandoc.convert_text(
+                        content,
+                        out_format,
+                        format="gfm+attributes+implicit_figures",
+                        outputfile=self.__location,
+                        cworkdir=working_dir,
+                    )
+                except (RuntimeError, OSError) as e:
+                    logging.warning(f"Failed to export {out_format} to {self.__location}: {e}")
+                    GLib.idle_add(self.emit, "failed", str(e))
+                else:
+                    logging.info(f"Exported {self.__out_format} to {self.__location}")
+                    GLib.idle_add(self.emit, "finished", self.__location)
             else:
-                logging.info(f"Exported {self.__out_format} to {self.__location}")
-                GLib.idle_add(self.emit, "finished", self.__location)
+                logging.warning(
+                    f"Failed to export {out_format} to {self.__location}: "
+                    "Pypandoc is not installed on your system."
+                )
+                self.emit("failed", "Pypandoc is not installed on your system.")
 
         thread = Thread(target=thread_do)
         thread.daemon = True
