From: Arnaud Ferraris <aferraris@debian.org>
Date: Thu, 19 Sep 2024 20:10:23 +0200
Subject: config_manager: use built-in StrEnum

Forwarded: not-needed
---
 iotas/config_manager.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iotas/config_manager.py b/iotas/config_manager.py
index 3bed0d8..3b84ba0 100644
--- a/iotas/config_manager.py
+++ b/iotas/config_manager.py
@@ -4,7 +4,7 @@ import logging
 from typing import Optional
 
 # Change to Python interval version from Python v3.11
-from strenum import StrEnum
+from enum import StrEnum
 
 from iotas import const
 
