From d539a57067cde3589755fdd406581f080c18a7ae Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 26 Aug 2026 11:20:57 +0200
Subject: [PATCH] Install symbolic icon

---
 pyproject.toml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pyproject.toml b/pyproject.toml
index d2a0e92..49b18e3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -72,6 +72,7 @@ version = {attr = "keysign._version.__version__"}
 "share/applications" = ["data/org.gnome.Keysign.desktop"]
 "share/metainfo" = ["data/org.gnome.Keysign.appdata.xml"]
 "share/icons/hicolor/scalable/apps" = ["data/org.gnome.Keysign.svg"]
+"share/icons/hicolor/symbolic/apps" = ["data/org.gnome.Keysign-symbolic.svg"]
 
 [tool.pytest.ini_options]
 pythonpath = ["."]
