diff --git a/init/postfix-tlspol.service.template b/init/postfix-tlspol.service.template
index 0ca03f2..4ad542f 100644
--- a/init/postfix-tlspol.service.template
+++ b/init/postfix-tlspol.service.template
@@ -1,22 +1,15 @@
 [Unit]
-Description=postfix-tlspol
+Description=postfix-tlspol daemon
 After=nss-lookup.target network-online.target
 Wants=nss-lookup.target
 
 [Service]
-WorkingDirectory=%%BASEDIR%%
-ExecStart=%%BASEDIR%%/build/postfix-tlspol -config %%BASEDIR%%/configs/config.yaml
+ExecStart=/usr/bin/postfix-tlspol -config /etc/postfix-tlspol/config.yaml
 ExecReload=/bin/kill -HUP $MAINPID
-DynamicUser=yes
 Restart=always
 RestartSec=5
-MemoryDenyWriteExecute=yes
-NoNewPrivileges=yes
-LockPersonality=yes
-PrivateUsers=yes
 PrivateDevices=yes
 PrivateTmp=yes
-ProtectHome=yes
 ProtectControlGroups=yes
 ProtectKernelLogs=yes
 ProtectKernelModules=yes
@@ -24,8 +17,6 @@ ProtectKernelTunables=yes
 ProtectClock=yes
 ProtectSystem=strict
 ProtectHostname=yes
-ProtectProc=invisible
-ProcSubset=pid
 RestrictRealtime=yes
 RestrictNamespaces=yes
 RestrictSUIDSGID=yes
@@ -35,12 +26,6 @@ SystemCallArchitectures=native
 SystemCallFilter=@system-service
 SystemCallFilter=~@privileged @resources
 SystemCallErrorNumber=EPERM
-CapabilityBoundingSet=
-ReadOnlyPaths=%%BASEDIR%%
-UMask=0111
-RuntimeDirectory=postfix-tlspol
-RuntimeDirectoryMode=1777
-BindPaths=/run/postfix-tlspol
 
 [Install]
 WantedBy=multi-user.target
