diff --git a/src/sage/libs/eclib/__init__.pxd b/src/sage/libs/eclib/__init__.pxd
index d44d4fba865..a40c604159b 100644
--- a/src/sage/libs/eclib/__init__.pxd
+++ b/src/sage/libs/eclib/__init__.pxd
@@ -55,7 +55,7 @@ cdef extern from "eclib/matrix.h":
     cdef cppclass mat:
         mat()
         mat(mat m)
-        scalar* get_entries()
+        vec get_entries()
         scalar sub(long, long)
         long nrows()
         long ncols()
@@ -67,7 +67,7 @@ cdef extern from "eclib/smatrix.h":
     cdef cppclass smat:
         smat()
         smat(smat m)
-        scalar* get_entries()
+        svec get_entries()
         scalar sub(long, long)
         long nrows()
         long ncols()
diff --git a/src/sage/libs/eclib/mat.pyx b/src/sage/libs/eclib/mat.pyx
index a5ef4f45c68..2b395819e7a 100644
--- a/src/sage/libs/eclib/mat.pyx
+++ b/src/sage/libs/eclib/mat.pyx
@@ -2,7 +2,7 @@
 Cremona matrices
 """
 
-from sage.libs.eclib cimport scalar, addscalar
+from sage.libs.eclib cimport vec, scalar, addscalar
 
 from sage.matrix.matrix_space import MatrixSpace
 from sage.rings.integer_ring import ZZ
@@ -213,7 +213,7 @@ cdef class Matrix:
         """
         cdef long n = self.nrows()
         cdef long i, j, k
-        cdef scalar* v = <scalar*> self.M.get_entries()   # coercion needed to deal with const
+        cdef vec v = <vec> self.M.get_entries()   # coercion needed to deal with const
 
         cdef Matrix_integer_dense Td
         cdef Matrix_integer_sparse Ts
