diff --git a/src/sage/graphs/generators/trees.pyx b/src/sage/graphs/generators/trees.pyx
index 2447e4d8595..5a8f2530213 100644
--- a/src/sage/graphs/generators/trees.pyx
+++ b/src/sage/graphs/generators/trees.pyx
@@ -480,7 +480,7 @@ def RandomLobster(n, p, q, seed=None, immutable=False):
     if seed is None:
         seed = int(current_randstate().long_seed() % sys.maxsize)
     import networkx
-    return Graph(networkx.random_lobster(n, p, q, seed=seed),
+    return Graph(networkx.random_lobster_graph(n, p, q, seed=seed),
                  immutable=immutable)
 
 

