diff --git a/src/libpcp/src/check-statics b/src/libpcp/src/check-statics
index b43cb4a..ddbc1f1 100755
--- a/src/libpcp/src/check-statics
+++ b/src/libpcp/src/check-statics
@@ -728,6 +728,7 @@ do
 		    -e 's/b \([_a-zA-Z][_a-zA-Z0-9]*\)\.[0-9]*.[0b]$/b \1/' \
 		    -e 's/\([bds] \).*\.\([_a-zA-Z]\)/\1\2/' \
 		    -e 's/s _glib_relative_date\./s /' \
+		    -e '/ B __gnu_lto_slim/d' \
 		    -e '/ s EH_/d' \
 		    -e '/ s str\.[0-9][0-9]*$/d' \
 		    -e '/ b \.bss/d' \
diff --git a/src/libpcp3/src/check-statics b/src/libpcp3/src/check-statics
index fdaa066..42005a4 100755
--- a/src/libpcp3/src/check-statics
+++ b/src/libpcp3/src/check-statics
@@ -727,6 +727,7 @@ do
 		    -e 's/b \([_a-zA-Z][_a-zA-Z0-9]*\)\.[0-9]*.[0b]$/b \1/' \
 		    -e 's/\([bds] \).*\.\([_a-zA-Z]\)/\1\2/' \
 		    -e 's/s _glib_relative_date\./s /' \
+		    -e '/ B __gnu_lto_slim/d' \
 		    -e '/ s EH_/d' \
 		    -e '/ s str\.[0-9][0-9]*$/d' \
 		    -e '/ b \.bss/d' \
