diff --git a/qa/openmetrics/openmetrics_endpoint.python b/qa/openmetrics/openmetrics_endpoint.python
index 354c16e..5e70aa1 100755
--- a/qa/openmetrics/openmetrics_endpoint.python
+++ b/qa/openmetrics/openmetrics_endpoint.python
@@ -16,10 +16,10 @@
 
 import time
 import re
-from six.moves import queue
+import queue
 import threading
 import socket
-from six.moves import BaseHTTPServer
+import http.server
 import argparse
 
 
@@ -33,7 +33,7 @@ def write_endpoint_metadata(args=None, endpoint=None):
     f.close()
 
 
-class FakeEndpoint(BaseHTTPServer.BaseHTTPRequestHandler):
+class FakeEndpoint(http.server.BaseHTTPRequestHandler):
     def sample_gauge_data(self, iteration=None, instances=None, metrics=None, endpointNum=None):
         gauge_value = 12.1
         instance_scale = 1
@@ -176,7 +176,7 @@ class PrometheusEndpoint(threading.Thread):
             pass
 
         finally:
-            httpd = BaseHTTPServer.HTTPServer(self.args.addr, FakeEndpoint, False)
+            httpd = http.server.HTTPServer(self.args.addr, FakeEndpoint, False)
             httpd.socket = self.args.sock
             httpd.pqueue = self.pqueue
             httpd.delay = self.args.delay
diff --git a/qa/opentelemetry/opentelemetry_endpoint.python b/qa/opentelemetry/opentelemetry_endpoint.python
index 96e59c3..bc3420a 100644
--- a/qa/opentelemetry/opentelemetry_endpoint.python
+++ b/qa/opentelemetry/opentelemetry_endpoint.python
@@ -16,10 +16,10 @@
 
 import time
 import re
-from six.moves import queue
+import queue
 import threading
 import socket
-from six.moves import BaseHTTPServer
+import http.server
 import argparse
 import json
 from copy import deepcopy
@@ -33,7 +33,7 @@ def write_endpoint_metadata(args=None, endpoint=None):
     f.write("{}\n".format(endpointStr))
     f.close()
 
-class FakeEndpoint(BaseHTTPServer.BaseHTTPRequestHandler):
+class FakeEndpoint(http.server.BaseHTTPRequestHandler):
     def sample_gauge_data(self, iteration=None, instances=None, metrics=None, endpointNum=None):
         gauge_value = 12.1
         instance_scale = 1
@@ -215,7 +215,7 @@ class PrometheusEndpoint(threading.Thread):
             pass
 
         finally:
-            httpd = BaseHTTPServer.HTTPServer(self.args.addr, FakeEndpoint, False)
+            httpd = http.server.HTTPServer(self.args.addr, FakeEndpoint, False)
             httpd.socket = self.args.sock
             httpd.pqueue = self.pqueue
             httpd.delay = self.args.delay
