diff --git i/src/bun.js/modules/BunJSCModule.h w/src/bun.js/modules/BunJSCModule.h
index 0b30cea5ac..443d5442c1 100644
--- i/src/bun.js/modules/BunJSCModule.h
+++ w/src/bun.js/modules/BunJSCModule.h
@@ -79,11 +79,14 @@ JSC_DEFINE_HOST_FUNCTION(functionStartRemoteDebugger,
     JSC::JSValue hostValue = callFrame->argument(0);
     JSC::JSValue portValue = callFrame->argument(1);
     const char* host = defaultHost;
+    CString hostCString;
     if (hostValue.isString()) {
 
         auto str = hostValue.toWTFString(globalObject);
-        if (!str.isEmpty())
-            host = toCString(str).span().data();
+        if (!str.isEmpty()) {
+            hostCString = toCString(str);
+            host = hostCString.span().data();
+        }
     } else if (!hostValue.isUndefined()) {
         throwVMError(globalObject, scope,
             createTypeError(globalObject, "host must be a string"_s));
