From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ralf Jung <post@ralfj.de>
Date: Wed, 19 Sep 2012 13:12:58 +0200
Subject: [PATCH] Allow one to set slave installation path during compilation

Forwarded: not-needed
---
 cmake/ossp-util.cmake |  4 +++-
 osspd.c               | 12 ++----------
 2 files changed, 5 insertions(+), 11 deletions(-)

diff --git a/cmake/ossp-util.cmake b/cmake/ossp-util.cmake
index 78bcf21388d5..fc0d561c9afd 100644
--- a/cmake/ossp-util.cmake
+++ b/cmake/ossp-util.cmake
@@ -36,9 +36,11 @@ macro(install_daemon TARGET)
 	)
 endmacro()
 
+option(SLAVESDIR "Slave installation path" "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBEXECDIR}/osspd")
+add_definitions("-DSLAVE_DEFAULT_PATH=\"${SLAVESDIR}\"")
 macro(install_slave TARGET)
 	install(TARGETS ${TARGET}
-		RUNTIME DESTINATION "${CMAKE_INSTALL_LIBEXECDIR}/ossp"
+		RUNTIME DESTINATION "${SLAVESDIR}"
 	)
 endmacro()
 
diff --git a/osspd.c b/osspd.c
index 7b95bf87fdd0..f5cd5cbee07c 100644
--- a/osspd.c
+++ b/osspd.c
@@ -2163,7 +2163,7 @@ static const char *usage =
 "    --max=MAX         maximum number of open streams (default 256)\n"
 "    --umax=MAX        maximum number of open streams per UID (default --max)\n"
 "    --exit-on-idle    exit if idle\n"
-"    --dsp-slave=PATH  DSP slave (default ossp-padsp in the same dir)\n"
+"    --dsp-slave=PATH  DSP slave (default: " SLAVE_DEFAULT_PATH "/ossp-padsp)\n"
 "    --log=LEVEL       log level (0..6)\n"
 "    --timestamp       timestamp log messages\n"
 "    -v                increase verbosity, can be specified multiple times\n"
@@ -2286,7 +2286,6 @@ int main(int argc, char **argv)
 		.max_streams = DFL_MAX_STREAMS,
 	};
 	struct fuse_args args = FUSE_ARGS_INIT(argc, argv);
-	char path_buf[PATH_MAX], *dir;
 	char adsp_buf[64] = "", mixer_buf[64] = "";
 	struct sigaction sa;
 	struct stat stat_buf;
@@ -2329,19 +2328,12 @@ int main(int argc, char **argv)
 	if (sigaction(SIGPIPE, &sa, NULL))
 		fatal_e(-errno, "failed to ignore SIGPIPE");
 
-	/* determine slave path and check for availability */
-	ret = readlink("/proc/self/exe", path_buf, PATH_MAX - 1);
-	if (ret < 0)
-		fatal_e(-errno, "failed to determine executable path");
-	path_buf[ret] = '\0';
-	dir = dirname(path_buf);
-
 	if (param.dsp_slave_path) {
 		strncpy(dsp_slave_path, param.dsp_slave_path, PATH_MAX - 1);
 		dsp_slave_path[PATH_MAX - 1] = '\0';
 	} else {
 		ret = snprintf(dsp_slave_path, PATH_MAX, "%s/%s",
-			       dir, "ossp-padsp");
+			       SLAVE_DEFAULT_PATH, "ossp-padsp");
 		if (ret >= PATH_MAX)
 			fatal("dsp slave pathname too long");
 	}
