diff --git i/awscli/autoprompt/prompttoolkit.py w/awscli/autoprompt/prompttoolkit.py
index 8f5f91512..21650f0e1 100644
--- i/awscli/autoprompt/prompttoolkit.py
+++ w/awscli/autoprompt/prompttoolkit.py
@@ -15,6 +15,19 @@ import shlex
 import sys
 from contextlib import contextmanager, nullcontext
 
+import importlib.metadata
+
+# Patch metadata.version to prevent crash due to missing dist-info
+real_metadata_version = importlib.metadata.version
+
+def safe_version(name):
+    if name == "prompt_toolkit":
+        return "0.0.0"  # dummy version
+    return real_metadata_version(name)
+
+importlib.metadata.version = safe_version
+
+
 from prompt_toolkit.application import Application
 from prompt_toolkit.completion import Completer, Completion, ThreadedCompleter
 from prompt_toolkit.document import Document
