diff --unified --recursive --text --new-file OvenMediaEngine-0.18.0.orig/src/projects/base/ovlibrary/string.h OvenMediaEngine-0.18.0/src/projects/base/ovlibrary/string.h
--- OvenMediaEngine-0.18.0.orig/src/projects/base/ovlibrary/string.h	2025-05-31 09:19:25.690715430 +0200
+++ OvenMediaEngine-0.18.0/src/projects/base/ovlibrary/string.h	2025-05-31 09:20:05.305013966 +0200
@@ -11,6 +11,7 @@
 #include <memory.h>
 
 #include <cstdarg>
+#include <cstdint>
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
