diff --git a/exec/src/lib.rs b/exec/src/lib.rs
index 45cc0c5f1..ee92ef1a0 100644
--- a/exec/src/lib.rs
+++ b/exec/src/lib.rs
@@ -2,6 +2,7 @@
 //   stdout is the final message (if any).
 // - In --json mode, stdout must be valid JSONL, one event per line.
 // For both modes, any other output must be written to stderr.
+#![recursion_limit = "256"]
 #![deny(clippy::print_stdout)]
 
 mod cli;
diff --git a/cli/src/main.rs b/cli/src/main.rs
index 22feb5f82..418241165 100644
--- a/cli/src/main.rs
+++ b/cli/src/main.rs
@@ -1,3 +1,5 @@
+#![recursion_limit = "256"]
+
 use clap::Args;
 use clap::CommandFactory;
 use clap::Parser;
