diff --git i/userspace/sysdig/CMakeLists.txt w/userspace/sysdig/CMakeLists.txt
index 4acbb0300..cc345d810 100644
--- i/userspace/sysdig/CMakeLists.txt
+++ w/userspace/sysdig/CMakeLists.txt
@@ -20,7 +20,9 @@ if(NOT WIN32)
 	include(ncurses)
 endif() # NOT WIN32
 
-include(zlib)
+if(USE_BUNDLED_FALCOSECURITY_LIBS)
+	include(zlib)
+endif()
 include(luajit)
 
 include_directories("${PROJECT_BINARY_DIR}/userspace/sinspui")
